# Raw data to variant calls

Processing Illumina reads to raw vcf file

### Data

6 Illumina HiSeq 2000 libraries. (4 libraries on HD - Elements. 2 on FGCZ server)

48 individuals pooled per library. 

9 populations - pops sequenced across at least 4 lanes each. Two additional populations (TP & GTT) were sequenced with this project, but are not part of the analyses. 

Paired-end ddRAD; 2 files per library

Barcodes and indiv names in Samples_Barcodes.txt

Restriction enzymes

```
PstI TGCAG - 3'

EcoRI AATT - 3'
```

## Demultiplex & adapter filtering

#### Demultiplex 

Samples were demultiplexed in ipyRAD. See [here](https://github.com/alexjvr1/BrownArgus/blob/master/2.ipyRADopt.ipynb). 

I compared the number of reads obtained from 0 or 1 mismatches in the barcode. This test showed that allowing 1 mismatch in the barcode allows more sequences to be assigned. Barcodes are 2+ bases different from each other, so a single mismatch won't mis-assign sequences to individuals.

The ipyRAD notebook also shows optimisation of clustering thresholds for de novo assembly. These were initially used before the A. agestis genome became available. Variants reported in the final manuscript are based on mapping to the genome and calling variants using mpileup and bcftools call. See below. 

#### Adapter filtering

