# Raw data to variant calls

Processing Illumina reads to raw vcf file

### Data

6 Illumina HiSeq 2000 libraries. (4 libraries on HD - Elements. 2 on FGCZ server)

48 individuals pooled per library. 

9 populations - pops sequenced across at least 4 lanes each. Two additional populations (TP & GTT) were sequenced with this project, but are not part of the analyses. 

Paired-end ddRAD; 2 files per library

Barcodes and indiv names in Samples_Barcodes.txt

Restriction enzymes

```
PstI TGCAG - 3'

EcoRI AATT - 3'
```

## Demultiplex & adapter filtering

#### Demultiplex 

Samples were demultiplexed in ipyRAD. See [here](https://github.com/alexjvr1/BrownArgus/blob/master/2.ipyRADopt.ipynb). 

I compared the number of reads obtained from 0 or 1 mismatches in the barcode. This test showed that allowing 1 mismatch in the barcode allows more sequences to be assigned. Barcodes are 2+ bases different from each other, so a single mismatch won't mis-assign sequences to individuals.

The ipyRAD notebook also shows optimisation of clustering thresholds for de novo assembly. These were initially used before the A. agestis genome became available. Variants reported in the final manuscript are based on mapping to the genome and calling variants using mpileup and bcftools call. See below. 

#### Adapter filtering

Adapters were trimmed using Trimmomatic v0.36: 

```
for i in *R1.fastq.gz; do java -jar Trimmomatic/0.36/trimmomatic-0.36.jar PE -threads 10 -trimlog test.log -basein $i -baseout $i ILLUMINACLIP:Trimmomatic/0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36; done

```


## Demultiplexed reads to variants

#### Draft genome

A draft A.agestis genome has been generated for the [Velocity project](https://gtr.ukri.org/projects?ref=NE/N015711/1). 
Data is from a paired-end HiSeq run. Initial assembly with Discovar (denovo). 

Assessment of this genome on the Velocity project GitPage. Here I plot the cumulative size of the genome represented by the scaffolds ordered from large to small. 

```
/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/03_variants
/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/RefGenome

##check the number of contigs in the Reference genome

cat Aricia_agestis_Red_MESPA.fasta.fai |wc -l
93568

##Check that all these contigs have been used during variant calling
grep "#contig" variants.raw.vcf |wc -l
93568  

##Read the contig lengths into R and plot the lengths as a proportion of the total estimated genome length. This is done on my personal mac so dataframes were copied to local computer. 

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RefGenome_RawSeqStats

R version 3.5.0 (2018-04-23) -- "Joy in Playing"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin15.6.0 (64-bit)

library(dplyr)
library(ggplot2)

contig.length <- read.table("Aricia_agestis_Red_MESPA.fasta.fai", header=F)
contig.length$prop.TotGenome <- (contig.length$V2/371368629)*100
colnames(contig.length) <- c("scaffold", "length", "bytesindex.fasta", "length.fasta", "bytes", "prop.GenomeTot")
contig.length <- contig.length[order(-contig.length$prop.GenomeTot),]
head(contig.length)
contig.length$index <- (1:nrow(contig.length))
contig.length2 <- contig.length %>% mutate(cumsum=cumsum(prop.GenomeTot))

pdf("C3_AriciaAgestis_CumulativeGenome_scaffolds.pdf")
ggplot(contig.length2, aes(x=index, y=cumsum)) + geom_line() + ggtitle("Aricia Agestis Reference Genome (predicted 371Mb)") + ylab("Cumulative fraction of genome (%)") + xlab("Number of scaffolds")
dev.off()
```


![alt_txt][cumGenTot]

[cumGenTot]:https://user-images.githubusercontent.com/12142475/53176921-f921c180-35e6-11e9-83ec-01245bfc9017.png




#### Map to draft genome

I used [this script](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/MapwithBWAmem.ARRAY.sh) to map all reads to the draft genome using BWA mem. Default parameters were used. 
Min mapping quality of Phred 20

#### Call variants

Variants were called using samtools mpileup and bcftools call functions using this [script](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/call_SNVs_bluecrystal.pl)

The dataset is broken up into sets of ~5000 contigs and variants are called simultaneously across all indivs for a locus. These are concatenated into a single bcffile for the entire dataset, and then converted to vcf. 

There are almost no filters at this point: only minMQS Phred > 20. 

ie. loci are *not* filtered on min and max depth. Everything is kept in the final raw vcf file. 

The final raw vcf file after excluding TP and GTT individuals: 
```
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf variants.raw.vcf
	--exclude removeTP.GTT.names
	--recode-INFO-all
	--out AAgestis.276.raw
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 276 out of 286 Individuals
Outputting VCF file...
After filtering, kept 282092 out of a possible 282092 Sites
Run Time = 270.00 seconds


```

#### Assess the raw data and unfiltered variant calls

*Mean fragment size per sample*

I calculated this from the cleaned fastq.gz files. I.e. the demultiplexed files after adapter trimming
```
for i in $(ls *fastq.gz); do awk '{if(NR%4==2) {count++; bases += length} } END{print bases/count}' $i; done > 1a_AAgestis.PopGen_mean.readlength_cleanedData

ls *fastq.gz > all.samples
```

Copy this to personal computer and draw a graph of read length per population
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RefGenome_RawSeqStats

read.length <- read.table("1a_AAgestis.PopGen_mean.readlength_cleanedData", header=F)
all.samples.names <- read.table("all.samples", header=F)
total <- data.frame(all.samples.names$V1, read.length$V1)  #bind into one dataframe

total$pop <- total$all.samples.names.V1 #add pop info
total$pop <- gsub("_.*", "", total$pop)
head(total)

total$read.nr <- total$all.samples.names.V1  ##add fwd vs rev read info
total$read.nr <- gsub("_.fastq.gz", "", total$read.nr)
total$read.nr <- gsub("^.*_", "", total$read.nr)
head(total)

pdf("1a_Aagestis_lengthRawReads.pdf")
ggplot(total, aes(x=pop, y=read.length.V1, colour=read.nr)) + geom_boxplot() + ggtitle("1a Aricia agestis - Length of cleaned raw reads") + ylab("Mean read length (bp)") + xlab("population")
dev.off()
```

![alt_txt][length_raw_reads]

[length_raw_reads]:https://user-images.githubusercontent.com/12142475/53186907-df3daa00-35f9-11e9-8af8-6b1eaf94e0a8.png




*Mean depth per site per population* 

Individual mean depth vs number of genotyped loci. Coloured per population. 

Coverage per individual and per pop

Number of variants per population (with individual variation)

Number of scaffolds with variants
	
	scaffold length vs number of variants. 

