# Raw data to variant calls

Processing Illumina reads to raw vcf file

### Data

6 Illumina HiSeq 2000 libraries. (4 libraries on HD - Elements. 2 on FGCZ server)

48 individuals pooled per library. 

9 populations - pops sequenced across at least 4 lanes each. Two additional populations (TP & GTT) were sequenced with this project, but are not part of the analyses. 

Paired-end ddRAD; 2 files per library

Barcodes and indiv names in Samples_Barcodes.txt

Restriction enzymes

```
PstI TGCAG - 3'

EcoRI AATT - 3'
```

## Demultiplex & adapter filtering

#### Demultiplex 

Samples were demultiplexed in ipyRAD. See [here](https://github.com/alexjvr1/BrownArgus/blob/master/2.ipyRADopt.ipynb). 

I compared the number of reads obtained from 0 or 1 mismatches in the barcode. This test showed that allowing 1 mismatch in the barcode allows more sequences to be assigned. Barcodes are 2+ bases different from each other, so a single mismatch won't mis-assign sequences to individuals.

The ipyRAD notebook also shows optimisation of clustering thresholds for de novo assembly. These were initially used before the A. agestis genome became available. Variants reported in the final manuscript are based on mapping to the genome and calling variants using mpileup and bcftools call. See below. 

#### Adapter filtering

Adapters were trimmed using Trimmomatic v0.36: 

```
for i in *R1.fastq.gz; do java -jar Trimmomatic/0.36/trimmomatic-0.36.jar PE -threads 10 -trimlog test.log -basein $i -baseout $i ILLUMINACLIP:Trimmomatic/0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36; done

```


## Demultiplexed reads to variants

#### Draft genome

A draft A.agestis genome has been generated for the [Velocity project](https://gtr.ukri.org/projects?ref=NE/N015711/1). 
Data is from a paired-end HiSeq run. Initial assembly with Discovar (denovo). 

#### Map to draft genome

I used [this script](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/MapwithBWAmem.ARRAY.sh) to map all reads to the draft genome using BWA mem. Default parameters were used. 
Min mapping quality of Phred 20

#### Call variants

Variants were called using samtools mpileup and bcftools call functions using this [script](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/call_SNVs_bluecrystal.pl)

The dataset is broken up into sets of ~5000 contigs and variants are called simultaneously across all indivs for a locus. These are concatenated into a single bcffile for the entire dataset, and then converted to vcf. 

There are almost no filters at this point: only minMQS Phred > 20. 

ie. loci are *not* filtered on min and max depth. Everything is kept in the final raw vcf file. 

The final raw vcf file after excluding TP and GTT individuals: 
```
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf variants.raw.vcf
	--exclude removeTP.GTT.names
	--recode-INFO-all
	--out AAgestis.276.raw
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 276 out of 286 Individuals
Outputting VCF file...
After filtering, kept 282092 out of a possible 282092 Sites
Run Time = 270.00 seconds


```

#### Assess the raw data and unfiltered variant calls

Mean fragment size per sample. 

Mean depth per site per population. 

Individual mean depth vs number of genotyped loci. Coloured per population. 

Coverage per individual and per pop

Number of variants per population (with individual variation)

Number of scaffolds with variants
	
	scaffold length vs number of variants. 

