# Filtering raw SNP file 

Filtering variants from the raw vcf output to produce two datasets: 

1) Population structure analyses (little missing data)

2) Signals of selection (include as many loci as possible). 


### Filters to apply

1) Remove loci with QUAL < 100 (i.e. Phred confidence in variant site)

2) Minimun mean depth of 6 (i.e. remove loci with lower than mean 6x depth)

3) Remove all loci genotyped in <30% of individuals

4) Remove individuals with >60% missingness

5) Remove loci with <5% MAF



##### Dataset1
```
vcftools --vcf AAgestis.276.raw.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.276.raw.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--out AAgestis_FINAL
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 66500 out of a possible 282092 Sites
Run Time = 162.00 seconds

```

Remove individuals with a high proportion of missingness. 

```
vcftools --vcf AAgestis_FINAL.recode.vcf --missing-indv

awk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF

```


![alt_txt][missing1]

[missing1]:https://user-images.githubusercontent.com/12142475/52953422-57e50200-337f-11e9-82ba-f5acce21e50f.png



##### Dataset2

MAF 5%
```

```
