# Filtering raw SNP file 

Filtering variants from the raw vcf output to produce two datasets: 

1) Population structure analyses (little missing data)

2) Signals of selection (include as many loci as possible). 


### Filters to apply

1) Remove loci with QUAL < 100 (i.e. Phred confidence in variant site)

2) Minimum mean depth of 10 (i.e. remove loci with lower than mean 6x depth)

3) Max mean depth of mean + 2xSD of meanDP (here 255)

4) remove all multi-allelic SNPs

5) Remove all loci genotyped in <30% of individuals

6) Remove individuals with >60% missingness

Finally 

7) Remove loci with <1% MAF (to exclude any errors).

8) Filter for <5% MAF when doing outlier detection. 



##### Dataset1
```
#find meanDP with --depth flag in vcftools

vcftools --vcf AAgestis.276.raw.recode.vcf --minQ 100 --min-meanDP 10 --max-meanDP 255 --max-alleles 2 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.276.raw.recode.vcf
	--recode-INFO-all
	--max-alleles 2
	--max-meanDP 255
	--min-meanDP 10
	--minQ 100
	--max-missing 0.3
	--out AAgestis_FINAL
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 282092 Sites
Run Time = 151.00 seconds

```

Remove individuals with a high proportion of missingness. 

```
vcftools --vcf AAgestis_FINAL.recode.vcf --missing-indv

awk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF

```

The vast majority of the samples have <60% missing data. 


![alt_txt][missing1]

[missing1]:https://user-images.githubusercontent.com/12142475/52953422-57e50200-337f-11e9-82ba-f5acce21e50f.png



Remove individuals with >60% missingness
```
awk '$5>0.6 {print $1}' out.imiss > indivstoremove

cat indivstoremove

INDV
BAR_1_2013_R1_.fastq.gz
BAR_23_2014_R1_.fastq.gz
BCH_38_2013_R1_.fastq.gz
BRO_4_2014_R1_.fastq.gz
HOD_13_2014_R1_.fastq.gz
HOD_18_2014_R1_.fastq.gz
HOD_6_2014_R1_.fastq.gz
HOD_8_2014_R1_.fastq.gz
LYD_20_2014_R1_.fastq.gz
LYD_28_2014_R1_.fastq.gz
MOF_42_2014_R1_.fastq.gz
SWD_17_2013_R1_.fastq.gz
```


Remove samples
```
vcftools --vcf AAgestis_FINAL.recode.vcf --remove indivstoremove --recode --recode-INFO-all --out AAgestis.264_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis_FINAL.recode.vcf
	--exclude indivstoremove
	--recode-INFO-all
	--out AAgestis.264_FINAL
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 264 out of 276 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 63054 Sites
Run Time = 47.00 seconds

```


And check if the filters remove any more loci: 
```
vcftools --vcf AAgestis.264_FINAL.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis.264.63054_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264_FINAL.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--out AAgestis.264.63054_FINAL
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 63054 Sites
Run Time = 51.00 seconds

```

Rename the individuals in this file
```
module load apps/bcftools-1.8

bcftools query -l AAgestis.264_FINAL.recode.vcf > A264.oldnames

sed 's/_R1_fastq.gz//g' A264.oldnames > A264.newnames

bcftools reheader AAgestis.264_FINAL.recode.vcf -s A264.newnames -o AAgestis.264_FINAL.newnames.vcf

```

And filter for MAF 1%
```
vcftools --vcf AAgestis.264_FINAL.newnames.vcf --maf 0.01 --recode --recode-INFO-all --out AA.264.41508_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264_FINAL.newnames.vcf
	--recode-INFO-all
	--maf 0.01
	--out AA.264.41508_FINAL
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 41508 out of a possible 63054 Sites
Run Time = 35.00 seconds

```



#### Assess the data

These stats are based on the dataset before filtering for MAF 1% (66054 loci)

*Mean depth per site per population* 

Individual mean depth vs number of genotyped loci. Coloured per population. 

Coverage per individual and per pop


vcftools finds these statistics. 

[OUTPUT DEPTH STATISTICS](http://vcftools.sourceforge.net/man_latest.html)

--depth

Generates a file containing the mean depth per individual. This file has the suffix ".idepth".

--site-depth

Generates a file containing the depth per site summed across all individuals. This output file has the suffix ".ldepth".

--site-mean-depth

Generates a file containing the mean depth per site averaged across all individuals. This output file has the suffix ".ldepth.mean".


1. Plot of mean depth per individual grouped by population: raw data first, and then filtered


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/StatsReseqData

R version 3.5.0 (2018-04-23) -- "Joy in Playing"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin15.6.0 (64-bit)

DP.indivs <- read.table("out.idepth_vcf.raw", header=T)
head(DP.indivs)
DP.indivs$pop <- gsub("_.*gz", "", DP.indivs$INDV)
DP.indivs$dataset <- "raw"
head(DP.indivs)

DP.indivs.filtered <- read.table("out.idepth", header=T)
head(DP.indivs.filtered)
DP.indivs.filtered$pop <- gsub("_.*_.*", "", DP.indivs.filtered$INDV)
DP.indivs.filtered$dataset <- "filtered"
head(DP.indivs.filtered)

Tot.DP.indivs <- rbind(DP.indivs, DP.indivs.filtered)

pdf("1a_AriciaAgestis_MeanDP.pdf")
ggplot(Tot.DP.indivs, aes(x=pop, y=MEAN_DEPTH, colour=dataset)) + geom_boxplot()
dev.off()
```

![alt_txt][meanDP]

[meanDP]:https://user-images.githubusercontent.com/12142475/53504646-3b954380-3aaa-11e9-9255-b6e5503f1744.png




2. Mean depth per site in raw & filtered data


```
DP.sites <- read.table("out.ldepth_vcf.raw", header=T)
DP.sites.filtered <- read.table("out.ldepth", header=T)
DP.sites.filtered$dataset <- "filtered"
DP.sites$dataset <- "raw"
TotDP.sites <- rbind(DP.sites, DP.sites.filtered)

pdf("a1_AriciaAgestis_DepthPerSite.pdf")
ggplot(TotDP.sites, aes(SUM_DEPTH, colour=dataset)) + geom_histogram()
dev.off()


summary(DP.sites.filtered)
         CHROM            POS           SUM_DEPTH      SUMSQ_DEPTH      
 contig_365  :   82   Min.   :     4   Min.   : 2518   Min.   :   59081  
 m_scaff_3898:   80   1st Qu.:  2275   1st Qu.:13229   1st Qu.: 1643958  
 m_scaff_714 :   80   Median :  5135   Median :29046   Median : 5223654  
 contig_2942 :   79   Mean   :  7790   Mean   :30056   Mean   : 6068789  
 m_scaff_5830:   78   3rd Qu.: 10551   3rd Qu.:45418   3rd Qu.: 9723112  
 contig_16698:   77   Max.   :147561   Max.   :68598   Max.   :18003826  
 (Other)     :62578                                                      
   dataset         
 Length:63054      
 Class :character  
 Mode  :character  
                   
                   
summary(DP.sites)
          CHROM             POS           SUM_DEPTH      SUMSQ_DEPTH      
 m_scaff_6129:   176   Min.   :     4   Min.   :    1   Min.   :       1  
 m_scaff_5252:   168   1st Qu.:  2197   1st Qu.:    4   1st Qu.:       9  
 m_scaff_6232:   153   Median :  5058   Median :   39   Median :     226  
 contig_2942 :   152   Mean   :  7780   Mean   : 7724   Mean   : 1528150  
 m_scaff_910 :   150   3rd Qu.: 10497   3rd Qu.: 4038   3rd Qu.:  390659  
 m_scaff_714 :   144   Max.   :147699   Max.   :71626   Max.   :18744992  
 (Other)     :281149                                                      

```

![alt_txt][meanDP_sites]

[meanDP_sites]:https://user-images.githubusercontent.com/12142475/53504859-a9416f80-3aaa-11e9-92d6-73c1f1969d9d.png




*Number of variants per population (with individual variation)*


```
out.imiss <- read.table("out.imiss", header=T)
out.imiss$N_Variants <- (out.imiss$N_DATA-out.imiss$N_MISS)
out.imiss$pop <- gsub("_.*_.*", "", out.imiss$INDV)

pdf("1a_AriciaAgestis_nrVariantsPerPop.pdf")
ggplot(out.imiss, aes(x=pop, y=N_Variants)) + geom_boxplot()
dev.off()
```

![alt_txt][Nvariants]

[Nvariants]:https://user-images.githubusercontent.com/12142475/53504951-d8f07780-3aaa-11e9-9efc-ad74416d245a.png


There's some variance in the number of variants between populations. Most concerning is FOR which seems to have several indivdiduals with a low number of variants. I need to check whether this is a problem with the data before I continue. 

An obvious problem could be coverage per sample: 


*Nr variants vs Depth*

```
Tot.filtered <- merge(out.imiss, DP.indivs.filtered, by="INDV")
head(Tot.filtered)


pdf("1a_AriciaAgestis_variantsVsDepth.pdf")
ggplot(Tot.filtered, aes(x=N_Variants, y=MEAN_DEPTH, colour=pop.x)) + geom_point()
dev.off()
```

![alt_txt][variants_DP]

[variants_DP]:https://user-images.githubusercontent.com/12142475/53505386-9ed3a580-3aab-11e9-8c81-ba6671317704.png



And depth vs het
```
het.264 <- read.table("out.het", header=T)
het.264$het <- (het.264$N_SITES-het.264$O.HOM.)/het.264$N_SITES
het.264$pop <- gsub("_.*_.*", "", het.264$INDV)
head(het.264)
Tot.filtered$het <- het.264$het


```

![atl_txt][Tot.filtered_DPvshet]

[Tot.filtered_DPvshet]:https://user-images.githubusercontent.com/12142475/53509175-2f61b400-3ab3-11e9-9465-5577737ba756.png



There's a worrying correlation between MEAN_DEPTH and number of variants called in an individual. However, the mean depth is really high (>50x), and I've filtered for minQ (variant quality) PHRED 100. 
However, I can't filter individual genotypes without GQ (Genotype Quality) flag in the vcf file. 


When I filter the dataset to include only unlinked loci (assuming a max ddRAD locus of 600bp based on the insert size in the library prep): 

```
vcftools --vcf AAgestis.264.thin600.recode.vcf --het

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.thin600.recode.vcf
	--het

After filtering, kept 264 out of 264 Individuals
Outputting Individual Heterozygosity
After filtering, kept 5625 out of a possible 5625 Sites
Run Time = 1.00 seconds

```

And then look at depth vs N_variants: 
```
out.het.unlinked <- read.table("AAgestis.264.thin600/out.het", header=T)
out.het.unlinked$het <- (out.het.unlinked$N_SITES-out.het.unlinked$O.HOM.)/out.het.unlinked$N_SITES
out.het.unlinked$pop <- gsub("_.*_.*", "", out.het.unlinked$INDV)

ggplot(out.het.unlinked, aes(N_SITES, MEAN_DEPTH)) + geom_point()

```

![alt_txt][unlinked_DPTHvsN_VAR]

[unlinked_DPTHvsN_VAR]:https://user-images.githubusercontent.com/12142475/53508333-7189f600-3ab1-11e9-94ce-197347ce24bc.png


and N-variants vs het:
```
ggplot(out.het.unlinked, aes(N_SITES, het)) + geom_point()

```

![alt_txt][unlinked_het]

[unlinked_het]:https://user-images.githubusercontent.com/12142475/53509025-db56cf80-3ab2-11e9-8dba-659e659ffde4.png


But the het doesn't increase with MEAN_DEPTH
```
ggplot(out.het.unlinked, aes(MEAN_DEPTH, het)) + geom_point()

```

![alt_txt][unlinked_meandpvshet]

[unlinked_meandpvshet]:https://user-images.githubusercontent.com/12142475/53508240-430c1b00-3ab1-11e9-94a8-555ce8c7b0c7.png




*F vs Nvariants*

F (inbreeding) vs Number of variants. 

![alt_txt][meanHet]

[meanHet]:https://user-images.githubusercontent.com/12142475/53207386-7c650680-362b-11e9-94e7-56c321d5d8b5.png





#### Population genetics statistics

These stats are based on the FINAL dataset which includes 264 individuals, 41508 loci. 

##### 1. Number of polymorphic sites (and genomic resolution) per indiv per pop


```

```
	
##### 2. Genome-wide nucleotide diversity (measured in windows)


	mean and variance per population
	
	
	
##### 3. Nucleotide diversity vs nr markers & DP

	Is the number of loci reflecting true diversity?
	




##### 4. Heterozygosity
	
Diversity measures per population. This can be calculated in hierfstat. 
To compare between populations I will keep only loci that are genotyped in at least 50% of individuals across all populations.
	

```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

library(adegenet)
library(vcfR)
library(hierfstat)

AA264 <- read.vcfR("AA.264.41508_FINAL.recode.vcf", nrows=41508)
Scanning file to determine attributes.
File attributes:
  meta lines: 93609
  header_line: 93610
  variant count: 41508
  column count: 273
Meta line 93609 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 41508
  Character matrix gt cols: 273
  skip: 0
  nrows: 41508
  row_num: 0
Processed variant: 41508
All variants processed

AA.genind <- vcfR2genind(AA264)  ##convert to genind
AA264.pop <- read.table("AA.264.names", header=F) ## add population information
AA264.pop$pop <- gsub("_.*_.*", "", AA264.pop$V1)
AA.genind@pop < AA264.pop$pop

hier.AA264 <- genind2hierfstat(AA.genind) ## convert to hierfstat
AA264.stats <- basic.stats(hier.AA264) ## calculate basic statistics (gene diversity, etc). 


```

Is this significantly different between populations? 




##### 5. Linkage disequilibrium

	Estimate r2 in windows of 25 variants (?). Assess only contigs that have this variation. 



##### 6. Genome-wide Tajima's D

	Affected by demographic history
	



	
