# Filtering raw SNP file 

Filtering variants from the raw vcf output to produce two datasets: 

1) Population structure analyses (little missing data)

2) Signals of selection (include as many loci as possible). 



#Filtering has been redone using reads mapped to the Sanger genome. See new filter sets [here](https://github.com/alexjvr1/AriciaAgestis_GWASMS/blob/main/2.VariantFiltering.md)



Final dataset for PopGen statistics (MAF 1%):

251 individuals

84117 variants

10097 loci



Final dataset for Outlier analyses (MAF 5%):

251 individuals

35463 variants

6785 loci






# See below for filtering on old dataset using deNovo Reference genome  ######################

### Filters to apply

1) Remove loci with QUAL < 100 (i.e. Phred confidence in variant site)

2) Minimum mean depth of 10 (i.e. remove loci with lower than mean 6x depth)

3) Max mean depth of mean + 2xSD of meanDP (here 255)

4) remove all multi-allelic SNPs

5) Remove all loci genotyped in <30% of individuals

6) Remove individuals with >60% missingness

Finally 

7) Remove loci with <1% MAF (to exclude any errors).

8) Filter for <5% MAF when doing outlier detection. 



##### Dataset1
```
#find meanDP with --depth flag in vcftools

vcftools --vcf AAgestis.276.raw.recode.vcf --minQ 100 --min-meanDP 10 --max-meanDP 255 --max-alleles 2 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.276.raw.recode.vcf
	--recode-INFO-all
	--max-alleles 2
	--max-meanDP 255
	--min-meanDP 10
	--minQ 100
	--max-missing 0.3
	--out AAgestis_FINAL
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 282092 Sites
Run Time = 151.00 seconds

```

Remove individuals with a high proportion of missingness. 

```
vcftools --vcf AAgestis_FINAL.recode.vcf --missing-indv

awk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF

```

The vast majority of the samples have <60% missing data. 


![alt_txt][missing1]

[missing1]:https://user-images.githubusercontent.com/12142475/52953422-57e50200-337f-11e9-82ba-f5acce21e50f.png



Remove individuals with >60% missingness
```
awk '$5>0.6 {print $1}' out.imiss > indivstoremove

cat indivstoremove

INDV
BAR_1_2013_R1_.fastq.gz
BAR_23_2014_R1_.fastq.gz
BCH_38_2013_R1_.fastq.gz
BRO_4_2014_R1_.fastq.gz
HOD_13_2014_R1_.fastq.gz
HOD_18_2014_R1_.fastq.gz
HOD_6_2014_R1_.fastq.gz
HOD_8_2014_R1_.fastq.gz
LYD_20_2014_R1_.fastq.gz
LYD_28_2014_R1_.fastq.gz
MOF_42_2014_R1_.fastq.gz
SWD_17_2013_R1_.fastq.gz
```


Remove samples
```
vcftools --vcf AAgestis_FINAL.recode.vcf --remove indivstoremove --recode --recode-INFO-all --out AAgestis.264_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis_FINAL.recode.vcf
	--exclude indivstoremove
	--recode-INFO-all
	--out AAgestis.264_FINAL
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 264 out of 276 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 63054 Sites
Run Time = 47.00 seconds

```


And check if the filters remove any more loci: 
```
vcftools --vcf AAgestis.264_FINAL.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis.264.63054_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264_FINAL.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--out AAgestis.264.63054_FINAL
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 63054 out of a possible 63054 Sites
Run Time = 51.00 seconds

```

Rename the individuals in this file
```
module load apps/bcftools-1.8

bcftools query -l AAgestis.264_FINAL.recode.vcf > A264.oldnames

sed 's/_R1_fastq.gz//g' A264.oldnames > A264.newnames

bcftools reheader AAgestis.264_FINAL.recode.vcf -s A264.newnames -o AAgestis.264_FINAL.newnames.vcf

```




#### Assess the data

These stats are based on the dataset before filtering for MAF 1% (66054 loci)

*Mean depth per site per population* 

Individual mean depth vs number of genotyped loci. Coloured per population. 

Coverage per individual and per pop


vcftools finds these statistics. 

[OUTPUT DEPTH STATISTICS](http://vcftools.sourceforge.net/man_latest.html)

--depth

Generates a file containing the mean depth per individual. This file has the suffix ".idepth".

--site-depth

Generates a file containing the depth per site summed across all individuals. This output file has the suffix ".ldepth".

--site-mean-depth

Generates a file containing the mean depth per site averaged across all individuals. This output file has the suffix ".ldepth.mean".


1. Plot of mean depth per individual grouped by population: raw data first, and then filtered


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/StatsReseqData

R version 3.5.0 (2018-04-23) -- "Joy in Playing"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin15.6.0 (64-bit)

DP.indivs <- read.table("out.idepth_vcf.raw", header=T)
head(DP.indivs)
DP.indivs$pop <- gsub("_.*gz", "", DP.indivs$INDV)
DP.indivs$dataset <- "raw"
head(DP.indivs)

DP.indivs.filtered <- read.table("out.idepth", header=T)
head(DP.indivs.filtered)
DP.indivs.filtered$pop <- gsub("_.*_.*", "", DP.indivs.filtered$INDV)
DP.indivs.filtered$dataset <- "filtered"
head(DP.indivs.filtered)

Tot.DP.indivs <- rbind(DP.indivs, DP.indivs.filtered)

pdf("1a_AriciaAgestis_MeanDP.pdf")
ggplot(Tot.DP.indivs, aes(x=pop, y=MEAN_DEPTH, colour=dataset)) + geom_boxplot()
dev.off()
```

![alt_txt][meanDP]

[meanDP]:https://user-images.githubusercontent.com/12142475/53504646-3b954380-3aaa-11e9-9255-b6e5503f1744.png




2. Mean depth per site in raw & filtered data


```
DP.sites <- read.table("out.ldepth_vcf.raw", header=T)
DP.sites.filtered <- read.table("out.ldepth", header=T)
DP.sites.filtered$dataset <- "filtered"
DP.sites$dataset <- "raw"
TotDP.sites <- rbind(DP.sites, DP.sites.filtered)

pdf("a1_AriciaAgestis_DepthPerSite.pdf")
ggplot(TotDP.sites, aes(SUM_DEPTH, colour=dataset)) + geom_histogram()
dev.off()


summary(DP.sites.filtered)
         CHROM            POS           SUM_DEPTH      SUMSQ_DEPTH      
 contig_365  :   82   Min.   :     4   Min.   : 2518   Min.   :   59081  
 m_scaff_3898:   80   1st Qu.:  2275   1st Qu.:13229   1st Qu.: 1643958  
 m_scaff_714 :   80   Median :  5135   Median :29046   Median : 5223654  
 contig_2942 :   79   Mean   :  7790   Mean   :30056   Mean   : 6068789  
 m_scaff_5830:   78   3rd Qu.: 10551   3rd Qu.:45418   3rd Qu.: 9723112  
 contig_16698:   77   Max.   :147561   Max.   :68598   Max.   :18003826  
 (Other)     :62578                                                      
   dataset         
 Length:63054      
 Class :character  
 Mode  :character  
                   
                   
summary(DP.sites)
          CHROM             POS           SUM_DEPTH      SUMSQ_DEPTH      
 m_scaff_6129:   176   Min.   :     4   Min.   :    1   Min.   :       1  
 m_scaff_5252:   168   1st Qu.:  2197   1st Qu.:    4   1st Qu.:       9  
 m_scaff_6232:   153   Median :  5058   Median :   39   Median :     226  
 contig_2942 :   152   Mean   :  7780   Mean   : 7724   Mean   : 1528150  
 m_scaff_910 :   150   3rd Qu.: 10497   3rd Qu.: 4038   3rd Qu.:  390659  
 m_scaff_714 :   144   Max.   :147699   Max.   :71626   Max.   :18744992  
 (Other)     :281149                                                      

```

![alt_txt][meanDP_sites]

[meanDP_sites]:https://user-images.githubusercontent.com/12142475/53504859-a9416f80-3aaa-11e9-92d6-73c1f1969d9d.png




*Number of variants per population (with individual variation)*


```
out.imiss <- read.table("out.imiss", header=T)
out.imiss$N_Variants <- (out.imiss$N_DATA-out.imiss$N_MISS)
out.imiss$pop <- gsub("_.*_.*", "", out.imiss$INDV)

pdf("1a_AriciaAgestis_nrVariantsPerPop.pdf")
ggplot(out.imiss, aes(x=pop, y=N_Variants)) + geom_boxplot()
dev.off()
```

![alt_txt][Nvariants]

[Nvariants]:https://user-images.githubusercontent.com/12142475/53504951-d8f07780-3aaa-11e9-9efc-ad74416d245a.png


There's some variance in the number of variants between populations. Most concerning is FOR which seems to have several indivdiduals with a low number of variants. I need to check whether this is a problem with the data before I continue. 

An obvious problem could be coverage per sample: 


*Nr variants vs Depth*

```
Tot.filtered <- merge(out.imiss, DP.indivs.filtered, by="INDV")
head(Tot.filtered)


pdf("1a_AriciaAgestis_variantsVsDepth.pdf")
ggplot(Tot.filtered, aes(x=N_Variants, y=MEAN_DEPTH, colour=pop.x)) + geom_point()
dev.off()
```

![alt_txt][variants_DP]

[variants_DP]:https://user-images.githubusercontent.com/12142475/53505386-9ed3a580-3aab-11e9-8c81-ba6671317704.png



And depth vs het
```
het.264 <- read.table("out.het", header=T)
het.264$het <- (het.264$N_SITES-het.264$O.HOM.)/het.264$N_SITES
het.264$pop <- gsub("_.*_.*", "", het.264$INDV)
head(het.264)
Tot.filtered$het <- het.264$het


```

![atl_txt][Tot.filtered_DPvshet]

[Tot.filtered_DPvshet]:https://user-images.githubusercontent.com/12142475/53509175-2f61b400-3ab3-11e9-9465-5577737ba756.png



There's a worrying correlation between MEAN_DEPTH and number of variants called in an individual. However, the mean depth is really high (>50x), and I've filtered for minQ (variant quality) PHRED 100. 
However, I can't filter individual genotypes without GQ (Genotype Quality) flag in the vcf file. 


When I filter the dataset to include only unlinked loci (assuming a max ddRAD locus of 600bp based on the insert size in the library prep): 

```
vcftools --vcf AAgestis.264.thin600.recode.vcf --het

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.thin600.recode.vcf
	--het

After filtering, kept 264 out of 264 Individuals
Outputting Individual Heterozygosity
After filtering, kept 5625 out of a possible 5625 Sites
Run Time = 1.00 seconds

```

And then look at depth vs N_variants: 
```
out.het.unlinked <- read.table("AAgestis.264.thin600/out.het", header=T)
out.het.unlinked$het <- (out.het.unlinked$N_SITES-out.het.unlinked$O.HOM.)/out.het.unlinked$N_SITES
out.het.unlinked$pop <- gsub("_.*_.*", "", out.het.unlinked$INDV)

ggplot(out.het.unlinked, aes(N_SITES, MEAN_DEPTH)) + geom_point()

```

![alt_txt][unlinked_DPTHvsN_VAR]

[unlinked_DPTHvsN_VAR]:https://user-images.githubusercontent.com/12142475/53508333-7189f600-3ab1-11e9-94ce-197347ce24bc.png


and N-variants (unlinked) vs het:
```
ggplot(out.het.unlinked, aes(N_SITES, het)) + geom_point()

```

![alt_txt][unlinked_het]

[unlinked_het]:https://user-images.githubusercontent.com/12142475/53509025-db56cf80-3ab2-11e9-8dba-659e659ffde4.png


But the het doesn't increase with MEAN_DEPTH
```
ggplot(out.het.unlinked, aes(MEAN_DEPTH, het)) + geom_point()

```

![alt_txt][unlinked_meandpvshet]

[unlinked_meandpvshet]:https://user-images.githubusercontent.com/12142475/53508240-430c1b00-3ab1-11e9-94a8-555ce8c7b0c7.png




F (inbreeding) vs Number of variants. 

![alt_txt][meanHet]

[meanHet]:https://user-images.githubusercontent.com/12142475/53207386-7c650680-362b-11e9-94e7-56c321d5d8b5.png








#### Population genetics statistics

Before continuing with these analyses I will filter the dataset to include only loci genotyped in at least 50% of individuals in each population. 
I'm also filtering for MAF 1% as these loci are more likely to include errors introduced during PCR or sequencing. 

I am also filtering out individuals that are related more than would be expected under random mating. 

##### 1. MAF 1%

Filter for MAF 1%
```
vcftools --vcf AAgestis.264_FINAL.newnames.vcf --maf 0.01 --recode --recode-INFO-all --out AA.264.41508_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264_FINAL.newnames.vcf
	--recode-INFO-all
	--maf 0.01
	--out AA.264.41508_FINAL
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 41508 out of a possible 63054 Sites
Run Time = 35.00 seconds

```



##### 2. Keep only loci genotyped across all populations

To compare between populations I will keep only loci that are genotyped in at least 50% of individuals across all populations.
	
First split the vcf file by population
```
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep BCH.names --recode --recode-INFO-all --out BCH
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep LYD.names --recode --recode-INFO-all --out LYD
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep SWD.names --recode --recode-INFO-all --out SWD
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep BAR.names --recode --recode-INFO-all --out BAR
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep HOD.names --recode --recode-INFO-all --out HOD
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep MOF.names --recode --recode-INFO-all --out MOF
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep WIS.names --recode --recode-INFO-all --out WIS
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep BRO.names --recode --recode-INFO-all --out BRO
vcftools --vcf AA.264.41508_FINAL.recode.vcf --keep FOR.names --recode --recode-INFO-all --out FOR

```


Then filter each of these for 50% missingness
```
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/03_variants/PerPopVCFfile
for i in $(ls *vcf); do vcftools --vcf $i --max-missing 0.5 --recode --recode-INFO-all --out $i.maxmiss

```

bgzip all of these (this was done on the mac) and find the intersection using bcftools. This keeps only loci that are genotyped in all of the files. Various options available. See [here](https://samtools.github.io/bcftools/bcftools.html#isec)
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

for i in $(ls *vcf); do bgzip $i; done
for i in $(ls test/*gz); do tabix $i; done

bcftools isec -n 9 BAR.recode.vcf.maxmiss.recode.vcf.gz BCH.recode.vcf.maxmiss.recode.vcf.gz BRO.recode.vcf.maxmiss.recode.vcf.gz FOR.recode.vcf.maxmiss.recode.vcf.gz HOD.recode.vcf.maxmiss.recode.vcf.gz LYD.recode.vcf.maxmiss.recode.vcf.gz MOF.recode.vcf.maxmiss.recode.vcf.gz SWD.recode.vcf.maxmiss.recode.vcf.gz WIS.recode.vcf.maxmiss.recode.vcf.gz -p test
```

And merge all these files using bcftools.
```

for i in $(ls test/*gz); do tabix $i; done

bcftools merge test/0000.vcf.gz test/0001.vcf.gz test/0002.vcf.gz test/0003.vcf.gz test/0004.vcf.gz test/0005.vcf.gz test/0006.vcf.gz test/0007.vcf.gz test/0008.vcf.gz -O v > test/AA264.merge0.5missing.vcf


vcftools --vcf test/AA264.merge0.5missing.vcf 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf test/AA264.merge0.5missing.vcf

After filtering, kept 264 out of 264 Individuals
After filtering, kept 31522 out of a possible 31522 Sites
Run Time = 3.00 seconds

```


This still leaves a really big file with 31k loci! 



##### 3. Relatedness filter

Estimate relatedness using the [KING method](https://academic.oup.com/bioinformatics/article/26/22/2867/228512) implemented in vcftools.  

```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/

#first thin to include only one variant per locus

vcftools --vcf AA264.merge0.5missing.vcf --thin 600 --recode --recode-INFO-all --out
AA264.merge0.5missing.thin600

vcftools --vcf AA264.merge0.5missing.thin600.recode.vcf --relatedness2
mv out.relatedness AA264.merged.thinned.relatedness2
```

Read into R and plot
```
AA264.thin.relatedness2 <- read.table("AA264.merged.thinned.relatedness2", header=T)
head(AA264.thin.relatedness2)
AA264.thin.relatedness2$pop2 <- gsub("_.*_.*", "", AA264.thin.relatedness2$INDV2)
AA264.thin.relatedness2$pop1 <- gsub("_.*_.*", "", AA264.thin.relatedness2$INDV1)

pdf("AA264.thin.relatedness2.pdf")
ggplot(AA264.thin.relatedness2[which(AA264.thin.relatedness2$pop1==AA264.thin.relatedness2$pop2),], aes(x=pop1, y=RELATEDNESS_PHI)) + geom_boxplot()
dev.off()
```

![alt_txt][AA264.relatedness2]

[AA264.relatedness2]:https://user-images.githubusercontent.com/12142475/53795582-30b73480-3f2a-11e9-92ae-5acb79cf5390.png




Figure1 from [Manichaikul et al. 2010](https://academic.oup.com/bioinformatics/article/26/22/2867/228512) shows that there us a lot of variance in the expected measure of relatedness when using only a few markers. As this method assumes unlinked markers, the thinned dataset includes only 3831 variants. 

![alt_txt][KING.fig1]

[KING.fig1]:https://user-images.githubusercontent.com/12142475/53795669-5a705b80-3f2a-11e9-8c6a-5ada97bc8836.png


Given the small number of markers I can only accurately identify first and second order relatives. Third order relatives can't be distinguished from unrelated individuals. 
Using a (second order relative) cut-off of 0.05 I identified a few related individuals, mostly within the WIS pop: 

```
AA264.thin.relatedness2[which(AA264.thin.relatedness2$pop1==AA264.thin.relatedness2$pop2 & AA264.thin.relatedness2$RELATEDNESS_PHI>0.05 & AA264.thin.relatedness2$RELATEDNESS_PHI<0.5),]
            INDV1       INDV2 N_AaAa N_AAaa N1_Aa N2_Aa RELATEDNESS_PHI pop1
20144 BRO_24_2013  BRO_6_2013    264    101   472   435       0.0683572  BRO
58832 WIS_13_2013 WIS_15_2013    241     75   460   432       0.1020180  WIS
58840 WIS_13_2013 WIS_22_2013    260     76   460   444       0.1194690  WIS
58849 WIS_13_2013 WIS_28_2013    267    104   460   479       0.0628328  WIS
59372 WIS_16_2013 WIS_24_2013    248     86   415   471       0.0857788  WIS
```

I'm using vcftools to check the missingness within each of these individuals to choose which to remove: 
```
vcftools --vcf AA264.merge0.5missing.vcf --missing-indv

for i in $(cat related.indivs.toremove); do grep $i out.imiss ; done
BRO_24_2013	31522	0	1615	0.0512341
BRO_6_2013	31522	0	2239	0.0710298
WIS_13_2013	31522	0	1682	0.0533596
WIS_15_2013	31522	0	1444	0.0458093
WIS_22_2013	31522	0	1867	0.0592285
WIS_28_2013	31522	0	1037	0.0328977
WIS_16_2013	31522	0	3662	0.116173
WIS_24_2013	31522	0	997	0.0316287
```

I'll remove: 

BRO_6_2013, WIS_13_2013, and WIS_16_2013


And apply all the previous filters: 
```

vcftools --vcf AA.264.41508_FINAL.recode.vcf --remove related.indivs.toremove --recode --recode-INFO-all --out AA261

vcftools --vcf AA261.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --maf 0.01 --recode --recode-INFO-all --out AA261.FINAL

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.recode.vcf
	--recode-INFO-all
	--maf 0.01
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--out AA261.FINAL
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 41462 out of a possible 41508 Sites
Run Time = 30.00 seconds

```

And filter to keep only loci genotyped in 50% of individuals across all species: 
First split the vcf file by population
```
vcftools --vcf AA261.FINAL.recode.vcf --keep BCH.names --recode --recode-INFO-all --out BCH
vcftools --vcf AA261.FINAL.recode.vcf --keep LYD.names --recode --recode-INFO-all --out LYD
vcftools --vcf AA261.FINAL.recode.vcf --keep SWD.names --recode --recode-INFO-all --out SWD
vcftools --vcf AA261.FINAL.recode.vcf --keep BAR.names --recode --recode-INFO-all --out BAR
vcftools --vcf AA261.FINAL.recode.vcf --keep HOD.names --recode --recode-INFO-all --out HOD
vcftools --vcf AA261.FINAL.recode.vcf --keep MOF.names --recode --recode-INFO-all --out MOF
vcftools --vcf AA261.FINAL.recode.vcf --keep WIS.names --recode --recode-INFO-all --out WIS
vcftools --vcf AA261.FINAL.recode.vcf --keep BRO.names --recode --recode-INFO-all --out BRO
vcftools --vcf AA261.FINAL.recode.vcf --keep FOR.names --recode --recode-INFO-all --out FOR
```


Then filter each of these for 50% missingness
```
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/03_variants/PerPopVCFfile_AA261/
for i in $(ls *vcf); do vcftools --vcf $i --max-missing 0.5 --recode --recode-INFO-all --out $i.maxmiss; done

```

bgzip all of these (this was done on the mac) and find the intersection using bcftools. This keeps only loci that are genotyped in all of the files. Various options available. See [here](https://samtools.github.io/bcftools/bcftools.html#isec)
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

for i in $(ls *maxmiss.recode.vcf); do bgzip $i; done
for i in $(ls *gz); do tabix $i; done


bcftools isec -n 9 BAR.recode.vcf.maxmiss.recode.vcf.gz BCH.recode.vcf.maxmiss.recode.vcf.gz BRO.recode.vcf.maxmiss.recode.vcf.gz FOR.recode.vcf.maxmiss.recode.vcf.gz HOD.recode.vcf.maxmiss.recode.vcf.gz LYD.recode.vcf.maxmiss.recode.vcf.gz MOF.recode.vcf.maxmiss.recode.vcf.gz SWD.recode.vcf.maxmiss.recode.vcf.gz WIS.recode.vcf.maxmiss.recode.vcf.gz -p AA261.0.5miss.9popsMerged
```

And merge all these files using bcftools.
```
for i in $(ls AA261.0.5miss.9popsMerged/*.vcf); do bgzip $i; done
for i in $(ls AA261.0.5miss.9popsMerged/*gz); do tabix $i; done

bcftools merge AA261.0.5miss.9popsMerged/0000.vcf.gz AA261.0.5miss.9popsMerged/0001.vcf.gz AA261.0.5miss.9popsMerged/0002.vcf.gz AA261.0.5miss.9popsMerged/0003.vcf.gz AA261.0.5miss.9popsMerged/0004.vcf.gz AA261.0.5miss.9popsMerged/0005.vcf.gz AA261.0.5miss.9popsMerged/0006.vcf.gz AA261.0.5miss.9popsMerged/0007.vcf.gz AA261.0.5miss.9popsMerged/0008.vcf.gz -O v > AA261.0.5miss.9popsMerged/AA261.0.5miss.9popsMerged.vcf


vcftools --vcf AA261.0.5miss.9popsMerged/AA261.0.5miss.9popsMerged

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.0.5miss.9popsMerged/AA261.0.5miss.9popsMerged.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 31381 out of a possible 31381 Sites
Run Time = 3.00 seconds
```

Final dataset for PopGen statistics (MAF 1%): 

261 individuals

31381 variants

3819 loci



Final dataset for Outlier analyses (MAF 5%):

261 individuals

18338 variants

3559 loci 


