# Filtering raw SNP file 

Filtering variants from the raw vcf output to produce two datasets: 

1) Population structure analyses (little missing data)

2) Signals of selection (include as many loci as possible). 


### Filters to apply

1) Remove loci with QUAL < 100 (i.e. Phred confidence in variant site)

2) Minimum mean depth of 6 (i.e. remove loci with lower than mean 6x depth)

3) Remove all loci genotyped in <30% of individuals

4) Remove individuals with >60% missingness

5) Remove loci with <5% MAF



##### Dataset1
```
vcftools --vcf AAgestis.276.raw.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.276.raw.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--out AAgestis_FINAL
	--recode

After filtering, kept 276 out of 276 Individuals
Outputting VCF file...
After filtering, kept 66500 out of a possible 282092 Sites
Run Time = 162.00 seconds

```

Remove individuals with a high proportion of missingness. 

```
vcftools --vcf AAgestis_FINAL.recode.vcf --missing-indv

awk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF

```

The vast majority of the samples have <60% missing data. 


![alt_txt][missing1]

[missing1]:https://user-images.githubusercontent.com/12142475/52953422-57e50200-337f-11e9-82ba-f5acce21e50f.png



Remove individuals with >60% missingness
```
awk '$5>0.6 {print $1}' out.imiss > indivstoremove

cat indivstoremove

INDV
BAR_1_2013_R1_.fastq.gz
BAR_23_2014_R1_.fastq.gz
BCH_38_2013_R1_.fastq.gz
BRO_4_2014_R1_.fastq.gz
HOD_13_2014_R1_.fastq.gz
HOD_18_2014_R1_.fastq.gz
HOD_6_2014_R1_.fastq.gz
HOD_8_2014_R1_.fastq.gz
LYD_20_2014_R1_.fastq.gz
LYD_28_2014_R1_.fastq.gz
MOF_42_2014_R1_.fastq.gz
SWD_17_2013_R1_.fastq.gz
```


Remove samples
```
vcftools --vcf AAgestis_FINAL.recode.vcf --remove indivstoremove --recode --recode-INFO-all --out AAgestis.264_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis_FINAL.recode.vcf
	--exclude indivstoremove
	--recode-INFO-all
	--out AAgestis.264_FINAL
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 264 out of 276 Individuals
Outputting VCF file...
After filtering, kept 66500 out of a possible 66500 Sites
Run Time = 44.00 seconds

```


And check if the filters remove any more loci: 
```
vcftools --vcf AAgestis.264_FINAL.recode.vcf --minQ 100 --min-meanDP 6 --max-missing 0.3 --recode --recode-INFO-all --out AAgestis.264.66491_FINAL

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264_FINAL.recode.vcf
	--recode-INFO-all
	--min-meanDP 6
	--minQ 100
	--max-missing 0.3
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 66491 out of a possible 66500 Sites
Run Time = 48.00 seconds
```



##### Dataset2

MAF 5%
```
vcftools --vcf AAgestis.264.66491_FINAL.recode.vcf --maf 0.05 --recode --recode-INFO-all --out AAgestis.264.25726_FINAL2

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.66491_FINAL.recode.vcf
	--recode-INFO-all
	--maf 0.05
	--out AAgestis.264_FINAL
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 25726 out of a possible 66491 Sites
Run Time = 20.00 seconds
```



#### Population genetics statistics

number of polymorphic sites (and genomic resolution)

	per indiv per pop
	
	variable only between populations

	nd in coding vs non-coding sites
	
Genome-wide nucleotide diversity (measured in windows)

	mean and variance per population

Heterozygosity
	
	Ho vs He for each individual
	
	Het per population
	
Linkage disequilibrium

	Estimate r2 in windows of 25 variants (?). Assess only contigs that have this variation. 
	
Genome-wide Tajima's D

	Affected by demographic history
	
Fis per population


	
