# Population structure and genetic diversity



### Heterozygosity and allelic diversity







### PCA

PCA is affecteted by missing data, so I'll remove all loci with >80% missingness. I'm also filtering for MAF 0.01.


```
bluecp3:/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/PCAdapt

vcftools --vcf AAgestis.264.66491_FINAL_newnames.recode.vcf --max-missing 0.8 --maf 0.01 --recode --recode-INFO-all --out AAgestis.forPCA

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.66491_FINAL_newnames.recode.vcf
	--recode-INFO-all
	--maf 0.01
	--max-missing 0.8
	--out AAgestis.forPCA
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 28980 out of a possible 66491 Sites
Run Time = 24.00 seconds

```

Convert to plink

```
vcftools --vcf AAgestis.forPCA.recode.vcf --plink --out AAgestis.forPCA.plink

```


Copy to Mac
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/input.data

plink --file AAgestis.forPCA.plink --recode A --out AAgestis.forPCA.plink

```

Open R and run analysis
```
R version 3.5.0

library(pcadapt)

AA264 <- read.pcadapt("AAgestis.forPCA.plink.ped", type="ped")

Summary:

	- input file:				AAgestis.forPCA.plink.ped
	- output file:				/var/folders/d1/2957hj5928x_7dc6xjx_2rx40000gq/T//RtmpjH9vMm/filef3a44aebc49d.pcadapt

	- number of individuals detected:	264
	- number of loci detected:		28980

28980 lines detected.
264 columns detected.


x.AA264 <- pcadapt(AA264, K=20)

Reading file CHall.932.7744.plink.pcadapt...
Number of SNPs: 7744
Number of individuals: 932
Number of SNPs with minor allele frequency lower than 0.05 ignored: 0
2148762 out of 7217408 missing data ignored.

plot(x.AA264, option="screeplot")  ##PC for pop structure = on the steep curve
```

![alt_txt][nr_pcaclusters]

[nr_pcaclusters]:https://user-images.githubusercontent.com/12142475/53089433-7b858500-3504-11e9-950c-2b00d82f9334.png



Based on this I choose K=2. The random eigenvalues fall in the straight line, while the ones explaining structure are above this.

See vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html on choosing K.

Plot the PCA using population information


```
pop.AA264 <- read.table("AA264.cluster.pop", header=F)

summary(pop.AA264)
           V1            V2            V3        V4            V5       
 BAR_10_2013:  1   WIS    :45   geranium:118   new:116   Min.   :1.000  
 BAR_11_2014:  1   SWD    :39   rockrose:146   old:148   1st Qu.:3.000  
 BAR_12_2013:  1   BCH    :38                            Median :5.000  
 BAR_13_2014:  1   HOD    :30                            Mean   :4.727  
 BAR_14_2013:  1   BAR    :28                            3rd Qu.:7.000  
 BAR_14_2014:  1   MOF    :25                            Max.   :9.000  
 (Other)    :258   (Other):59                                           
       V6              V7                V8                  V9     
 Min.   :51.00   Min.   :-1.2189   Min.   :22.00   darkorchid3: 88  
 1st Qu.:51.62   1st Qu.:-1.0394   1st Qu.:22.00   darkorchid4: 30  
 Median :52.97   Median :-0.6404   Median :22.00   gold1      : 28  
 Mean   :52.50   Mean   :-0.4365   Mean   :22.88   goldenrod  :118  
 3rd Qu.:53.19   3rd Qu.:-0.1689   3rd Qu.:24.00                    
 Max.   :54.16   Max.   : 1.2694   Max.   :24.00                    
                                                                    
summary(pop.AA264$V2)  ##pop sizes
BAR BCH BRO FOR HOD LYD MOF SWD WIS 
 28  38  18  20  30  21  25  39  45 



poplist <- as.character(pop.AA264[,2])

plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,3]) ##pca coloured by host plant presence. 
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,4]) ##pca coloured by new vs old
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified
```


![alt_txt][PCApops]

[PCApops]:https://user-images.githubusercontent.com/12142475/53090273-c902f180-3506-11e9-9e59-9d52ab970e6a.png


![alt_txt][PCAcolHist]

[PCAcolHist]:https://user-images.githubusercontent.com/12142475/53090546-b0470b80-3507-11e9-8476-483f42f7fe83.png



![alt_txt][PCAhostplant]

[PCAcolhostplant]:https://user-images.githubusercontent.com/12142475/53090576-c785f900-3507-11e9-8056-2883892005bc.png




### fineRADstructure

[fineRADstructure](http://cichlid.gurdon.cam.ac.uk/fineRADstructure.html)[(Malinksy et al. 2018)](https://academic.oup.com/mbe/article/35/5/1284/4883220)
uses haplotype information from RAD data to infer a co-ancestry matrix. The dataset utilises linkage information, so all loci should be kept. 

To create haplotype data, all variant loci from a single scaffold is concatenated into a sequence string. These data can be extracted from the vcf file. 

I've used the [vcf2hapmatrix.py](https://github.com/pimbongaerts/radseq/blob/master/vcf2hapmatrix.py) script to convert my vcf file to an input file for fineRADstructure. 
This script assumes that the scaffolds are in order. It concatenates all the nucleotide bases found on a specific scaffold together for each individual. This means many linked loci are preserved in the final dataset. 

The file conversion uses python3, and needs the module PyVCF. Conversions are run on the mac laptop. 
```
./vcf2hapmatrix.py 

```

