# Population structure and genetic diversity


Final datasets used here: 

##### 1. Final dataset for PopGen statistics (MAF 1%):

261 individuals

31381 variants

3819 loci

##### 2. Final dataset for Outlier analyses (MAF 5%):

261 individuals

18338 variants

3559 loci




#### Pop Gen Stats



##### 1. Number of polymorphic sites (and genomic resolution) per indiv per pop

The total number of sites typed per individual can be found in the out.het and snp density estimates from vcftools 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --het

for i in $(ls *vcf.gz); do vcftools --gzvcf $i --SNPdensity 600 && mv out.snpden $i.snpden; done
```
	

And plot in R
```
snpden.9pops <- read.table("PerPopVCFfiles_AA261/9pops.snpden", header=T)
snpden.9pops$pop <- "pop"
snpden.9pops$pop[1:12990] <- "BCH"
snpden.9pops$pop[12991:24929] <- "LYD"
snpden.9pops$pop[24930:37414] <- "SWD"
snpden.9pops$pop[37415:49938] <- "BAR"
snpden.9pops$pop[49939:62333] <- "HOD"
snpden.9pops$pop[62334:75199] <- "MOF"
snpden.9pops$pop[75200:87909] <- "WIS"
snpden.9pops$pop[87910:99819] <- "BRO"
snpden.9pops$pop[99820:110692] <- "FOR"

pdf("snpDensity.9pops.pdf")
ggplot(snpden.9pops[which(snpden.9pops$VARIANTS.KB>0),], aes(x=pop, y=VARIANTS.KB)) + geom_boxplot() + ggtitle("number of SNPs (>0) per kb within each population") + ylab("Number of SNPs (per kb)") +xlab("population")
dev.off()
```

	
![alt_txt][Nsites.pop]

[Nsites.pop]:https://user-images.githubusercontent.com/12142475/53816343-4d1f9500-3f5b-11e9-89dd-27b28d036e8a.png

There is very little difference between populations. 

The interesting thing to note here are the hypervariable regions found across all populations. The top two of these: 

```
snpden.9pops[which(snpden.9pops$VARIANTS.KB>60),]
          BCH_CHROM BIN_START SNP_COUNT VARIANTS.KB pop
4979   m_scaff_5956     19200        42          70 BCH
6892   contig_17335      6600        57          95 BCH
17537  m_scaff_5956     19200        42          70 LYD
19282  contig_17335      6600        57          95 LYD
29677  m_scaff_5956     19200        42          70 SWD
31543  contig_17335      6600        57          95 SWD
42192  m_scaff_5956     19200        42          70 BAR
44102  contig_17335      6600        57          95 BAR
54733  m_scaff_5956     19200        42          70 HOD
56637  contig_17335      6600        57          95 HOD
67362  m_scaff_5956     19200        42          70 MOF
69213  contig_17335      6600        57          95 MOF
80123  m_scaff_5956     19200        42          70 WIS
81914  contig_17335      6600        57          95 WIS
92571  m_scaff_5956     19200        42          70 BRO
104312 m_scaff_5956     19200        42          70 FOR
```

When I look at these contigs in the reference genome there seems to be a high frequency of runs of a single bp in these scaffolds. All indels have been removed from this dataset, and I've applied filters for mapping quality, depth of variant calls, and minor allele frequencies. 

m_scaff_5956 occurs in the gff file and codes for several mRNA in Heliconius: 

```
grep m_scaff_5956 Aricia_agestis_Red_MESPA.gff

m_scaff_5956	ALN	cds	19212	19222	15	+	1	ID=cds37451;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 287 289 +
m_scaff_5956	ALN	cds	19465	19467	17	+	2	ID=cds37452;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 290 290 +
m_scaff_5956	ALN	cds	19721	19732	7	+	2	ID=cds37453;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 291 295 +
m_scaff_5956	ALN	cds	19784	19786	23	+	2	ID=cds37454;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 296 296 +
m_scaff_5956	ALN	cds	19910	19936	34	+	2	ID=cds37455;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 297 304 +
m_scaff_5956	ALN	cds	20015	20035	28	+	2	ID=cds37456;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 305 311 +
m_scaff_5956	ALN	cds	20100	20104	25	+	2	ID=cds37457;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 312 313 +
m_scaff_5956	ALN	cds	20143	20163	11	+	0	ID=cds37458;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 314 317 +
m_scaff_5956	ALN	cds	20381	20405	23	+	0	ID=cds37459;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 318 325 +
m_scaff_5956	ALN	cds	20507	20517	17	+	2	ID=cds37460;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 326 329 +
m_scaff_5956	ALN	cds	20706	20737	26	+	0	ID=cds37461;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 330 340 +
m_scaff_5956	ALN	cds	20826	20868	24	+	1	ID=cds37462;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 341 352 +
[aj18951

```


##### 2. Genome-wide nucleotide diversity (measured in windows)

This can be estimated using vcftools. Here I estimated nucleotide diversity in windows of 1kb. 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

vcftools --vcf AA261.0.5miss.9popsMerged.vcf --window-pi 1000

mv out.windowed.pi AA261.windowed.pi
```

And plot in R

```
winowpi.9pops_A261$pop <- "pop"
winowpi.9pops_A261$pop[1:5233] <- "BCH"
winowpi.9pops_A261$pop[5234:10069] <- "LYD"
winowpi.9pops_A261$pop[10070:15159] <- "SWD"
winowpi.9pops_A261$pop[15160:20168] <- "HOD"
winowpi.9pops_A261$pop[20169:25165] <- "MOF"
winowpi.9pops_A261$pop[25166:30278] <- "WIS"
winowpi.9pops_A261$pop[30279:34951] <- "BRO"
winowpi.9pops_A261$pop[34952:39329] <- "FOR"
winowpi.9pops_A261$pop[39330:nrow(winowpi.9pops_A261)] <- "BAR"
```

![alt_txt][NucDiv]

[NucDiv]:
	
	
##### 3. Nucleotide diversity vs nr markers & DP

Is the number of loci reflecting true diversity

```


```
	


![alt_txt][NucDivvsNVariants]

[NucDivvsNVariants]:



![alt_txt][NucDivvsDP]

[NucDivvsDP]:



##### 4. Expected heterozygosity per populations
	
	
Diversity measures per population. This can be calculated in hierfstat. 


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

library(adegenet)
library(vcfR)
library(hierfstat)

AA261 <- read.vcfR("AA261.0.5miss.9popsMerged.vcf", nrows=31381)
Scanning file to determine attributes.
File attributes:
  meta lines: 93613
  header_line: 93614
  variant count: 31381
  column count: 270
Meta line 93613 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 31381
  Character matrix gt cols: 270
  skip: 0
  nrows: 31381
  row_num: 0
Processed variant: 31381
All variants processed

AA.genind <- vcfR2genind(AA261)  ##convert to genind
AA261.pop <- read.table("AA.261.names", header=F) ## add population information
AA261.pop$pop <- gsub("_.*_.*", "", AA261.pop$V1)
AA.genind@pop <- as.factor(AA261.pop$pop)

hier.AA261 <- genind2hierfstat(AA.genind) ## convert to hierfstat
AA261.stats <- basic.stats(hier.AA261) ## calculate basic statistics (gene diversity, etc). 

```

Plot expected heterozygosity frequency per population (boxplot and histogram)

```
Hs.9pops <- melt(AA261.stats$Hs)

pdf("Hs.9pops_maxmiss0.5.pdf")
ggplot(Hs.9pops, aes(x=Var2, y=value)) + geom_boxplot() + ggtitle("Aricia agestis: Genetic diversity (Hs) per pop") +ylab("Gene diversity (Hs)") + xlab("Population")
dev.off()
```

Draw a histogram for each species separately and combine graphs in Adobe Illustrator. 

![alt_txt][Hs.allpops]

[Hs.allpops]:https://user-images.githubusercontent.com/12142475/53811655-c914df80-3f51-11e9-960a-951712a1a60b.png




Is this significantly different between populations? 

Dwass-Steel all-pairs test for significance. 
```



```


Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```

```

This seems to be driven by the MOF and FOR. These differences are only significant when at least one of these populations is included.  




##### 5. Allelic diversity

Total allelic diversity (rarefaction method) per population. 
```

```

And plot in R
```

```


Number of private alleles per population
```

```


And plot in R
```

```




##### 6. Linkage disequilibrium

As this is RAD data we can't use windows of x number of variants - this leaves us with too few scaffolds to work with. Instead I assume that all loci are independent and I calculate linkage between them. 








##### 7. Genome-wide Tajima's D

This needs to be calculated for each population separately. 

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261

```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TajimaD 1000 && mv out.TajimaD > $i.TajimaD; done

for i in $(ls *TajimaD); do rename 's/recode.vcf.maxmiss.recode.vcf.gz.//' $i; done

```
	
Read into R and plot
```


```


	



## Population Structure

### PCA

PCA is affecteted by missing data, so I'll remove all loci with >80% missingness. I'm also filtering for MAF 0.01.


```
bluecp3:/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/PCAdapt

vcftools --vcf AAgestis.264.66491_FINAL_newnames.recode.vcf --max-missing 0.8 --maf 0.01 --recode --recode-INFO-all --out AAgestis.forPCA

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.66491_FINAL_newnames.recode.vcf
	--recode-INFO-all
	--maf 0.01
	--max-missing 0.8
	--out AAgestis.forPCA
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 28980 out of a possible 66491 Sites
Run Time = 24.00 seconds

```

Convert to plink

```
vcftools --vcf AAgestis.forPCA.recode.vcf --plink --out AAgestis.forPCA.plink

```


Copy to Mac
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/input.data

plink --file AAgestis.forPCA.plink --recode A --out AAgestis.forPCA.plink

```

Open R and run analysis
```
R version 3.5.0

library(pcadapt)

AA264 <- read.pcadapt("AAgestis.forPCA.plink.ped", type="ped")

Summary:

	- input file:				AAgestis.forPCA.plink.ped
	- output file:				/var/folders/d1/2957hj5928x_7dc6xjx_2rx40000gq/T//RtmpjH9vMm/filef3a44aebc49d.pcadapt

	- number of individuals detected:	264
	- number of loci detected:		28980

28980 lines detected.
264 columns detected.


x.AA264 <- pcadapt(AA264, K=20)

Reading file CHall.932.7744.plink.pcadapt...
Number of SNPs: 7744
Number of individuals: 932
Number of SNPs with minor allele frequency lower than 0.05 ignored: 0
2148762 out of 7217408 missing data ignored.

plot(x.AA264, option="screeplot")  ##PC for pop structure = on the steep curve
```

![alt_txt][nr_pcaclusters]

[nr_pcaclusters]:https://user-images.githubusercontent.com/12142475/53089433-7b858500-3504-11e9-950c-2b00d82f9334.png



Based on this I choose K=2. The random eigenvalues fall in the straight line, while the ones explaining structure are above this.

See vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html on choosing K.

Plot the PCA using population information


```
pop.AA264 <- read.table("AA264.cluster.pop", header=F)

summary(pop.AA264)
           V1            V2            V3        V4            V5       
 BAR_10_2013:  1   WIS    :45   geranium:118   new:116   Min.   :1.000  
 BAR_11_2014:  1   SWD    :39   rockrose:146   old:148   1st Qu.:3.000  
 BAR_12_2013:  1   BCH    :38                            Median :5.000  
 BAR_13_2014:  1   HOD    :30                            Mean   :4.727  
 BAR_14_2013:  1   BAR    :28                            3rd Qu.:7.000  
 BAR_14_2014:  1   MOF    :25                            Max.   :9.000  
 (Other)    :258   (Other):59                                           
       V6              V7                V8                  V9     
 Min.   :51.00   Min.   :-1.2189   Min.   :22.00   darkorchid3: 88  
 1st Qu.:51.62   1st Qu.:-1.0394   1st Qu.:22.00   darkorchid4: 30  
 Median :52.97   Median :-0.6404   Median :22.00   gold1      : 28  
 Mean   :52.50   Mean   :-0.4365   Mean   :22.88   goldenrod  :118  
 3rd Qu.:53.19   3rd Qu.:-0.1689   3rd Qu.:24.00                    
 Max.   :54.16   Max.   : 1.2694   Max.   :24.00                    
                                                                    
summary(pop.AA264$V2)  ##pop sizes
BAR BCH BRO FOR HOD LYD MOF SWD WIS 
 28  38  18  20  30  21  25  39  45 



poplist <- as.character(pop.AA264[,2])

plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,3]) ##pca coloured by host plant presence. 
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,4]) ##pca coloured by new vs old
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified
```


![alt_txt][PCApops]

[PCApops]:https://user-images.githubusercontent.com/12142475/53090273-c902f180-3506-11e9-9e59-9d52ab970e6a.png


![alt_txt][PCAcolHist]

[PCAcolHist]:https://user-images.githubusercontent.com/12142475/53090546-b0470b80-3507-11e9-8476-483f42f7fe83.png



![alt_txt][PCAhostplant]

[PCAcolhostplant]:https://user-images.githubusercontent.com/12142475/53090576-c785f900-3507-11e9-8056-2883892005bc.png




### fineRADstructure

[fineRADstructure](http://cichlid.gurdon.cam.ac.uk/fineRADstructure.html)[(Malinksy et al. 2018)](https://academic.oup.com/mbe/article/35/5/1284/4883220)
uses haplotype information from RAD data to infer a co-ancestry matrix. The dataset utilises linkage information, so all loci should be kept. 

To create haplotype data, all variant loci from a single scaffold is concatenated into a sequence string. These data can be extracted from the vcf file. 

I've used the [vcf2hapmatrix.py](https://github.com/pimbongaerts/radseq/blob/master/vcf2hapmatrix.py) script to convert my vcf file to an input file for fineRADstructure. 
This script assumes that the scaffolds are in order. It concatenates all the nucleotide bases found on a specific scaffold together for each individual. This means many linked loci are preserved in the final dataset. 

The file conversion uses python3, and needs the module PyVCF. Conversions are run on the mac laptop. 
```
./vcf2hapmatrix.py 

```

