# Population structure and genetic diversity


Final datasets used here: 

##### 1. Final dataset for PopGen statistics (MAF 1%):

261 individuals

31381 variants

3819 loci

##### 2. Final dataset for Outlier analyses (MAF 5%):

261 individuals

18338 variants

3559 loci




#### Pop Gen Stats



##### 1. Number of polymorphic sites (and genomic resolution) per indiv per pop

The total number of sites typed per individual can be found in the out.het and snp density estimates from vcftools 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --het

for i in $(ls *vcf.gz); do vcftools --gzvcf $i --SNPdensity 600 && mv out.snpden $i.snpden; done
```
	

And plot in R
```
snpden.9pops <- read.table("PerPopVCFfiles_AA261/9pops.snpden", header=T)
snpden.9pops$pop <- "pop"
snpden.9pops$pop[1:12990] <- "BCH"
snpden.9pops$pop[12991:24929] <- "LYD"
snpden.9pops$pop[24930:37414] <- "SWD"
snpden.9pops$pop[37415:49938] <- "BAR"
snpden.9pops$pop[49939:62333] <- "HOD"
snpden.9pops$pop[62334:75199] <- "MOF"
snpden.9pops$pop[75200:87909] <- "WIS"
snpden.9pops$pop[87910:99819] <- "BRO"
snpden.9pops$pop[99820:110692] <- "FOR"

pdf("snpDensity.9pops.pdf")
ggplot(snpden.9pops[which(snpden.9pops$VARIANTS.KB>0),], aes(x=pop, y=VARIANTS.KB)) + geom_boxplot() + ggtitle("number of SNPs (>0) per kb within each population") + ylab("Number of SNPs (per kb)") +xlab("population")
dev.off()
```

	
![alt_txt][Nsites.pop]

[Nsites.pop]:https://user-images.githubusercontent.com/12142475/53816343-4d1f9500-3f5b-11e9-89dd-27b28d036e8a.png

There is very little difference between populations. 

The interesting thing to note here are the hypervariable regions found across all populations. The top two of these: 

```
snpden.9pops[which(snpden.9pops$VARIANTS.KB>60),]
          BCH_CHROM BIN_START SNP_COUNT VARIANTS.KB pop
4979   m_scaff_5956     19200        42          70 BCH
6892   contig_17335      6600        57          95 BCH
17537  m_scaff_5956     19200        42          70 LYD
19282  contig_17335      6600        57          95 LYD
29677  m_scaff_5956     19200        42          70 SWD
31543  contig_17335      6600        57          95 SWD
42192  m_scaff_5956     19200        42          70 BAR
44102  contig_17335      6600        57          95 BAR
54733  m_scaff_5956     19200        42          70 HOD
56637  contig_17335      6600        57          95 HOD
67362  m_scaff_5956     19200        42          70 MOF
69213  contig_17335      6600        57          95 MOF
80123  m_scaff_5956     19200        42          70 WIS
81914  contig_17335      6600        57          95 WIS
92571  m_scaff_5956     19200        42          70 BRO
104312 m_scaff_5956     19200        42          70 FOR
```

When I look at these contigs in the reference genome there seems to be a high frequency of runs of a single bp in these scaffolds. All indels have been removed from this dataset, and I've applied filters for mapping quality, depth of variant calls, and minor allele frequencies. 

m_scaff_5956 occurs in the gff file and codes for an mRNA in Heliconius: 

```
grep m_scaff_5956 Aricia_agestis_Red_MESPA.gff

m_scaff_5956	ALN	cds	19212	19222	15	+	1	ID=cds37451;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 287 289 +
m_scaff_5956	ALN	cds	19465	19467	17	+	2	ID=cds37452;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 290 290 +
m_scaff_5956	ALN	cds	19721	19732	7	+	2	ID=cds37453;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 291 295 +
m_scaff_5956	ALN	cds	19784	19786	23	+	2	ID=cds37454;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 296 296 +
m_scaff_5956	ALN	cds	19910	19936	34	+	2	ID=cds37455;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 297 304 +
m_scaff_5956	ALN	cds	20015	20035	28	+	2	ID=cds37456;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 305 311 +
m_scaff_5956	ALN	cds	20100	20104	25	+	2	ID=cds37457;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 312 313 +
m_scaff_5956	ALN	cds	20143	20163	11	+	0	ID=cds37458;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 314 317 +
m_scaff_5956	ALN	cds	20381	20405	23	+	0	ID=cds37459;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 318 325 +
m_scaff_5956	ALN	cds	20507	20517	17	+	2	ID=cds37460;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 326 329 +
m_scaff_5956	ALN	cds	20706	20737	26	+	0	ID=cds37461;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 330 340 +
m_scaff_5956	ALN	cds	20826	20868	24	+	1	ID=cds37462;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 341 352 +
[aj18951

```


##### 2. Genome-wide nucleotide diversity (measured in windows)

This can be estimated using vcftools - here in windows of 1kb. 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

vcftools --vcf AA261.0.5miss.9popsMerged.vcf --window-pi 1000

mv out.windowed.pi AA261.windowed.pi
```

Calculate mean and SD
```
tapply(winowpi.9pops_A261$PI, winowpi.9pops_A261$pop, mean, na.rm=T)
         BAR          BCH          BRO          FOR          HOD          LYD 
0.0010411435 0.0011215100 0.0010182363 0.0008825968 0.0010366159 0.0009963501 
         MOF          SWD          WIS 
0.0011155415 0.0010591159 0.0010719579 


tapply(winowpi.9pops_A261$PI, winowpi.9pops_A261$pop, sd, na.rm=T)
         BAR          BCH          BRO          FOR          HOD          LYD 
0.0009572797 0.0010310719 0.0009232863 0.0008207372 0.0009646674 0.0008943332 
         MOF          SWD          WIS 
0.0010311528 0.0009765223 0.0009983996 
```

And plot in R

```
winowpi.9pops_A261$pop <- "pop"
winowpi.9pops_A261$pop[1:5233] <- "BCH"
winowpi.9pops_A261$pop[5234:10069] <- "LYD"
winowpi.9pops_A261$pop[10070:15159] <- "SWD"
winowpi.9pops_A261$pop[15160:20168] <- "HOD"
winowpi.9pops_A261$pop[20169:25165] <- "MOF"
winowpi.9pops_A261$pop[25166:30278] <- "WIS"
winowpi.9pops_A261$pop[30279:34951] <- "BRO"
winowpi.9pops_A261$pop[34952:39329] <- "FOR"
winowpi.9pops_A261$pop[39330:nrow(winowpi.9pops_A261)] <- "BAR"


pdf("AA261_PIperpop.pdf")
ggplot(winowpi.9pops_A261, aes(x=pop, y=PI)) + geom_boxplot() + ggtitle("Mean nucleotide diversity measured across 1kb windows for each population") + ylab("PI") + xlab("population")
dev.off()
```

![alt_txt][NucDiv]

[NucDiv]:https://user-images.githubusercontent.com/12142475/53818593-37609e80-3f60-11e9-808e-40a5bcb68b3a.png
	
	


Check if this is significantly different between populations, and also between History and hostPlant

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=winowpi.9pops_A261$PI, g=winowpi.9pops_A261$pop, data=winowpi.9pops_A261, formula=winowpi.9pops_A261$PI~winowpi.9pops_A261$pop))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: winowpi.9pops_A261$PI and winowpi.9pops_A261$pop
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -4.487 0.04033001   *
BRO - BAR == 0  -0.495 0.99999377    
FOR - BAR == 0 -10.216 1.8217e-11 ***
HOD - BAR == 0  -0.676 0.99993058    
LYD - BAR == 0  -1.615 0.96782256    
MOF - BAR == 0  -3.961 0.11502517    
SWD - BAR == 0  -0.822 0.99969430    
WIS - BAR == 0  -0.993 0.99877014    
BRO - BCH == 0  -4.960 0.01344163   *
FOR - BCH == 0 -14.542 9.6811e-14 ***
HOD - BCH == 0  -5.143 0.00845932  **
LYD - BCH == 0  -6.176 0.00042837 ***
MOF - BCH == 0  -0.496 0.99999362    
SWD - BCH == 0  -3.685 0.18390036    
WIS - BCH == 0  -3.459 0.25917632    
FOR - BRO == 0  -9.794 1.5667e-10 ***
HOD - BRO == 0  -0.165 1.00000000    
LYD - BRO == 0  -1.109 0.99730639    
MOF - BRO == 0  -4.399 0.04868001   *
SWD - BRO == 0  -1.317 0.99121524    
WIS - BRO == 0  -1.458 0.98295736    
HOD - FOR == 0  -9.520 6.0255e-10 ***
LYD - FOR == 0  -8.769 2.0167e-08 ***
MOF - FOR == 0 -13.827 9.7811e-14 ***
SWD - FOR == 0 -10.951 4.0146e-13 ***
WIS - FOR == 0 -10.981 3.4539e-13 ***
LYD - HOD == 0  -0.938 0.99919296    
MOF - HOD == 0  -4.586 0.03246386   *
SWD - HOD == 0  -1.471 0.98193927    
WIS - HOD == 0  -1.637 0.96503030    
MOF - LYD == 0  -5.554 0.00277393  **
SWD - LYD == 0  -2.461 0.72152683    
WIS - LYD == 0  -2.595 0.65878453    
SWD - MOF == 0  -3.129 0.39745764    
WIS - MOF == 0  -2.920 0.49793426    
WIS - SWD == 0  -0.180 1.00000000    
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

winowpi.9pops_A261$History <- (winowpi.9pops_A261$pop %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))


winowpi.9pops_A261$HostPlant <- (winowpi.9pops_A261$pop %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

winowpi.9pops_A261$History <- as.factor(winowpi.9pops_A261$History)
kruskal.test(PI~History, data=winowpi.9pops_A261)


kruskal.test(PI~History, data=winowpi.9pops_A261)

	Kruskal-Wallis rank sum test

data:  PI by History
Kruskal-Wallis chi-squared = 12.062, df = 1, p-value = 0.0005147



winowpi.9pops_A261$HostPlant <- as.factor(winowpi.9pops_A261$HostPlant)
kruskal.test(PI~HostPlant, data=winowpi.9pops_A261)

	Kruskal-Wallis rank sum test

data:  PI by HostPlant
Kruskal-Wallis chi-squared = 9.1441, df = 1, p-value = 0.002495

```

This remained significant after removing one population at a time, except for MOF and FOR.
I also checked whether a ranomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 1 1 2 1 2 2 1

winowpi.9pops_A261$rand <- (winowpi.9pops_A261$pop %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "1", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "2", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
winowpi.9pops_A261$rand <- as.factor(winowpi.9pops_A261$rand)

kruskal.test(PI~rand, data=winowpi.9pops_A261)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  PI by rand
Kruskal-Wallis chi-squared = 0.47368, df = 1, p-value = 0.4913

```

Not significant. 
So there is a significant difference in nucleotide diversity, but not Het (see below) between HostPlant and History groups. But this seems to be driven by FOR and MOF. 



##### 3. Expected heterozygosity per populations
	
	
Diversity measures per population. This can be calculated in hierfstat. 


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

library(adegenet)
library(vcfR)
library(hierfstat)

AA261 <- read.vcfR("AA261.0.5miss.9popsMerged.vcf", nrows=31381)
Scanning file to determine attributes.
File attributes:
  meta lines: 93613
  header_line: 93614
  variant count: 31381
  column count: 270
Meta line 93613 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 31381
  Character matrix gt cols: 270
  skip: 0
  nrows: 31381
  row_num: 0
Processed variant: 31381
All variants processed

AA.genind <- vcfR2genind(AA261)  ##convert to genind
AA261.pop <- read.table("AA.261.names", header=F) ## add population information
AA261.pop$pop <- gsub("_.*_.*", "", AA261.pop$V1)
AA.genind@pop <- as.factor(AA261.pop$pop)

hier.AA261 <- genind2hierfstat(AA.genind) ## convert to hierfstat
AA261.stats <- basic.stats(hier.AA261) ## calculate basic statistics (gene diversity, etc). 

```

Plot expected heterozygosity frequency per population (boxplot and histogram)

```
Hs.9pops <- melt(AA261.stats$Hs)

pdf("Hs.9pops_maxmiss0.5.pdf")
ggplot(Hs.9pops, aes(x=Var2, y=value)) + geom_boxplot() + ggtitle("Aricia agestis: Genetic diversity (Hs) per pop") +ylab("Gene diversity (Hs)") + xlab("Population")
dev.off()
```

Draw a histogram for each species separately and combine graphs in Adobe Illustrator. 

![alt_txt][Hs.allpops]

[Hs.allpops]:https://user-images.githubusercontent.com/12142475/53811655-c914df80-3f51-11e9-960a-951712a1a60b.png




Is this significantly different between populations? 

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=Hs.9pops$value, g=Hs.9pops$Var2, data=Hs.9pops, formula=Hs.9pops$value~Hs.9pops$Var2))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: Hs.9pops$value and Hs.9pops$Var2
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -6.719 7.0788e-05 ***
BRO - BAR == 0 -16.465 < 2.22e-16 ***
FOR - BAR == 0 -10.728 1.2434e-12 ***
HOD - BAR == 0  -1.380  0.9880742    
LYD - BAR == 0  -0.709  0.9998992    
MOF - BAR == 0  -4.202  0.0729229   .
SWD - BAR == 0  -8.940 9.3057e-09 ***
WIS - BAR == 0  -3.142  0.3914241    
BRO - BCH == 0 -23.000 < 2.22e-16 ***
FOR - BCH == 0 -18.207 < 2.22e-16 ***
HOD - BCH == 0  -2.764  0.5754429    
LYD - BCH == 0  -8.839 1.4710e-08 ***
MOF - BCH == 0 -13.966 8.4821e-14 ***
SWD - BCH == 0  -1.247  0.9939241    
WIS - BCH == 0 -13.022 1.0236e-13 ***
FOR - BRO == 0  -3.993  0.1085177    
HOD - BRO == 0 -20.060 < 2.22e-16 ***
LYD - BRO == 0 -11.742 1.0214e-13 ***
MOF - BRO == 0  -9.872 1.0573e-10 ***
SWD - BRO == 0 -25.396 < 2.22e-16 ***
WIS - BRO == 0 -15.728 3.3973e-14 ***
HOD - FOR == 0 -14.020 8.0491e-14 ***
LYD - FOR == 0  -7.534 3.5540e-06 ***
MOF - FOR == 0  -3.687  0.1832615    
SWD - FOR == 0 -20.599 < 2.22e-16 ***
WIS - FOR == 0  -9.667 2.9347e-10 ***
LYD - HOD == 0  -4.448  0.0438712   *
MOF - HOD == 0  -8.687 2.9208e-08 ***
SWD - HOD == 0  -4.999  0.0121987   *
WIS - HOD == 0  -6.819 4.9851e-05 ***
MOF - LYD == 0  -7.001 2.6079e-05 ***
SWD - LYD == 0 -11.181 2.2315e-13 ***
WIS - LYD == 0  -1.911  0.9156345    
SWD - MOF == 0 -16.750 < 2.22e-16 ***
WIS - MOF == 0  -5.426  0.0039668  **
WIS - SWD == 0 -15.949 1.3989e-14 ***
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

Hs.9pops$History <- Hs.9pops$Var2
Hs.9pops$History <- (Hs.9pops$History %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))
head(Hs.9pops)

Hs.9pops$HostPlatn <- Hs.9pops$Var2
Hs.9pops$HostPlatn <- (Hs.9pops$HostPlatn %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

Hs.9pops$History <- as.factor(Hs.9pops$History)
kruskal.test(value~History, data=Hs.9pops)


	Kruskal-Wallis rank sum test

data:  value by History
Kruskal-Wallis chi-squared = 143.24, df = 1, p-value < 2.2e-16



Hs.9pops$HostPlatn <- as.factor(Hs.9pops$HostPlatn)
kruskal.test(value~HostPlatn, data=Hs.9pops)

	Kruskal-Wallis rank sum test

data:  value by HostPlatn
Kruskal-Wallis chi-squared = 95.455, df = 1, p-value < 2.2e-16

```

This remained significant after removing one population at a time. I also checked whether a ranomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 2 1 2 1 1 2 1

Hs.9pops$rand <- (Hs.9pops$rand %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "2", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "1", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
Hs.9pops$rand <- as.factor(Hs.9pops$rand)

kruskal.test(value~rand, data=Hs.9pops)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  value by rand
Kruskal-Wallis chi-squared = 4.9303, df = 1, p-value = 0.02639

```

This is significant, but much less so than the History and HostPlant groupings. 








##### 4. Allelic diversity

Total allelic diversity (rarefaction method) per population. 
```

```

And plot in R
```

```


Number of private alleles per population
```

```


And plot in R
```

```




##### 5. Linkage disequilibrium

As this is RAD data we can't use windows of x number of variants - this leaves us with too few scaffolds to work with. Instead I assume that all loci are independent and I calculate linkage between them. 








##### 6. Genome-wide Tajima's D

This needs to be calculated for each population separately. 

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261

```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TajimaD 1000 && mv out.TajimaD > $i.TajimaD; done

for i in $(ls *TajimaD); do rename 's/recode.vcf.maxmiss.recode.vcf.gz.//' $i; done

```
	
Read into R and plot
```


```


	



## Population Structure

### PCA

PCA is affecteted by missing data, so I'll remove all loci with >80% missingness. I'm also filtering for MAF 0.01.


```
bluecp3:/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/PCAdapt

vcftools --vcf AAgestis.264.66491_FINAL_newnames.recode.vcf --max-missing 0.8 --maf 0.01 --recode --recode-INFO-all --out AAgestis.forPCA

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.264.66491_FINAL_newnames.recode.vcf
	--recode-INFO-all
	--maf 0.01
	--max-missing 0.8
	--out AAgestis.forPCA
	--recode

After filtering, kept 264 out of 264 Individuals
Outputting VCF file...
After filtering, kept 28980 out of a possible 66491 Sites
Run Time = 24.00 seconds

```

Convert to plink

```
vcftools --vcf AAgestis.forPCA.recode.vcf --plink --out AAgestis.forPCA.plink

```


Copy to Mac
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/input.data

plink --file AAgestis.forPCA.plink --recode A --out AAgestis.forPCA.plink

```

Open R and run analysis
```
R version 3.5.0

library(pcadapt)

AA264 <- read.pcadapt("AAgestis.forPCA.plink.ped", type="ped")

Summary:

	- input file:				AAgestis.forPCA.plink.ped
	- output file:				/var/folders/d1/2957hj5928x_7dc6xjx_2rx40000gq/T//RtmpjH9vMm/filef3a44aebc49d.pcadapt

	- number of individuals detected:	264
	- number of loci detected:		28980

28980 lines detected.
264 columns detected.


x.AA264 <- pcadapt(AA264, K=20)

Reading file CHall.932.7744.plink.pcadapt...
Number of SNPs: 7744
Number of individuals: 932
Number of SNPs with minor allele frequency lower than 0.05 ignored: 0
2148762 out of 7217408 missing data ignored.

plot(x.AA264, option="screeplot")  ##PC for pop structure = on the steep curve
```

![alt_txt][nr_pcaclusters]

[nr_pcaclusters]:https://user-images.githubusercontent.com/12142475/53089433-7b858500-3504-11e9-950c-2b00d82f9334.png



Based on this I choose K=2. The random eigenvalues fall in the straight line, while the ones explaining structure are above this.

See vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html on choosing K.

Plot the PCA using population information


```
pop.AA264 <- read.table("AA264.cluster.pop", header=F)

summary(pop.AA264)
           V1            V2            V3        V4            V5       
 BAR_10_2013:  1   WIS    :45   geranium:118   new:116   Min.   :1.000  
 BAR_11_2014:  1   SWD    :39   rockrose:146   old:148   1st Qu.:3.000  
 BAR_12_2013:  1   BCH    :38                            Median :5.000  
 BAR_13_2014:  1   HOD    :30                            Mean   :4.727  
 BAR_14_2013:  1   BAR    :28                            3rd Qu.:7.000  
 BAR_14_2014:  1   MOF    :25                            Max.   :9.000  
 (Other)    :258   (Other):59                                           
       V6              V7                V8                  V9     
 Min.   :51.00   Min.   :-1.2189   Min.   :22.00   darkorchid3: 88  
 1st Qu.:51.62   1st Qu.:-1.0394   1st Qu.:22.00   darkorchid4: 30  
 Median :52.97   Median :-0.6404   Median :22.00   gold1      : 28  
 Mean   :52.50   Mean   :-0.4365   Mean   :22.88   goldenrod  :118  
 3rd Qu.:53.19   3rd Qu.:-0.1689   3rd Qu.:24.00                    
 Max.   :54.16   Max.   : 1.2694   Max.   :24.00                    
                                                                    
summary(pop.AA264$V2)  ##pop sizes
BAR BCH BRO FOR HOD LYD MOF SWD WIS 
 28  38  18  20  30  21  25  39  45 



poplist <- as.character(pop.AA264[,2])

plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,3]) ##pca coloured by host plant presence. 
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA264[,4]) ##pca coloured by new vs old
plot(x.AA264, option="scores", pop=poplist) ##pca with all populations specified
```


![alt_txt][PCApops]

[PCApops]:https://user-images.githubusercontent.com/12142475/53090273-c902f180-3506-11e9-9e59-9d52ab970e6a.png


![alt_txt][PCAcolHist]

[PCAcolHist]:https://user-images.githubusercontent.com/12142475/53090546-b0470b80-3507-11e9-8476-483f42f7fe83.png



![alt_txt][PCAhostplant]

[PCAcolhostplant]:https://user-images.githubusercontent.com/12142475/53090576-c785f900-3507-11e9-8056-2883892005bc.png




### fineRADstructure

[fineRADstructure](http://cichlid.gurdon.cam.ac.uk/fineRADstructure.html)[(Malinksy et al. 2018)](https://academic.oup.com/mbe/article/35/5/1284/4883220)
uses haplotype information from RAD data to infer a co-ancestry matrix. The dataset utilises linkage information, so all loci should be kept. 

To create haplotype data, all variant loci from a single scaffold is concatenated into a sequence string. These data can be extracted from the vcf file. 

I've used the [vcf2hapmatrix.py](https://github.com/pimbongaerts/radseq/blob/master/vcf2hapmatrix.py) script to convert my vcf file to an input file for fineRADstructure. 
This script assumes that the scaffolds are in order. It concatenates all the nucleotide bases found on a specific scaffold together for each individual. This means many linked loci are preserved in the final dataset. 

The file conversion uses python3, and needs the module PyVCF. Conversions are run on the mac laptop. 
```
./vcf2hapmatrix.py 

```

