# Population structure and genetic diversity


Final datasets used here: 

##### 1. Final dataset for PopGen statistics (MAF 1%):


251 individuals

84117 variants

10097 loci



##### 2. Final dataset for Outlier analyses (MAF 5%):


251 individuals

35463 variants

6785 loci




#### Pop Gen Stats



##### 1. Number of polymorphic sites (and genomic resolution) per indiv per pop

The total number of sites typed per individual can be found in the out.het and snp density estimates from vcftools 
```
pwd
/newhome/aj18951/1a_Aricia_agestis_GWASdata/04_Outlier

vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --het

vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --SNPdensity 600

R
summary(all.snpden)
    CHROM             BIN_START          SNP_COUNT         VARIANTS.KB     
 Length:671682      Min.   :       0   Min.   : 0.00000   Min.   : 0.0000  
 Class :character   1st Qu.: 4656000   1st Qu.: 0.00000   1st Qu.: 0.0000  
 Mode  :character   Median : 9246000   Median : 0.00000   Median : 0.0000  
                    Mean   :10366696   Mean   : 0.08589   Mean   : 0.1431  
                    3rd Qu.:14206200   3rd Qu.: 0.00000   3rd Qu.: 0.0000  
                    Max.   :41908200   Max.   :44.00000   Max.   :73.3333  
		   

summary(all.snpden[which(all.snpden$VARIANTS.KB>0),])   
    CHROM             BIN_START          SNP_COUNT       VARIANTS.KB    
 Length:8370        Min.   :       0   Min.   : 1.000   Min.   : 1.667  
 Class :character   1st Qu.: 4435200   1st Qu.: 3.000   1st Qu.: 5.000  
 Mode  :character   Median : 9146700   Median : 5.000   Median : 8.333  
                    Mean   :10002836   Mean   : 6.892   Mean   :11.487  
                    3rd Qu.:14054100   3rd Qu.:10.000   3rd Qu.:16.667  
                    Max.   :41908200   Max.   :44.000   Max.   :73.333  
		    
##73 SNPs per kb = one every 14 bases!

#Where are these very variable regions? 
#looks like they're all over the genome

all.snpden[which(all.snpden$VARIANTS.KB>50),]
          CHROM BIN_START SNP_COUNT VARIANTS.KB
981     SUPER_9    634200        32     53.3333
31912  SUPER_10    759000        31     51.6667
69611  SUPER_11   5737200        33     55.0000
88233  SUPER_11  16910400        40     66.6667
95080  SUPER_12   3615000        35     58.3333
96764  SUPER_12   4625400        31     51.6667
116614 SUPER_12  16535400        36     60.0000
144077 SUPER_14    447600        36     60.0000
176463 SUPER_15   4125600        33     55.0000
211602 SUPER_16  10261800        34     56.6667
251565 SUPER_18   4681800        34     56.6667
301747  SUPER_Z  20203200        32     53.3333
392294 SUPER_21   4296000        44     73.3333
397214 SUPER_21   7248000        38     63.3333
450888  SUPER_2  19687800        34     56.6667
462559  SUPER_3   2023200        38     63.3333
468715  SUPER_3   5716800        39     65.0000
535752  SUPER_4  21775200        35     58.3333
580696  SUPER_6   4542600        37     61.6667
661129  SUPER_8  12534600        38     63.3333

##SUPER_5 doesn't feature at all!, and 

```
	


The interesting thing to note here are the hypervariable regions. The top two of these: 

```
module load languages/R-4.0.3-gcc9.1.0

snpden.9pops <- read.table("out.snpden", header=T)
snpden.9pops[which(snpden.9pops$VARIANTS.KB>40),]
          CHROM BIN_START SNP_COUNT VARIANTS.KB
235383 SUPER_21   4296000        62          62    ##SNP every 16bp
305504  SUPER_3   2023000        45          45    ##SNP every 22bp
```



##### 2. Genome-wide nucleotide diversity (measured in windows)

This can be estimated using vcftools - here in windows of 1kb. 
```
#Calculate pi in 1kb windows for each population

cd /newhome/aj18951/1a_Aricia_agestis_GWASdata/04_Outlier/VCF_perpop/test


#eg
vcftools --gzvcf 0008.vcf.gz --window-pi 1000
mv out.windowed.pi WIS.winpi    

vcftools --vcf ../../AA251.FINAL.MAF0.01.missing0.5perpop.vcf --window-pi 1000

mv out.windowed.pi AA251.windowed.pi
```

And move everything to the mac
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA251_FINAL

#Add pop name to each column. Headers removed from all but BAR
awk '{s=(NR==1)?"pop":"BAR";$0=$0 OFS s}1' BAR.winpi > BAR.winpi2
awk '{s="BCH";$0=$0 OFS s}1' BCH.winpi > BCH.winpi2
awk '{s="BRO";$0=$0 OFS s}1' BRO.winpi > BRO.winpi2
awk '{s="FOR";$0=$0 OFS s}1' FOR.winpi > FOR.winpi2
awk '{s="HOD";$0=$0 OFS s}1' HOD.winpi > HOD.winpi2
awk '{s="LYD";$0=$0 OFS s}1' LYD.winpi > LYD.winpi2
awk '{s="MOF";$0=$0 OFS s}1' MOF.winpi > MOF.winpi2
awk '{s="SWD";$0=$0 OFS s}1' SWD.winpi > SWD.winpi2
awk '{s="WIS";$0=$0 OFS s}1' WIS.winpi > WIS.winpi2



cat all together: 
cat BAR.winpi2 BCH.winpi2 BRO.winpi2 FOR.winpi2 HOD.winpi2 LYD.winpi2 MOF.winpi2 SWD.winpi2 WIS.winpi2 >> windowpi.9pops_A251

```


Calculate mean and SD
```
R

winpi <- read.table("windowpi.9pops_A251", header=T)


tapply(winpi$PI, winpi$pop, mean, na.rm=T)
        BAR          BCH          BRO          FOR          HOD          LYD 
0.0011460419 0.0012352851 0.0011048816 0.0009373671 0.0011466826 0.0010669760 
         MOF          SWD          WIS 
0.0012212952 0.0011609863 0.0011854702 

tapply(winpi$PI, winpi$pop, sd, na.rm=T)
         BAR          BCH          BRO          FOR          HOD          LYD 
0.0010113329 0.0010844298 0.0009646065 0.0008474430 0.0010108379 0.0009264965 
         MOF          SWD          WIS 
0.0010723023 0.0010215381 0.0010522020 

```

And plot in R

```
library(ggplot2)

pdf("AA251_PIperpop.pdf")
ggplot(winpi, aes(x=pop, y=PI)) + geom_boxplot() + ggtitle("Mean nucleotide diversity measured across 1kb windows for each population") + ylab("PI") + xlab("population")
dev.off()
```

![alt_txt][NucDiv]

[NucDiv]:https://user-images.githubusercontent.com/12142475/118273937-71bd1300-b4bc-11eb-89ce-78b2867e36a4.png
	
	


Check if this is significantly different between populations, and also between History and hostPlant

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=winpi$PI, g=winpi$pop, data=winpi, formula=winpi$PI~winpi$pop))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: winowpi.9pops_A261$PI and winowpi.9pops_A261$pop
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -6.770 5.9210e-05 ***
BRO - BAR == 0  -2.110 0.85979741    
FOR - BAR == 0 -17.596 < 2.22e-16 ***
HOD - BAR == 0  -0.352 0.99999957    
LYD - BAR == 0  -5.060 0.01044763   *
MOF - BAR == 0  -5.778 0.00145179  **
SWD - BAR == 0  -1.186 0.99570582    
WIS - BAR == 0  -2.577 0.66724630    
BRO - BCH == 0  -8.946 9.0427e-09 ***
FOR - BCH == 0 -24.308 < 2.22e-16 ***
HOD - BCH == 0  -6.481 0.00015912 ***
LYD - BCH == 0 -11.974 8.8707e-14 ***
MOF - BCH == 0  -1.022 0.99849644    
SWD - BCH == 0  -5.633 0.00221651  **
WIS - BCH == 0  -4.164 0.07851061   .
FOR - BRO == 0 -15.780 2.7978e-14 ***
HOD - BRO == 0  -2.482 0.71177974    
LYD - BRO == 0  -2.960 0.47797614    
MOF - BRO == 0  -7.874 9.2247e-07 ***
SWD - BRO == 0  -3.344 0.30368873    
WIS - BRO == 0  -4.696 0.02523162   *
HOD - FOR == 0 -18.065 < 2.22e-16 ***
LYD - FOR == 0 -12.908 1.1058e-13 ***
MOF - FOR == 0 -23.191 < 2.22e-16 ***
SWD - FOR == 0 -18.822 < 2.22e-16 ***
WIS - FOR == 0 -20.002 < 2.22e-16 ***
LYD - HOD == 0  -5.417 0.00406431  **
MOF - HOD == 0  -5.463 0.00357974  **
SWD - HOD == 0  -0.878 0.99950219    
WIS - HOD == 0  -2.265 0.80448731    
MOF - LYD == 0 -10.830 7.3419e-13 ***
SWD - LYD == 0  -6.322 0.00026850 ***
WIS - LYD == 0  -7.684 1.9743e-06 ***
SWD - MOF == 0  -4.561 0.03426165   *
WIS - MOF == 0  -3.132 0.39605924    
WIS - SWD == 0  -1.420 0.98562195    
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

winpi$History <- (winpi$pop %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))


winpi$HostPlant <- (winpi$pop %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

winpi$History <- as.factor(winpi$History)
kruskal.test(PI~History, data=winpi)


kruskal.test(PI~History, data=winowpi.9pops_A261)

	Kruskal-Wallis rank sum test

data:  PI by History
Kruskal-Wallis chi-squared = 42.182, df = 1, p-value = 8.317e-11



winpi$HostPlant <- as.factor(winpi$HostPlant)
kruskal.test(PI~HostPlant, data=winpi)

	Kruskal-Wallis rank sum test

data:  PI by HostPlant
Kruskal-Wallis chi-squared = 45.291, df = 1, p-value = 1.698e-11

```

This remained significant after removing one population at a time, except for MOF and FOR.
I also checked whether a ranomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 1 1 2 1 2 2 1

winpi$rand <- (winpi$pop %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "1", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "2", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
winpi$rand <- as.factor(winpi$rand)

kruskal.test(PI~rand, data=winpi)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  PI by rand
Kruskal-Wallis chi-squared = 0.23796, df = 1, p-value = 0.6257

```

Not significant. 
So there is a significant difference in nucleotide diversity, but not Het (see below) between HostPlant and History groups. But this seems to be driven by FOR and MOF. 



##### 3. Expected heterozygosity per populations
	
	
Diversity measures per population. This can be calculated in hierfstat. 


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA251_FINAL

bcftools query -l AA251.FINAL.MAF0.01.missing0.5perpop.vcf > AA251.names


R

library(adegenet)
library(vcfR)
library(hierfstat)

NamesAA <- read.table("AA251.names", header=F)
colnames(NamesAA) <- "Indiv"
NamesAA$pop <- gsub(_[0-9]., , NamesAA$Indiv)

AA251 <- read.vcfR("AA251.FINAL.MAF0.01.missing0.5perpop.vcf", nrows=61210)
Scanning file to determine attributes.
File attributes:
  meta lines: 71
  header_line: 72
  variant count: 61210
  column count: 260
Meta line 71 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 61210
  Character matrix gt cols: 260
  skip: 0
  nrows: 61210
  row_num: 0
Processed variant: 61210
All variants processed

AA.genind <- vcfR2genind(AA251)  ##convert to genind
AA251.pop <- read.table("../PCAdapt/AA251.names", header=T) ##Add pop info
> head(AA251.pop)
        Indiv Pop     Host History Colour Shape
1 BAR_10_2013 BAR Rockrose     New  gold1    17
2 BAR_11_2014 BAR Rockrose     New  gold1    17
3 BAR_12_2013 BAR Rockrose     New  gold1    17
4 BAR_13_2014 BAR Rockrose     New  gold1    17
5 BAR_14_2013 BAR Rockrose     New  gold1    17
6 BAR_14_2014 BAR Rockrose     New  gold1    17

AA.genind@pop <- as.factor(AA251.pop$Pop)

hier.AA251 <- genind2hierfstat(AA.genind) ## convert to hierfstat
AA251.stats <- basic.stats(hier.AA251) ## calculate basic statistics (gene diversity, etc). 

```

Plot expected heterozygosity frequency per population (boxplot and histogram)

```
Hs.9pops <- melt(AA251.stats$Hs)

pdf("Hs.9pops_maxmiss0.5.pdf")
ggplot(Hs.9pops, aes(x=Var2, y=value)) + geom_boxplot() + ggtitle("Aricia agestis: Genetic diversity (Hs) per pop") +ylab("Gene diversity (Hs)") + xlab("Population")
dev.off()
```

Draw a histogram for each species separately and combine graphs in Adobe Illustrator. 

![alt_txt][Hs.allpops]

[Hs.allpops]:https://user-images.githubusercontent.com/12142475/53811655-c914df80-3f51-11e9-960a-951712a1a60b.png




Is this significantly different between populations? 

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=Hs.9pops$value, g=Hs.9pops$Var2, data=Hs.9pops, formula=Hs.9pops$value~Hs.9pops$Var2))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: Hs.9pops$value and Hs.9pops$Var2
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -6.719 7.0788e-05 ***
BRO - BAR == 0 -16.465 < 2.22e-16 ***
FOR - BAR == 0 -10.728 1.2434e-12 ***
HOD - BAR == 0  -1.380  0.9880742    
LYD - BAR == 0  -0.709  0.9998992    
MOF - BAR == 0  -4.202  0.0729229   .
SWD - BAR == 0  -8.940 9.3057e-09 ***
WIS - BAR == 0  -3.142  0.3914241    
BRO - BCH == 0 -23.000 < 2.22e-16 ***
FOR - BCH == 0 -18.207 < 2.22e-16 ***
HOD - BCH == 0  -2.764  0.5754429    
LYD - BCH == 0  -8.839 1.4710e-08 ***
MOF - BCH == 0 -13.966 8.4821e-14 ***
SWD - BCH == 0  -1.247  0.9939241    
WIS - BCH == 0 -13.022 1.0236e-13 ***
FOR - BRO == 0  -3.993  0.1085177    
HOD - BRO == 0 -20.060 < 2.22e-16 ***
LYD - BRO == 0 -11.742 1.0214e-13 ***
MOF - BRO == 0  -9.872 1.0573e-10 ***
SWD - BRO == 0 -25.396 < 2.22e-16 ***
WIS - BRO == 0 -15.728 3.3973e-14 ***
HOD - FOR == 0 -14.020 8.0491e-14 ***
LYD - FOR == 0  -7.534 3.5540e-06 ***
MOF - FOR == 0  -3.687  0.1832615    
SWD - FOR == 0 -20.599 < 2.22e-16 ***
WIS - FOR == 0  -9.667 2.9347e-10 ***
LYD - HOD == 0  -4.448  0.0438712   *
MOF - HOD == 0  -8.687 2.9208e-08 ***
SWD - HOD == 0  -4.999  0.0121987   *
WIS - HOD == 0  -6.819 4.9851e-05 ***
MOF - LYD == 0  -7.001 2.6079e-05 ***
SWD - LYD == 0 -11.181 2.2315e-13 ***
WIS - LYD == 0  -1.911  0.9156345    
SWD - MOF == 0 -16.750 < 2.22e-16 ***
WIS - MOF == 0  -5.426  0.0039668  **
WIS - SWD == 0 -15.949 1.3989e-14 ***
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

Hs.9pops$History <- Hs.9pops$Var2
Hs.9pops$History <- (Hs.9pops$History %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))
head(Hs.9pops)

Hs.9pops$HostPlant <- Hs.9pops$Var2
Hs.9pops$HostPlant <- (Hs.9pops$HostPlant %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

Hs.9pops$History <- as.factor(Hs.9pops$History)
kruskal.test(value~History, data=Hs.9pops)


	Kruskal-Wallis rank sum test

data:  value by History
Kruskal-Wallis chi-squared = 298.52, df = 1, p-value < 2.2e-16



Hs.9pops$HostPlant <- as.factor(Hs.9pops$HostPlant)
kruskal.test(value~HostPlant, data=Hs.9pops)

	Kruskal-Wallis rank sum test

data:  value by HostPlant
Kruskal-Wallis chi-squared = 207.35, df = 1, p-value < 2.2e-16


Hs.8pops <- Hs.9pops %>% filter(!grepl("MOF", Var2)) ##exclude one pop at a time and check significance
kruskal.test(value~HostPlant, data=Hs.8pops)

	Kruskal-Wallis rank sum test

data:  value by HostPlant
Kruskal-Wallis chi-squared = 207.35, df = 1, p-value < 2.2e-16



```

This remained significant after removing one population at a time. I also checked whether a randomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 2 1 2 1 1 2 1

Hs.9pops$rand <- Hs.9pops$Var2
Hs.9pops$rand <- (Hs.9pops$rand %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "2", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "1", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
Hs.9pops$rand <- as.factor(Hs.9pops$rand)

kruskal.test(value~rand, data=Hs.9pops)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  value by rand
Kruskal-Wallis chi-squared = 0.078113, df = 1, p-value = 0.7799

```

This is non-significant, so genetic diversity is significantly correlated with host plant and history. 





##### 4. Linkage disequilibrium

######## BASED ON DRAFT GENOME, NOT SANGER #########

As this is RAD data we can't use windows of x number of variants - this leaves us with too few scaffolds to work with. Instead I assume that all loci are independent and I calculate linkage between them. 


```
kruskal.test(R.2~History, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by History
Kruskal-Wallis chi-squared = 1474.5, df = 1, p-value < 2.2e-16

> kruskal.test(R.2~HostPlant, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by HostPlant
Kruskal-Wallis chi-squared = 609.84, df = 1, p-value < 2.2e-16

> sample(1:2, 9, replace=T)
[1] 2 2 2 1 2 2 2 1 2
> ld.9pops$rand <- (ld.9pops$pop %>% gsub("BCH", "2", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "2", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "2", .) %>% gsub("WIS", "2", .) %>% gsub("BRO", "1", .) %>% gsub("FOR", "2", .))

> ld.9pops$rand <- as.factor(ld.9pops$rand)
> kruskal.test(R.2~rand, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by rand
Kruskal-Wallis chi-squared = 6466.8, df = 1, p-value < 2.2e-16

```


Everything is significantly different. 
There might be something interesting going on with the differences in LD. See the plot below



And plot
```
pdf("9pops.LD.pdf")
ggplot(ld.9pops, aes(R.2, x=pop)) + geom_boxplot()
dev.off()

```

![alt_txt][LD.9pops]

[LD.9pops]:https://user-images.githubusercontent.com/12142475/54035457-4f7b2c80-41b1-11e9-99a6-1b1916222a28.png



##### 5. Genome-wide Tajima's D

######## BASED ON DRAFT GENOME, NOT SANGER #########

This needs to be calculated for each population separately. 

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261

```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TajimaD 1000 && mv out.TajimaD > $i.TajimaD; done

for i in $(ls *TajimaD); do rename 's/recode.vcf.maxmiss.recode.vcf.gz.//' $i; done

```
	
Read into R and plot
```
TajimaD.9pops <- read.table("PerPopVCFfiles_AA261/9pops.TajimaD", header=T)
head(TajimaD.9pops)
TajimaD.9pops$pop <- "pop"
TajimaD.9pops$pop[1:8850] <- "BAR"
TajimaD.9pops$pop[8851:18063] <- "BCH"
TajimaD.9pops$pop[18064:26154] <- "BRO"
TajimaD.9pops$pop[26155:33720] <- "FOR"
TajimaD.9pops$pop[33721:42498] <- "HOD"
TajimaD.9pops$pop[42499:50841] <- "LYD"
TajimaD.9pops$pop[50842:59722] <- "MOF"
TajimaD.9pops$pop[59723:68609] <- "SWD"
TajimaD.9pops$pop[68610:77643] <- "WIS"
TajimaD.9pops$pop[68610:77642] <- "WIS"
head(TajimaD.9pops)

TajimaD.9pops$History <- (TajimaD.9pops$pop %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))
TajimaD.9pops$HostPlant <- (TajimaD.9pops$HostPlant %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


TajimaD.9pops$HostPlant <- (TajimaD.9pops$pop %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))

pdf("TajimaD.9pops.pdf")
ggplot(TajimaD.9pops, aes(pop, TajimaD)) + geom_boxplot()
dev.off()
```

![alt_txt][Tajima.9pops]

[Tajima.9pops]:https://user-images.githubusercontent.com/12142475/54032947-ef818780-41aa-11e9-86fd-018d1f9f7b9c.png



Are these significantly different between groups? and populations?

```
summary(dscfAllPairsTest(x=TajimaD.9pops$TajimaD, g=TajimaD.9pops$pop, data=TajimaD.9pops, formula=TajimaD.9pops$TajimaD~TajimaD.9pops$pop))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: TajimaD.9pops$TajimaD and TajimaD.9pops$pop
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -2.172 0.83881716    
BRO - BAR == 0  -1.299 0.99198927    
FOR - BAR == 0  -4.697 0.02521420   *
HOD - BAR == 0  -0.915 0.99932365    
LYD - BAR == 0  -5.135 0.00863204  **
MOF - BAR == 0  -2.455 0.72401045    
SWD - BAR == 0  -5.665 0.00202078  **
WIS - BAR == 0  -0.245 0.99999998    
BRO - BCH == 0  -3.444 0.26474808    
FOR - BCH == 0  -6.905 3.6782e-05 ***
HOD - BCH == 0  -1.313 0.99140709    
LYD - BCH == 0  -3.019 0.44927805    
MOF - BCH == 0  -0.402 0.99999877    
SWD - BCH == 0  -3.626 0.20205289    
WIS - BCH == 0  -2.382 0.75651465    
FOR - BRO == 0  -3.249 0.34371176    
HOD - BRO == 0  -2.267 0.80365939    
LYD - BRO == 0  -6.282 0.00030581 ***
MOF - BRO == 0  -3.540 0.23007207    
SWD - BRO == 0  -6.726 6.9144e-05 ***
WIS - BRO == 0  -1.045 0.99823664    
HOD - FOR == 0  -5.660 0.00205234  **
LYD - FOR == 0  -9.649 3.2173e-10 ***
MOF - FOR == 0  -6.831 4.7887e-05 ***
SWD - FOR == 0 -10.140 2.7060e-11 ***
WIS - FOR == 0  -4.370 0.05175128   .
LYD - HOD == 0  -4.253 0.06576526   .
MOF - HOD == 0  -1.558 0.97414199    
SWD - HOD == 0  -4.830 0.01845235   *
WIS - HOD == 0  -1.142 0.99669757    
MOF - LYD == 0  -2.564 0.67372302    
SWD - LYD == 0  -0.516 0.99999132    
WIS - LYD == 0  -5.298 0.00561958  **
SWD - MOF == 0  -2.982 0.46750966    
WIS - MOF == 0  -2.667 0.62346674    
WIS - SWD == 0  -5.812 0.00131175  **
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

And for History and Hostplant

```
> kruskal.test(TajimaD~History, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by History
Kruskal-Wallis chi-squared = 6.3129, df = 1, p-value = 0.01199

> kruskal.test(TajimaD~HostPlant, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by HostPlant
Kruskal-Wallis chi-squared = 3.6057, df = 1, p-value = 0.05758


##check if this holds true for a random rearrangement of the History and HostPlant catagories

sample(1:2, 9, replace=T)
[1] 1,2,1,1,1,2,1,1,2

TajimaD.9pops$rand <- (TajimaD.9pops$pop %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "1", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "1", .) %>% gsub("MOF", "2", .) %>% gsub("WIS", "1", .) %>% gsub("BRO", "1", .) %>% gsub("FOR", "2", .))


kruskal.test(TajimaD~rand, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by rand
Kruskal-Wallis chi-squared = 0.019755, df = 1, p-value = 0.8882
```

Again, FOR and MOF seem to be driving this pattern.
When either of these are removed the test is highly significant. But NS if any of the other populations are removed. 




##### 6. Estimate Ts/Tv ration in every population

######## BASED ON DRAFT GENOME, NOT SANGER #########

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261
```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TsTv 1000 && mv out.TsTv $i.TsTv; done

BAR.vcf.gz
Ts/Tv ratio: 1.232

BCH.vcf.gz
Ts/Tv ratio: 1.232

BRO.vcf.gz
Ts/Tv ratio: 1.239

FOR.vcf.gz
Ts/Tv ratio: 1.241

HOD.vcf.gz
Ts/Tv ratio: 1.231

LYD.vcf.gz
Ts/Tv ratio: 1.23

MOF.vcf.gz
Ts/Tv ratio: 1.232

SWD.vcf.gz
Ts/Tv ratio: 1.229

WIS.vcf.gz
Ts/Tv ratio: 1.234
```
	
Pretty much the same between all populations. 


## Population Structure

### Fst and IBD

First calculate mean Fst overall and then for all the group comparisons: 

Vignette [here](http://adegenet.r-forge.r-project.org/files/tutorial.pdf)

First subset to 1000 random loci
```
##Thin to include only one variant per locus

/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF

module load apps/vcftools-0.1.17b

vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --thin 600 --recode --recode-INFO-all --out AA251.thin.MAF0.01

#write all the sites in the latest vcf
vcftools --vcf AA251.thin.MAF0.01.recode.vcf --kept-sites

#use gnu shuf to randomly choose 1000 SNPs and 3000 SNPs after removing the header
shuf -n1000 out.kept.sites > AA251.1000sites
shuf -n3000 out.kept.sites > AA251.3000sites

#Create new vcf files with only these sites

vcftools --vcf AA251.thin.MAF0.01.recode.vcf --positions AA251.1000sites --recode --recode-INFO-all --out AA251.1000sites
vcftools --vcf AA251.thin.MAF0.01.recode.vcf --positions AA251.3000sites --recode --recode-INFO-all --out AA251.3000sites



##Convert to Plink

#First create a chrom map 

#first create a chromosome map
module load apps/bcftools-1.8
bcftools view -H AA251.1000sites.recode.vcf |cut -f 1 |uniq | awk '{print $o"\t"$0}' > AA251.1000sites.chrom-map.txt
bcftools view -H AA251.3000sites.recode.vcf |cut -f 1 |uniq | awk '{print $o"\t"$0}' > AA251.3000sites.chrom-map.txt

##then convert

vcftools --vcf  AA251.1000sites.recode.vcf --plink --chrom-map AA251.1000sites.chrom-map.txt --out AA251.1000.plink
vcftools --vcf  AA251.3000sites.recode.vcf --plink --chrom-map AA251.3000sites.chrom-map.txt --out AA251.3000.plink


module load apps/plink2

plink --file AA251.1000.plink --recode A --out AA251.1000.plink --allow-extra-chr
plink --file AA251.3000.plink --recode A --out AA251.3000.plink --allow-extra-chr


##Copy to mac and read into R
pwd
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA251_FINAL

scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF/AA251.1000.plink* .
scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF/AA251.3000.plink* .

```


```
getwd()
"/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA251_FINAL"

library(adegenet)
library(vcfR)
library(hierfstat)

###Read in vcf files
AA251.1000 <- read.vcfR("AA251.1000sites.recode.vcf", nrows=1000)
AA251.3000 <- read.vcfR("AA251.3000sites.recode.vcf", nrows=3000)

#convert to genind
AA.genind.1000 <- vcfR2genind(AA251.1000)
AA.genind.3000 <- vcfR2genind(AA251.3000)


##Add pop info
AA251.pop <- read.table("../PCAdapt/AA251.names", header=T) ##Add pop info
> head(AA251.pop)
        Indiv Pop     Host History Colour Shape
1 BAR_10_2013 BAR Rockrose     New  gold1    17
2 BAR_11_2014 BAR Rockrose     New  gold1    17
3 BAR_12_2013 BAR Rockrose     New  gold1    17
4 BAR_13_2014 BAR Rockrose     New  gold1    17
5 BAR_14_2013 BAR Rockrose     New  gold1    17
6 BAR_14_2014 BAR Rockrose     New  gold1    17

AA.genind.1000@pop <- as.factor(AA251.pop$Pop)
AA.genind.3000@pop <- as.factor(AA251.pop$Pop)


####
fstat(AA.genind.1000)

fstat(AA264.1000.genind)  ##overall Fst from previous ddRAD mapped to draft reference
             pop       Ind
Total 0.03372121 0.3598001
pop   0.00000000 0.3374584

fstat(AA.genind.1000)      ##overall Fst based on 1000 radom independent SNPs mapped to Sanger genome
             pop       Ind
Total 0.02791767 0.3003377
pop   0.00000000 0.2802438

fstat(AA.genind.3000)      ##overall Fst based on 3000 radom independent SNPs mapped to Sanger genome
             pop       Ind
Total 0.03100062 0.3201592
pop   0.00000000 0.2984095


#convert to hierfstat

AA1000.hier <- genind2hierfstat(AA.genind.1000)
AA3000.hier <- genind2hierfstat(AA.genind.3000)


boot.fst1000.ALLpops <- boot.ppfst(data.frame(as.numeric(pop),AA1000.hier[,-c(1)]))
 boot.fst1000.ALLpops$ul
      [,1]   [,2]   [,3]   [,4]   [,5]   [,6]   [,7]   [,8]   [,9]
 [1,]   NA 0.0506 0.0258 0.0284 0.0366 0.0442 0.0110 0.0432 0.0187
 [2,]   NA     NA 0.0757 0.0538 0.0339 0.0295 0.0578 0.0137 0.0707
 [3,]   NA     NA     NA 0.0346 0.0546 0.0694 0.0262 0.0636 0.0161
 [4,]   NA     NA     NA     NA 0.0362 0.0574 0.0203 0.0401 0.0260
 [5,]   NA     NA     NA     NA     NA 0.0440 0.0376 0.0294 0.0423
 [6,]   NA     NA     NA     NA     NA     NA 0.0642 0.0255 0.0657
 [7,]   NA     NA     NA     NA     NA     NA     NA 0.0480 0.0157
 [8,]   NA     NA     NA     NA     NA     NA     NA     NA 0.0588
 [9,]   NA     NA     NA     NA     NA     NA     NA     NA     NA
 
 boot.fst1000.ALLpops$ll
      [,1]   [,2]   [,3]   [,4]   [,5]   [,6]   [,7]   [,8]   [,9]
 [1,]   NA 0.0363 0.0108 0.0123 0.0237 0.0262 0.0028 0.0289 0.0094
 [2,]   NA     NA 0.0544 0.0375 0.0223 0.0177 0.0386 0.0070 0.0503
 [3,]   NA     NA     NA 0.0168 0.0373 0.0468 0.0093 0.0456 0.0043
 [4,]   NA     NA     NA     NA 0.0199 0.0333 0.0069 0.0243 0.0136
 [5,]   NA     NA     NA     NA     NA 0.0254 0.0251 0.0191 0.0302
 [6,]   NA     NA     NA     NA     NA     NA 0.0387 0.0119 0.0471
 [7,]   NA     NA     NA     NA     NA     NA     NA 0.0350 0.0068
 [8,]   NA     NA     NA     NA     NA     NA     NA     NA 0.0453
 [9,]   NA     NA     NA     NA     NA     NA     NA     NA     NA



boot.fst3000.ALLpops <- boot.ppfst(data.frame(as.numeric(AA251.pop$Pop),AA3000.hier[,-c(1)]))
 boot.fst3000.ALLpops$ul


boot.fst3000.ALLpops$ll

#Now change the pops to Old and New
pop.ColHist <- gsub("BAR", "New", AA251.pop$Pop)
pop.ColHist <- gsub("BRO", "New", pop.ColHist)
pop.ColHist <- gsub("FOR", "New", pop.ColHist)
pop.ColHist <- gsub("MOF", "New", pop.ColHist)
pop.ColHist <- gsub("WIS", "New", pop.ColHist)
pop.ColHist <- gsub("BCH", "Est", pop.ColHist)
pop.ColHist <- gsub("HOD", "Est", pop.ColHist)
pop.ColHist <- gsub("SWD", "Est", pop.ColHist)
pop.ColHist <- gsub("LYD", "Est", pop.ColHist)

pop.ColHist <- as.factor(pop.ColHist)
AA251.3000.genind.ColHist <- AA.genind.3000
AA251.3000.genind.ColHist@pop <- pop.ColHist

fstat(AA251.3000.genind.ColHist)
             pop       Ind
Total 0.02606285 0.3265014
pop   0.00000000 0.3084783


##Change pops to Ger and RR
pop.HostPlant <- gsub("BAR", "RR", AA251.pop$Pop)
pop.HostPlant <- gsub("BCH", "RR", pop.HostPlant)
pop.HostPlant <- gsub("FOR", "RR", pop.HostPlant)
pop.HostPlant <- gsub("LYD", "RR", pop.HostPlant)
pop.HostPlant <- gsub("SWD", "RR", pop.HostPlant)
pop.HostPlant <- gsub("BRO", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("HOD", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("MOF", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("WIS", "Ger", pop.HostPlant)

pop.HostPlant <- as.factor(pop.HostPlant)
AA251.3000.genind.HostPlant <- AA.genind.3000
AA251.3000.genind.HostPlant@pop <- pop.HostPlant

fstat(AA251.3000.genind.HostPlant)
             pop       Ind
Total 0.01755635 0.3237100
pop   0.00000000 0.3116246



####Couldn't get bootstrap to work for subset populations. And I couldn't subset the datasets (i.e. to test all new pops or all Ger pops)

#Overall Weir & Cockerham Fst

wc(AA.genind.3000)
$FST
[1] 0.03100062

$FIS
[1] 0.2984095



```


I calculated population pairwise Fst using adegenet. This was based on a subset of 3000 loci, one SNP per locus. 
```
library(adegenet)

AA251.3000.fst <- pairwise.fst(AA.3000.genind)

AA251.3000.fst
           1          2          3          4          5          6          7         8
2 0.02695511                                                                  
3 0.02900239 0.03465201                                                       
4 0.03011466 0.03272610 0.03257589                                            
5 0.02570288 0.02550700 0.03361358 0.03407234                                 
6 0.03322799 0.02161129 0.04704231 0.04213453 0.03081127                      
7 0.02147975 0.03397874 0.02566609 0.02713527 0.02730942 0.04015687           
8 0.02490943 0.01553652 0.03260854 0.03049523 0.02410006 0.02152653 0.03240735
9 0.01829222 0.03433768 0.01859020 0.02225308 0.02704452 0.03546717 0.01307305 0.03204089


```


I tested if genetic distance increased significantly with 1) geographic distance, 2) Difference in colonisation history (0/1), and 3) Difference in dominant host plant (0/1). 

```
library(fields)
library(vegan)

AA.genpop.3000 <- genind2genpop(AA.genind.3000)

BA.site.info <- read.table("../BA_SiteInfo", header=T)
BA.site.info$Pop <- BA.site.info$Site_full_name %>% gsub("Barnack", "BAR", .) %>% gsub("Beacon_Hill", "BCH", .) %>% gsub("Brockadale", "BRO", .) %>% gsub("Fordon", "FOR", .) %>% gsub("Holme_Dunes", "HOD", .) %>% gsub("Lydden", "LYD", .) %>% gsub("Moor_Farm", "MOF", .) %>% gsub("Swyncombe", "SWD", .) %>% gsub("Whisby", "WIS", .)


Dgen <- dist.genpop(AA.genpop.3000, method=1)  ##Nei's genetic distance (1972)
AA_lon.lat <- cbind(BA.site.info$Long, BA.site.info$Lat)
Dgeo <- rdist.earth(AA_lon.lat, miles=F)
Dgeo <- log(Dgeo) ##log of distance as suggested for tests of IBD by Rousset. 
DHostPlant <- dist(as.numeric(BA.site.info$Site_dominant_hostplant))

DColHist <- dist(as.numeric(BA.site.info$Site_colonization_history))

##Significance of genetic distance vs geographic distance
mantel(Dgen, Dgeo)

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = Dgeo) 

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = Dgeo) 

Mantel statistic r: 0.7861 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.243 0.300 0.373 0.455 
Permutation: free
Number of permutations: 999


##Significance of genetic distance vs HostPlant

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = DHostPlant) 

Mantel statistic r: 0.2917 
      Significance: 0.074 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.217 0.313 0.467 0.588 
Permutation: free
Number of permutations: 999


##Significance of genetic distance vs ColHistory
Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = DColHist) 

Mantel statistic r: 0.4538 
      Significance: 0.04 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.220 0.313 0.588 0.694 
Permutation: free
Number of permutations: 999
```






### PCA

PCA is affecteted by missing data. As this file contains only loci that have been genotyped across all populations, I will use AA261 as is. 

```
bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/PCA

vcftools --vcf ../FINAL_VCF/AA251.FINAL.MAF0.01.missing0.5perpop.vcf

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf ../FINAL_VCF/AA251.FINAL.MAF0.01.missing0.5perpop.vcf

After filtering, kept 251 out of 251 Individuals
After filtering, kept 61210 out of a possible 61210 Sites
Run Time = 0.00 seconds

#This still includes the MT genome, but this has just one SNP
vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --not-chr scaffold_MT --recode --recode-INFO-all --out AAgestis.251.MAF0.05.FINAL.noMT

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.251.MAF0.05.FINAL.recode.vcf
	--not-chr scaffold_MT
	--recode-INFO-all
	--out AAgestis.251.MAF0.05.FINAL.noMT
	--recode

After filtering, kept 251 out of 251 Individuals
Outputting VCF file...
After filtering, kept 61209 out of a possible 61210 Sites
Run Time = 21.00 seconds

```

Convert to plink

```
#first create a chromosome map
bcftools view -H ../FINAL_VCF/AAgestis.251.MAF0.05.FINAL.noMT.recode.vcf |cut -f 1 |uniq | awk '{print $o"\t"$0}' > ../FINAL_VCF/AA251.chrom-map.txt

##then convert

vcftools --vcf ../FINAL_VCF/AAgestis.251.MAF0.05.FINAL.noMT.recode.vcf --plink --chrom-map ../FINAL_VCF/AA251.chrom-map.txt --out AA251.plink


Parameters as interpreted:
	--vcf ../FINAL_VCF/AAgestis.251.MAF0.05.FINAL.noMT.recode.vcf
	--chrom-map ../FINAL_VCF/AA251.chrom-map.txt
	--out AA251.plink
	--plink

After filtering, kept 251 out of 251 Individuals
Writing PLINK PED and MAP files ... 
	Read 24 chromosome mapping file entries.
Done.
After filtering, kept 61209 out of a possible 61209 Sites
Run Time = 6.00 seconds


module load apps/plink2

plink --file AA251.plink --recode A --out AA251.plink --allow-extra-chr
PLINK v1.90b3f 64-bit (2 Mar 2015)         https://www.cog-genomics.org/plink2
(C) 2005-2015 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to AA251.plink.log.
129152 MB RAM detected; reserving 64576 MB for main workspace.
.ped scan complete (for binary autoconversion).
Performing single-pass .bed write (61209 variants, 251 people).
--file: AA251.plink-temporary.bed + AA251.plink-temporary.bim +
AA251.plink-temporary.fam written.
61209 variants loaded from .bim file.
251 people (0 males, 0 females, 251 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA251.plink.nosex .
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 251 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921893.
61209 variants and 251 people pass filters and QC.
Note: No phenotypes present.
--recode A to AA251.plink.raw ... done.
```


92% genotyping rate


Copy to Mac
```
pwd
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/PCAdapt

scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/PCA/AA251.plink* .

```



Open R and run analysis
```
R version 3.5.0

library(pcadapt)

AA251 <- read.pcadapt("AA251.plink.ped", type="ped")
Summary:

	- input file:				AA251.plink.ped
	- output file:				/var/folders/d1/2957hj5928x_7dc6xjx_2rx40000gq/T//RtmpFhZYZa/filead0854684416.pcadapt

	- number of individuals detected:	251
	- number of loci detected:		61209

61209 lines detected.
251 columns detected.



x.AA251 <- pcadapt(AA251, K=20)



plot(x.AA251, option="screeplot")  ##PC for pop structure = on the steep curve
```

![alt_txt][nr_pcaclusters]

[nr_pcaclusters]:https://user-images.githubusercontent.com/12142475/54189040-e1807f00-44a8-11e9-9d0f-93f9d59b3652.png



Based on this I choose K=2. The random eigenvalues fall in the straight line, while the ones explaining structure are above this.

See vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html on choosing K.

Plot the PCA using population information


```
pop.AA251 <- read.table("AA251.names", header=T) ##read in names acquired from the vcf file. 

summary(pop.AA251)
         Indiv          Pop           Host            History   
 BAR_10_2013:  1   BCH    :38   Geranium:107   Established:125  
 BAR_11_2014:  1   SWD    :38   Rockrose:144   New        :126  
 BAR_12_2013:  1   WIS    :38                                   
 BAR_13_2014:  1   HOD    :29                                   
 BAR_14_2013:  1   BAR    :28                                   
 BAR_14_2014:  1   MOF    :25                                   
 (Other)    :245   (Other):55                                   
         Colour  
 darkorchid3:78  
 darkorchid4:29  
 gold1      :48  
 goldenrod  :96  
               
                                                                    
summary(pop.AA251$Pop)
BAR BCH BRO FOR HOD LYD MOF SWD WIS 
 28  38  15  20  29  20  25  38  38 



poplist <- as.character(pop.AA251$Pop)

plot(x.AA251, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA251$Host) ##pca coloured by host plant presence. 
plot(x.AA251, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA251$History) ##pca coloured by new vs old
plot(x.AA251, option="scores", pop=poplist) ##pca with all populations specified
```


![alt_txt][PCApops]

[PCApops]:https://user-images.githubusercontent.com/12142475/118281545-3c68f300-b4c5-11eb-851c-b3639a76e2f8.png


![alt_txt][PCAcolHist]

[PCAcolHist]:https://user-images.githubusercontent.com/12142475/118281600-51de1d00-b4c5-11eb-9942-1151781d5290.png



![alt_txt][PCAhostplant]

[PCAhostplant]:https://user-images.githubusercontent.com/12142475/118281642-5efb0c00-b4c5-11eb-86a4-c0f270512ba6.png



Plot with the correct shape and colours. 
```
library(ggplot2)

pop.AA251$PC1 <- x.AA251$scores[,1]
pop.AA251$PC2 <- x.AA251$scores[,2]

pdf("PCA251.forMS.pdf")
ggplot(pop.AA251, aes(x=PC1, y=PC2, colour=Colour, shape=Shape)) + geom_point() + scale_shape_manual(values = c(15,17)) + scale_color_manual(values=c("darkorchid3", "darkorchid4", "gold1", "goldenrod1"))
dev.off()


##Proportion of variance explained by PC1 and PC2
#singular.values is a vector containing the K ordered square root of the proportion of variance explained by each PC.

head(x.AA251$singular.values)
[1] 0.19313949 0.11547493 0.10087238 0.09903393 0.09793860 0.09448205

##PC1: 
100*(0.19313949^2)
[1] 3.730286


##PC2
100*(0.11547493^2)
[1] 1.333446

```


![alt_txt][PCAfinal]

[PCAfinal]:https://user-images.githubusercontent.com/12142475/54742011-09ae6300-4bb8-11e9-9a96-172bdc3595cd.png


### fineRADstructure

[fineRADstructure](http://cichlid.gurdon.cam.ac.uk/fineRADstructure.html)[(Malinksy et al. 2018)](https://academic.oup.com/mbe/article/35/5/1284/4883220)
uses haplotype information from RAD data to infer a co-ancestry matrix. The dataset utilises linkage information, so all loci should be kept. 

To create haplotype data, all variant loci from a single scaffold is concatenated into a sequence string. These data can be extracted from the vcf file. 

I've used the [vcf2hapmatrix.py](https://github.com/pimbongaerts/radseq/blob/master/vcf2hapmatrix.py) script to convert my vcf file to an input file for fineRADstructure. 
This script assumes that the scaffolds are in order. It concatenates all the nucleotide bases found on a specific scaffold together for each individual. This means many linked loci are preserved in the final dataset. 

The file conversion uses python3, and needs the module PyVCF.
```
pwd 
bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF

languages/python-anaconda3.8.5-2020.11
pip install PyVCF --user

./vcf2hapmatrix.py AA251.FINAL.MAF0.01.missing0.5perpop.vcf > AA251.fineradstr


##On mac:
cd /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure
scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF/AA251.fineradstr .

cp ~/Software/fineRADstructure/RADpainter .

./RADpainter paint AA251.fineradstr 
Painting RAD tags from: AA251.fineradstr
The file seems to be in a Matrix format
Processed: 100 tag loci in 7.10749 seconds(0.0710749 seconds per RAD locus)
```




### fastStructure

Get the population assignment Q values at K=2

Convert vcf to fastStructure format using pgdspider. Remember to add the extension .str to the input file. Run at K=2
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fastStructure

python /Users/alexjvr/Applications/fastStructure/fastStructure/structure.py -K 2 --format=str --input=AA251.FINAL.MAF0.01.missing0.5perpop --output=AA251.faststr_K2.2


```

Import the Q-values into R and plot

```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fastStructure

library(ggplot2)
library(reshape2)

df <- read.table("AA251.faststr_K2.2.2.meanQ", header=F)
popnames <- read.table("popnames.pgdspider")
df$pop <- popnames$V2
df$indiv <- popnames$V1

df$N2S <- df$pop
head(df$N2S)
df$N2S <- gsub("BCH", "1", df$N2S)
df$N2S <- gsub("LYD", "2", df$N2S)
df$N2S <- gsub("SWD", "3", df$N2S)
df$N2S <- gsub("BAR", "4", df$N2S)
df$N2S <- gsub("HOD", "5", df$N2S)
df$N2S <- gsub("MOF", "6", df$N2S)
df$N2S <- gsub("WIS", "7", df$N2S)
df$N2S <- gsub("BRO", "8", df$N2S)
df$N2S <- gsub("FOR", "9", df$N2S)

df$index <- 1:nrow(df)

attach(df)
df.sorted <- df[order(N2S),]
df.sorted$index2 <- 1:nrow(df.sorted)
df2 <- melt(df.sorted, id=c("pop", "indiv", "index2", "index", "N2S"),)


pdf("AA251.faststrK2.pdf")
ggplot(df2) + geom_bar(aes(x=index2, y=value, fill=variable), stat="identity")
dev.off()
```


### Figure

Combine the map, PCA, and fastructure together in a figure. 

![alt_txt][Fig1]

[Fig1]:https://user-images.githubusercontent.com/12142475/54742678-fdc3a080-4bb9-11e9-9483-c5d10468f1a7.png
