# Population structure and genetic diversity


Final datasets used here: 

##### 1. Final dataset for PopGen statistics (MAF 1%):

261 individuals

57689  variants

6845 loci (--thin 600)

##### 2. Final dataset for Outlier analyses (MAF 5%):

251 individuals

45693 variants

8660 loci




#### Pop Gen Stats



##### 1. Number of polymorphic sites (and genomic resolution) per indiv per pop

The total number of sites typed per individual can be found in the out.het and snp density estimates from vcftools 
```
pwd
/newhome/aj18951/1a_Aricia_agestis_GWASdata/04_Outlier

vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --het

vcftools --vcf AA251.FINAL.MAF0.01.missing0.5perpop.vcf --SNPdensity 600

R
summary(all.snpden)
    CHROM             BIN_START          SNP_COUNT         VARIANTS.KB     
 Length:671682      Min.   :       0   Min.   : 0.00000   Min.   : 0.0000  
 Class :character   1st Qu.: 4656000   1st Qu.: 0.00000   1st Qu.: 0.0000  
 Mode  :character   Median : 9246000   Median : 0.00000   Median : 0.0000  
                    Mean   :10366696   Mean   : 0.08589   Mean   : 0.1431  
                    3rd Qu.:14206200   3rd Qu.: 0.00000   3rd Qu.: 0.0000  
                    Max.   :41908200   Max.   :44.00000   Max.   :73.3333  
		   

summary(all.snpden[which(all.snpden$VARIANTS.KB>0),])   
    CHROM             BIN_START          SNP_COUNT       VARIANTS.KB    
 Length:8370        Min.   :       0   Min.   : 1.000   Min.   : 1.667  
 Class :character   1st Qu.: 4435200   1st Qu.: 3.000   1st Qu.: 5.000  
 Mode  :character   Median : 9146700   Median : 5.000   Median : 8.333  
                    Mean   :10002836   Mean   : 6.892   Mean   :11.487  
                    3rd Qu.:14054100   3rd Qu.:10.000   3rd Qu.:16.667  
                    Max.   :41908200   Max.   :44.000   Max.   :73.333  
		    
##73 SNPs per kb = one every 14 bases!

#Where are these very variable regions? 
#looks like they're all over the genome

all.snpden[which(all.snpden$VARIANTS.KB>50),]
          CHROM BIN_START SNP_COUNT VARIANTS.KB
981     SUPER_9    634200        32     53.3333
31912  SUPER_10    759000        31     51.6667
69611  SUPER_11   5737200        33     55.0000
88233  SUPER_11  16910400        40     66.6667
95080  SUPER_12   3615000        35     58.3333
96764  SUPER_12   4625400        31     51.6667
116614 SUPER_12  16535400        36     60.0000
144077 SUPER_14    447600        36     60.0000
176463 SUPER_15   4125600        33     55.0000
211602 SUPER_16  10261800        34     56.6667
251565 SUPER_18   4681800        34     56.6667
301747  SUPER_Z  20203200        32     53.3333
392294 SUPER_21   4296000        44     73.3333
397214 SUPER_21   7248000        38     63.3333
450888  SUPER_2  19687800        34     56.6667
462559  SUPER_3   2023200        38     63.3333
468715  SUPER_3   5716800        39     65.0000
535752  SUPER_4  21775200        35     58.3333
580696  SUPER_6   4542600        37     61.6667
661129  SUPER_8  12534600        38     63.3333

##SUPER_5 doesn't feature at all!, and 

```
	
	
The interesting thing to note here are the hypervariable regions found across all populations. The top two of these: 

```
snpden.9pops[which(snpden.9pops$VARIANTS.KB>60),]
          BCH_CHROM BIN_START SNP_COUNT VARIANTS.KB pop
4979   m_scaff_5956     19200        42          70 BCH
6892   contig_17335      6600        57          95 BCH
17537  m_scaff_5956     19200        42          70 LYD
19282  contig_17335      6600        57          95 LYD
29677  m_scaff_5956     19200        42          70 SWD
31543  contig_17335      6600        57          95 SWD
42192  m_scaff_5956     19200        42          70 BAR
44102  contig_17335      6600        57          95 BAR
54733  m_scaff_5956     19200        42          70 HOD
56637  contig_17335      6600        57          95 HOD
67362  m_scaff_5956     19200        42          70 MOF
69213  contig_17335      6600        57          95 MOF
80123  m_scaff_5956     19200        42          70 WIS
81914  contig_17335      6600        57          95 WIS
92571  m_scaff_5956     19200        42          70 BRO
104312 m_scaff_5956     19200        42          70 FOR
```

When I look at these contigs in the reference genome there seems to be a high frequency of runs of a single bp in these scaffolds. All indels have been removed from this dataset, and I've applied filters for mapping quality, depth of variant calls, and minor allele frequencies. 

m_scaff_5956 occurs in the gff file and codes for an mRNA in Heliconius: 

```
grep m_scaff_5956 Aricia_agestis_Red_MESPA.gff

m_scaff_5956	ALN	cds	19212	19222	15	+	1	ID=cds37451;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 287 289 +
m_scaff_5956	ALN	cds	19465	19467	17	+	2	ID=cds37452;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 290 290 +
m_scaff_5956	ALN	cds	19721	19732	7	+	2	ID=cds37453;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 291 295 +
m_scaff_5956	ALN	cds	19784	19786	23	+	2	ID=cds37454;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 296 296 +
m_scaff_5956	ALN	cds	19910	19936	34	+	2	ID=cds37455;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 297 304 +
m_scaff_5956	ALN	cds	20015	20035	28	+	2	ID=cds37456;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 305 311 +
m_scaff_5956	ALN	cds	20100	20104	25	+	2	ID=cds37457;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 312 313 +
m_scaff_5956	ALN	cds	20143	20163	11	+	0	ID=cds37458;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 314 317 +
m_scaff_5956	ALN	cds	20381	20405	23	+	0	ID=cds37459;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 318 325 +
m_scaff_5956	ALN	cds	20507	20517	17	+	2	ID=cds37460;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 326 329 +
m_scaff_5956	ALN	cds	20706	20737	26	+	0	ID=cds37461;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 330 340 +
m_scaff_5956	ALN	cds	20826	20868	24	+	1	ID=cds37462;Parent=mRNA02345;Name=m_scaff_5956_11;Target=HMEL025049_RA_Hmel203003o_gene_HMEL025049_Hmel203003o 341 352 +
[aj18951

```


##### 2. Genome-wide nucleotide diversity (measured in windows)

This can be estimated using vcftools - here in windows of 1kb. 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

vcftools --vcf AA261.0.5miss.9popsMerged.vcf --window-pi 1000

mv out.windowed.pi AA261.windowed.pi
```

Calculate mean and SD
```
tapply(winowpi.9pops_A261$PI, winowpi.9pops_A261$pop, mean, na.rm=T)
         BAR          BCH          BRO          FOR          HOD          LYD 
0.0010411435 0.0011215100 0.0010182363 0.0008825968 0.0010366159 0.0009963501 
         MOF          SWD          WIS 
0.0011155415 0.0010591159 0.0010719579 


tapply(winowpi.9pops_A261$PI, winowpi.9pops_A261$pop, sd, na.rm=T)
         BAR          BCH          BRO          FOR          HOD          LYD 
0.0009572797 0.0010310719 0.0009232863 0.0008207372 0.0009646674 0.0008943332 
         MOF          SWD          WIS 
0.0010311528 0.0009765223 0.0009983996 
```

And plot in R

```
winowpi.9pops_A261$pop <- "pop"
winowpi.9pops_A261$pop[1:5233] <- "BCH"
winowpi.9pops_A261$pop[5234:10069] <- "LYD"
winowpi.9pops_A261$pop[10070:15159] <- "SWD"
winowpi.9pops_A261$pop[15160:20168] <- "HOD"
winowpi.9pops_A261$pop[20169:25165] <- "MOF"
winowpi.9pops_A261$pop[25166:30278] <- "WIS"
winowpi.9pops_A261$pop[30279:34951] <- "BRO"
winowpi.9pops_A261$pop[34952:39329] <- "FOR"
winowpi.9pops_A261$pop[39330:nrow(winowpi.9pops_A261)] <- "BAR"


pdf("AA261_PIperpop.pdf")
ggplot(winowpi.9pops_A261, aes(x=pop, y=PI)) + geom_boxplot() + ggtitle("Mean nucleotide diversity measured across 1kb windows for each population") + ylab("PI") + xlab("population")
dev.off()
```

![alt_txt][NucDiv]

[NucDiv]:https://user-images.githubusercontent.com/12142475/53818593-37609e80-3f60-11e9-808e-40a5bcb68b3a.png
	
	


Check if this is significantly different between populations, and also between History and hostPlant

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=winowpi.9pops_A261$PI, g=winowpi.9pops_A261$pop, data=winowpi.9pops_A261, formula=winowpi.9pops_A261$PI~winowpi.9pops_A261$pop))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: winowpi.9pops_A261$PI and winowpi.9pops_A261$pop
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -4.487 0.04033001   *
BRO - BAR == 0  -0.495 0.99999377    
FOR - BAR == 0 -10.216 1.8217e-11 ***
HOD - BAR == 0  -0.676 0.99993058    
LYD - BAR == 0  -1.615 0.96782256    
MOF - BAR == 0  -3.961 0.11502517    
SWD - BAR == 0  -0.822 0.99969430    
WIS - BAR == 0  -0.993 0.99877014    
BRO - BCH == 0  -4.960 0.01344163   *
FOR - BCH == 0 -14.542 9.6811e-14 ***
HOD - BCH == 0  -5.143 0.00845932  **
LYD - BCH == 0  -6.176 0.00042837 ***
MOF - BCH == 0  -0.496 0.99999362    
SWD - BCH == 0  -3.685 0.18390036    
WIS - BCH == 0  -3.459 0.25917632    
FOR - BRO == 0  -9.794 1.5667e-10 ***
HOD - BRO == 0  -0.165 1.00000000    
LYD - BRO == 0  -1.109 0.99730639    
MOF - BRO == 0  -4.399 0.04868001   *
SWD - BRO == 0  -1.317 0.99121524    
WIS - BRO == 0  -1.458 0.98295736    
HOD - FOR == 0  -9.520 6.0255e-10 ***
LYD - FOR == 0  -8.769 2.0167e-08 ***
MOF - FOR == 0 -13.827 9.7811e-14 ***
SWD - FOR == 0 -10.951 4.0146e-13 ***
WIS - FOR == 0 -10.981 3.4539e-13 ***
LYD - HOD == 0  -0.938 0.99919296    
MOF - HOD == 0  -4.586 0.03246386   *
SWD - HOD == 0  -1.471 0.98193927    
WIS - HOD == 0  -1.637 0.96503030    
MOF - LYD == 0  -5.554 0.00277393  **
SWD - LYD == 0  -2.461 0.72152683    
WIS - LYD == 0  -2.595 0.65878453    
SWD - MOF == 0  -3.129 0.39745764    
WIS - MOF == 0  -2.920 0.49793426    
WIS - SWD == 0  -0.180 1.00000000    
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

winowpi.9pops_A261$History <- (winowpi.9pops_A261$pop %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))


winowpi.9pops_A261$HostPlant <- (winowpi.9pops_A261$pop %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

winowpi.9pops_A261$History <- as.factor(winowpi.9pops_A261$History)
kruskal.test(PI~History, data=winowpi.9pops_A261)


kruskal.test(PI~History, data=winowpi.9pops_A261)

	Kruskal-Wallis rank sum test

data:  PI by History
Kruskal-Wallis chi-squared = 12.062, df = 1, p-value = 0.0005147



winowpi.9pops_A261$HostPlant <- as.factor(winowpi.9pops_A261$HostPlant)
kruskal.test(PI~HostPlant, data=winowpi.9pops_A261)

	Kruskal-Wallis rank sum test

data:  PI by HostPlant
Kruskal-Wallis chi-squared = 9.1441, df = 1, p-value = 0.002495

```

This remained significant after removing one population at a time, except for MOF and FOR.
I also checked whether a ranomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 1 1 2 1 2 2 1

winowpi.9pops_A261$rand <- (winowpi.9pops_A261$pop %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "1", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "2", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
winowpi.9pops_A261$rand <- as.factor(winowpi.9pops_A261$rand)

kruskal.test(PI~rand, data=winowpi.9pops_A261)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  PI by rand
Kruskal-Wallis chi-squared = 0.47368, df = 1, p-value = 0.4913

```

Not significant. 
So there is a significant difference in nucleotide diversity, but not Het (see below) between HostPlant and History groups. But this seems to be driven by FOR and MOF. 



##### 3. Expected heterozygosity per populations
	
	
Diversity measures per population. This can be calculated in hierfstat. 


```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

library(adegenet)
library(vcfR)
library(hierfstat)

AA261 <- read.vcfR("AA261.0.5miss.9popsMerged.vcf", nrows=31381)
Scanning file to determine attributes.
File attributes:
  meta lines: 93613
  header_line: 93614
  variant count: 31381
  column count: 270
Meta line 93613 read in.
All meta lines processed.
gt matrix initialized.
Character matrix gt created.
  Character matrix gt rows: 31381
  Character matrix gt cols: 270
  skip: 0
  nrows: 31381
  row_num: 0
Processed variant: 31381
All variants processed

AA.genind <- vcfR2genind(AA261)  ##convert to genind
AA261.pop <- read.table("AA.261.names", header=F) ## add population information
AA261.pop$pop <- gsub("_.*_.*", "", AA261.pop$V1)
AA.genind@pop <- as.factor(AA261.pop$pop)

hier.AA261 <- genind2hierfstat(AA.genind) ## convert to hierfstat
AA261.stats <- basic.stats(hier.AA261) ## calculate basic statistics (gene diversity, etc). 

```

Plot expected heterozygosity frequency per population (boxplot and histogram)

```
Hs.9pops <- melt(AA261.stats$Hs)

pdf("Hs.9pops_maxmiss0.5.pdf")
ggplot(Hs.9pops, aes(x=Var2, y=value)) + geom_boxplot() + ggtitle("Aricia agestis: Genetic diversity (Hs) per pop") +ylab("Gene diversity (Hs)") + xlab("Population")
dev.off()
```

Draw a histogram for each species separately and combine graphs in Adobe Illustrator. 

![alt_txt][Hs.allpops]

[Hs.allpops]:https://user-images.githubusercontent.com/12142475/53811655-c914df80-3f51-11e9-960a-951712a1a60b.png




Is this significantly different between populations? 

Dwass-Steel all-pairs test for significance. 
```
##To test for differences between multiple groups. This test is for non-parametric data. 
library(PMCMRplus)

summary(dscfAllPairsTest(x=Hs.9pops$value, g=Hs.9pops$Var2, data=Hs.9pops, formula=Hs.9pops$value~Hs.9pops$Var2))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: Hs.9pops$value and Hs.9pops$Var2
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -6.719 7.0788e-05 ***
BRO - BAR == 0 -16.465 < 2.22e-16 ***
FOR - BAR == 0 -10.728 1.2434e-12 ***
HOD - BAR == 0  -1.380  0.9880742    
LYD - BAR == 0  -0.709  0.9998992    
MOF - BAR == 0  -4.202  0.0729229   .
SWD - BAR == 0  -8.940 9.3057e-09 ***
WIS - BAR == 0  -3.142  0.3914241    
BRO - BCH == 0 -23.000 < 2.22e-16 ***
FOR - BCH == 0 -18.207 < 2.22e-16 ***
HOD - BCH == 0  -2.764  0.5754429    
LYD - BCH == 0  -8.839 1.4710e-08 ***
MOF - BCH == 0 -13.966 8.4821e-14 ***
SWD - BCH == 0  -1.247  0.9939241    
WIS - BCH == 0 -13.022 1.0236e-13 ***
FOR - BRO == 0  -3.993  0.1085177    
HOD - BRO == 0 -20.060 < 2.22e-16 ***
LYD - BRO == 0 -11.742 1.0214e-13 ***
MOF - BRO == 0  -9.872 1.0573e-10 ***
SWD - BRO == 0 -25.396 < 2.22e-16 ***
WIS - BRO == 0 -15.728 3.3973e-14 ***
HOD - FOR == 0 -14.020 8.0491e-14 ***
LYD - FOR == 0  -7.534 3.5540e-06 ***
MOF - FOR == 0  -3.687  0.1832615    
SWD - FOR == 0 -20.599 < 2.22e-16 ***
WIS - FOR == 0  -9.667 2.9347e-10 ***
LYD - HOD == 0  -4.448  0.0438712   *
MOF - HOD == 0  -8.687 2.9208e-08 ***
SWD - HOD == 0  -4.999  0.0121987   *
WIS - HOD == 0  -6.819 4.9851e-05 ***
MOF - LYD == 0  -7.001 2.6079e-05 ***
SWD - LYD == 0 -11.181 2.2315e-13 ***
WIS - LYD == 0  -1.911  0.9156345    
SWD - MOF == 0 -16.750 < 2.22e-16 ***
WIS - MOF == 0  -5.426  0.0039668  **
WIS - SWD == 0 -15.949 1.3989e-14 ***
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

Almost all pair-wise comparisons are significant!



Is there significant difference between pops using different HostPlants or with different colonisation histories? 

```
##First add in the info about history and hostplant

Hs.9pops$History <- Hs.9pops$Var2
Hs.9pops$History <- (Hs.9pops$History %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))
head(Hs.9pops)

Hs.9pops$HostPlatn <- Hs.9pops$Var2
Hs.9pops$HostPlatn <- (Hs.9pops$HostPlatn %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


## Then calculate whether the groups are significantly different. 

Hs.9pops$History <- as.factor(Hs.9pops$History)
kruskal.test(value~History, data=Hs.9pops)


	Kruskal-Wallis rank sum test

data:  value by History
Kruskal-Wallis chi-squared = 143.24, df = 1, p-value < 2.2e-16



Hs.9pops$HostPlatn <- as.factor(Hs.9pops$HostPlatn)
kruskal.test(value~HostPlatn, data=Hs.9pops)

	Kruskal-Wallis rank sum test

data:  value by HostPlatn
Kruskal-Wallis chi-squared = 95.455, df = 1, p-value < 2.2e-16

```

This remained significant after removing one population at a time. I also checked whether a ranomly assigned pop would be significant: 

```
sample(1:2, 9, replace=T) #randomly assign 1 or 2 (with replacement) to each population. 
[1] 1 2 2 1 2 1 1 2 1

Hs.9pops$rand <- (Hs.9pops$rand %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "2", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "1", .) %>% gsub("WIS", "1", .) %>% gsub("BRO", "2", .) %>% gsub("FOR", "1", .))
Hs.9pops$rand <- as.factor(Hs.9pops$rand)

kruskal.test(value~rand, data=Hs.9pops)  ##check whether a random grouping of these data would be significant. 

	Kruskal-Wallis rank sum test

data:  value by rand
Kruskal-Wallis chi-squared = 4.9303, df = 1, p-value = 0.02639

```

This is significant, but much less so than the History and HostPlant groupings. 





##### 4. Linkage disequilibrium

As this is RAD data we can't use windows of x number of variants - this leaves us with too few scaffolds to work with. Instead I assume that all loci are independent and I calculate linkage between them. 


```
kruskal.test(R.2~History, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by History
Kruskal-Wallis chi-squared = 1474.5, df = 1, p-value < 2.2e-16

> kruskal.test(R.2~HostPlant, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by HostPlant
Kruskal-Wallis chi-squared = 609.84, df = 1, p-value < 2.2e-16

> sample(1:2, 9, replace=T)
[1] 2 2 2 1 2 2 2 1 2
> ld.9pops$rand <- (ld.9pops$pop %>% gsub("BCH", "2", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "2", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "2", .) %>% gsub("MOF", "2", .) %>% gsub("WIS", "2", .) %>% gsub("BRO", "1", .) %>% gsub("FOR", "2", .))

> ld.9pops$rand <- as.factor(ld.9pops$rand)
> kruskal.test(R.2~rand, data=ld.9pops)

	Kruskal-Wallis rank sum test

data:  R.2 by rand
Kruskal-Wallis chi-squared = 6466.8, df = 1, p-value < 2.2e-16

```


Everything is significantly different. 
There might be something interesting going on with the differences in LD. See the plot below



And plot
```
pdf("9pops.LD.pdf")
ggplot(ld.9pops, aes(R.2, x=pop)) + geom_boxplot()
dev.off()

```

![alt_txt][LD.9pops]

[LD.9pops]:https://user-images.githubusercontent.com/12142475/54035457-4f7b2c80-41b1-11e9-99a6-1b1916222a28.png



##### 5. Genome-wide Tajima's D

This needs to be calculated for each population separately. 

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261

```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TajimaD 1000 && mv out.TajimaD > $i.TajimaD; done

for i in $(ls *TajimaD); do rename 's/recode.vcf.maxmiss.recode.vcf.gz.//' $i; done

```
	
Read into R and plot
```
TajimaD.9pops <- read.table("PerPopVCFfiles_AA261/9pops.TajimaD", header=T)
head(TajimaD.9pops)
TajimaD.9pops$pop <- "pop"
TajimaD.9pops$pop[1:8850] <- "BAR"
TajimaD.9pops$pop[8851:18063] <- "BCH"
TajimaD.9pops$pop[18064:26154] <- "BRO"
TajimaD.9pops$pop[26155:33720] <- "FOR"
TajimaD.9pops$pop[33721:42498] <- "HOD"
TajimaD.9pops$pop[42499:50841] <- "LYD"
TajimaD.9pops$pop[50842:59722] <- "MOF"
TajimaD.9pops$pop[59723:68609] <- "SWD"
TajimaD.9pops$pop[68610:77643] <- "WIS"
TajimaD.9pops$pop[68610:77642] <- "WIS"
head(TajimaD.9pops)

TajimaD.9pops$History <- (TajimaD.9pops$pop %>% gsub("BCH", "old", .) %>% gsub("LYD", "old", .) %>% gsub("SWD","old", .) %>% gsub("BAR", "new", .) %>% gsub("HOD","old", .) %>% gsub("MOF", "new", .) %>% gsub("WIS", "new", .) %>% gsub("BRO", "new", .) %>% gsub("FOR", "old", .))
TajimaD.9pops$HostPlant <- (TajimaD.9pops$HostPlant %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))


TajimaD.9pops$HostPlant <- (TajimaD.9pops$pop %>% gsub("BCH", "Rockrose", .) %>% gsub("LYD", "Rockrose", .) %>% gsub("SWD","Rockrose", .) %>% gsub("BAR", "Rockrose", .) %>% gsub("HOD","Geranium", .) %>% gsub("MOF", "Geranium", .) %>% gsub("WIS", "Geranium", .) %>% gsub("BRO", "Geranium", .) %>% gsub("FOR", "Rockrose", .))

pdf("TajimaD.9pops.pdf")
ggplot(TajimaD.9pops, aes(pop, TajimaD)) + geom_boxplot()
dev.off()
```

![alt_txt][Tajima.9pops]

[Tajima.9pops]:https://user-images.githubusercontent.com/12142475/54032947-ef818780-41aa-11e9-86fd-018d1f9f7b9c.png



Are these significantly different between groups? and populations?

```
summary(dscfAllPairsTest(x=TajimaD.9pops$TajimaD, g=TajimaD.9pops$pop, data=TajimaD.9pops, formula=TajimaD.9pops$TajimaD~TajimaD.9pops$pop))

	Pairwise comparisons using Dwass-Steele-Critchlow-Fligner all-pairs test

data: TajimaD.9pops$TajimaD and TajimaD.9pops$pop
P value adjustment method: single-step
H0
               q value   Pr(>|q|)    
BCH - BAR == 0  -2.172 0.83881716    
BRO - BAR == 0  -1.299 0.99198927    
FOR - BAR == 0  -4.697 0.02521420   *
HOD - BAR == 0  -0.915 0.99932365    
LYD - BAR == 0  -5.135 0.00863204  **
MOF - BAR == 0  -2.455 0.72401045    
SWD - BAR == 0  -5.665 0.00202078  **
WIS - BAR == 0  -0.245 0.99999998    
BRO - BCH == 0  -3.444 0.26474808    
FOR - BCH == 0  -6.905 3.6782e-05 ***
HOD - BCH == 0  -1.313 0.99140709    
LYD - BCH == 0  -3.019 0.44927805    
MOF - BCH == 0  -0.402 0.99999877    
SWD - BCH == 0  -3.626 0.20205289    
WIS - BCH == 0  -2.382 0.75651465    
FOR - BRO == 0  -3.249 0.34371176    
HOD - BRO == 0  -2.267 0.80365939    
LYD - BRO == 0  -6.282 0.00030581 ***
MOF - BRO == 0  -3.540 0.23007207    
SWD - BRO == 0  -6.726 6.9144e-05 ***
WIS - BRO == 0  -1.045 0.99823664    
HOD - FOR == 0  -5.660 0.00205234  **
LYD - FOR == 0  -9.649 3.2173e-10 ***
MOF - FOR == 0  -6.831 4.7887e-05 ***
SWD - FOR == 0 -10.140 2.7060e-11 ***
WIS - FOR == 0  -4.370 0.05175128   .
LYD - HOD == 0  -4.253 0.06576526   .
MOF - HOD == 0  -1.558 0.97414199    
SWD - HOD == 0  -4.830 0.01845235   *
WIS - HOD == 0  -1.142 0.99669757    
MOF - LYD == 0  -2.564 0.67372302    
SWD - LYD == 0  -0.516 0.99999132    
WIS - LYD == 0  -5.298 0.00561958  **
SWD - MOF == 0  -2.982 0.46750966    
WIS - MOF == 0  -2.667 0.62346674    
WIS - SWD == 0  -5.812 0.00131175  **
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

And for History and Hostplant

```
> kruskal.test(TajimaD~History, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by History
Kruskal-Wallis chi-squared = 6.3129, df = 1, p-value = 0.01199

> kruskal.test(TajimaD~HostPlant, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by HostPlant
Kruskal-Wallis chi-squared = 3.6057, df = 1, p-value = 0.05758


##check if this holds true for a random rearrangement of the History and HostPlant catagories

sample(1:2, 9, replace=T)
[1] 1,2,1,1,1,2,1,1,2

TajimaD.9pops$rand <- (TajimaD.9pops$pop %>% gsub("BCH", "1", .) %>% gsub("LYD", "2", .) %>% gsub("SWD", "1", .) %>% gsub("BAR", "1", .) %>% gsub("HOD", "1", .) %>% gsub("MOF", "2", .) %>% gsub("WIS", "1", .) %>% gsub("BRO", "1", .) %>% gsub("FOR", "2", .))


kruskal.test(TajimaD~rand, data=TajimaD.9pops)

	Kruskal-Wallis rank sum test

data:  TajimaD by rand
Kruskal-Wallis chi-squared = 0.019755, df = 1, p-value = 0.8882
```

Again, FOR and MOF seem to be driving this pattern.
When either of these are removed the test is highly significant. But NS if any of the other populations are removed. 




##### 6. Estimate Ts/Tv ration in every population

/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL/PerPopVCFfiles_AA261
```
for i in $(ls *vcf.gz); do vcftools --gzvcf $i --TsTv 1000 && mv out.TsTv $i.TsTv; done

BAR.vcf.gz
Ts/Tv ratio: 1.232

BCH.vcf.gz
Ts/Tv ratio: 1.232

BRO.vcf.gz
Ts/Tv ratio: 1.239

FOR.vcf.gz
Ts/Tv ratio: 1.241

HOD.vcf.gz
Ts/Tv ratio: 1.231

LYD.vcf.gz
Ts/Tv ratio: 1.23

MOF.vcf.gz
Ts/Tv ratio: 1.232

SWD.vcf.gz
Ts/Tv ratio: 1.229

WIS.vcf.gz
Ts/Tv ratio: 1.234
```
	
Pretty much the same between all populations. 


## Population Structure

### Fst and IBD

First calculate mean Fst overall and then for all the group comparisons: 

Vignette [here](http://adegenet.r-forge.r-project.org/files/tutorial.pdf)
```
getwd()
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/DiversityStats_AA264.41508_FINAL

library(adegenet)
library(hierfstat)

fstat(AA264.1000.genind)  ##overall Fst
             pop       Ind
Total 0.03372121 0.3598001
pop   0.00000000 0.3374584

boot.fst.ALLpops <- boot.ppfst(data.frame(as.numeric(pop),AA264.1000.hier[,-c(1)]))
 boot.fst.ALLpops$ul
      [,1]   [,2]   [,3]   [,4]   [,5]   [,6]   [,7]   [,8]   [,9]
 [1,]   NA 0.0506 0.0258 0.0284 0.0366 0.0442 0.0110 0.0432 0.0187
 [2,]   NA     NA 0.0757 0.0538 0.0339 0.0295 0.0578 0.0137 0.0707
 [3,]   NA     NA     NA 0.0346 0.0546 0.0694 0.0262 0.0636 0.0161
 [4,]   NA     NA     NA     NA 0.0362 0.0574 0.0203 0.0401 0.0260
 [5,]   NA     NA     NA     NA     NA 0.0440 0.0376 0.0294 0.0423
 [6,]   NA     NA     NA     NA     NA     NA 0.0642 0.0255 0.0657
 [7,]   NA     NA     NA     NA     NA     NA     NA 0.0480 0.0157
 [8,]   NA     NA     NA     NA     NA     NA     NA     NA 0.0588
 [9,]   NA     NA     NA     NA     NA     NA     NA     NA     NA
 
 boot.fst.ALLpops$ll
      [,1]   [,2]   [,3]   [,4]   [,5]   [,6]   [,7]   [,8]   [,9]
 [1,]   NA 0.0363 0.0108 0.0123 0.0237 0.0262 0.0028 0.0289 0.0094
 [2,]   NA     NA 0.0544 0.0375 0.0223 0.0177 0.0386 0.0070 0.0503
 [3,]   NA     NA     NA 0.0168 0.0373 0.0468 0.0093 0.0456 0.0043
 [4,]   NA     NA     NA     NA 0.0199 0.0333 0.0069 0.0243 0.0136
 [5,]   NA     NA     NA     NA     NA 0.0254 0.0251 0.0191 0.0302
 [6,]   NA     NA     NA     NA     NA     NA 0.0387 0.0119 0.0471
 [7,]   NA     NA     NA     NA     NA     NA     NA 0.0350 0.0068
 [8,]   NA     NA     NA     NA     NA     NA     NA     NA 0.0453
 [9,]   NA     NA     NA     NA     NA     NA     NA     NA     NA


#Now change the pops to Old and New
pop.ColHist <- gsub("BAR", "New", pop)
pop.ColHist <- gsub("BRO", "New", pop.ColHist)
pop.ColHist <- gsub("FOR", "New", pop.ColHist)
pop.ColHist <- gsub("MOF", "New", pop.ColHist)
pop.ColHist <- gsub("WIS", "New", pop.ColHist)
pop.ColHist <- gsub("BCH", "Est", pop.ColHist)
pop.ColHist <- gsub("HOD", "Est", pop.ColHist)
pop.ColHist <- gsub("SWD", "Est", pop.ColHist)
pop.ColHist <- gsub("LYD", "Est", pop.ColHist)

pop.ColHist <- as.factor(pop.ColHist)
AA264.1000.genind.ColHist <- AA264.1000.genind
AA264.1000.genind.ColHist@pop <- pop.ColHist

fstat(AA264.1000.genind.ColHist)
             pop       Ind
Total 0.03340853 0.3678850
pop   0.00000000 0.3460371

AA264.1000.hier.ColHist <- AA264.1000.hier
AA264.1000.hier.ColHist$pop <- pop.ColHist


##Change pops to Ger and RR
pop.HostPlant <- gsub("BAR", "RR", pop)
pop.HostPlant <- gsub("BCH", "RR", pop.HostPlant)
pop.HostPlant <- gsub("FOR", "RR", pop.HostPlant)
pop.HostPlant <- gsub("LYD", "RR", pop.HostPlant)
pop.HostPlant <- gsub("SWD", "RR", pop.HostPlant)
pop.HostPlant <- gsub("BRO", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("HOD", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("MOF", "Ger", pop.HostPlant)
pop.HostPlant <- gsub("WIS", "Ger", pop.HostPlant)

pop.HostPlant <- as.factor(pop.HostPlant)
AA264.1000.genind.HostPlant <- AA264.1000.genind
AA264.1000.genind.HostPlant@pop <- pop.HostPlant

fstat(AA264.1000.genind.HostPlant)
             pop       Ind
Total 0.01973744 0.3635152
pop   0.00000000 0.3506997

AA264.1000.hier.HostPlant <- AA264.1000.hier
AA264.1000.hier.HostPlant$pop <- pop.HostPlant

####Couldn't get bootstrap to work for subset populations. And I couldn't subset the datasets (i.e. to test all new pops or all Ger pops)

#Overall Weir & Cockerham Fst

wc(AA264.1000.genind)
$FST
[1] 0.03372121

$FIS
[1] 0.3374584



```


I calculated population pairwise Fst using adegenet. This was based on a subset of 1000 loci, one SNP per locus. 
```
library(adegenet)

AA264.1000.fst <- pairwise.fst(AA264.genind)
```


I tested if genetic distance increased significantly with 1) geographic distance, 2) Difference in colonisation history (0/1), and 3) Difference in dominant host plant (0/1). 

```
Aa264.genpop <- genind2genpop(AA264.genind)
Dgen <- dist.genpop(AA264.genpop, method=1)  ##Nei's genetic distance (1972)
AA_lon.lat <- cbind(pops.9.coords$Long, CHall.coords$Lat)
Dgeo <- rdist.earth(AA_lon.lat, miles=F)
Dgeo <- log(Dgeo) ##log of distance as suggeted for tests of IBD by Rousset. 
DHostPlant <- dist(pops.9.coords$HostPlant)

DColHist <- dist(pops.9.coords$colhist)

##Significance of genetic distance vs geographic distance
mantel(Dgen, Dgeo)

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = Dgeo) 

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = Dgeo) 

Mantel statistic r: 0.7606 
      Significance: 0.001 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.237 0.309 0.395 0.442 
Permutation: free
Number of permutations: 999


##Significance of genetic distance vs HostPlant

Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = DHostPlant) 

Mantel statistic r: 0.2183 
      Significance: 0.103 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.218 0.318 0.453 0.501 
Permutation: free
Number of permutations: 999


##Significance of genetic distance vs ColHistory
Mantel statistic based on Pearson's product-moment correlation 

Call:
mantel(xdis = Dgen, ydis = DColHist) 

Mantel statistic r: 0.5009 
      Significance: 0.012 

Upper quantiles of permutations (null model):
  90%   95% 97.5%   99% 
0.193 0.380 0.453 0.501 
Permutation: free
Number of permutations: 999
```






### PCA

PCA is affecteted by missing data. As this file contains only loci that have been genotyped across all populations, I will use AA261 as is. 

```
bluecp3:/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/PCAdapt

vcftools --vcf AA261.0.5miss.9popsMerged.vcf 

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.0.5miss.9popsMerged.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 31381 out of a possible 31381 Sites
Run Time = 1.00 seconds




```

Convert to plink

```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --plink --out AA261.plink

```





Copy to Mac
```
scp aj18951@bluecrystalp3.acrc.bris.ac.uk:/panfs/panasas01/bisc/aj18951/1a_Aricia_agestis_PopGenomics/PCAdapt/AA261.plink* .

plink --file AA261.plink --recode A --out AA261.plink

PLINK v1.90b6.2 64-bit (12 Jun 2018)           www.cog-genomics.org/plink/1.9/
(C) 2005-2018 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to AA261.plink.log.
Options in effect:
  --file AA261.plink
  --out AA261.plink
  --recode A

16384 MB RAM detected; reserving 8192 MB for main workspace.
.ped scan complete (for binary autoconversion).
Performing single-pass .bed write (31381 variants, 261 people).
--file: AA261.plink-temporary.bed + AA261.plink-temporary.bim +
AA261.plink-temporary.fam written.
31381 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.plink.nosex .
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921137.
31381 variants and 261 people pass filters and QC.
Note: No phenotypes present.
--recode A to AA261.plink.raw ... done.


```

92% genotyping rate


Open R and run analysis
```
R version 3.5.0

library(pcadapt)

AA261 <- read.pcadapt("", type="ped")

Summary:

	- input file:				AA261.plink.ped
	- output file:				/var/folders/d1/2957hj5928x_7dc6xjx_2rx40000gq/T//Rtmp0bqHRx/file130141339c9fc.pcadapt

	- number of individuals detected:	261
	- number of loci detected:		31381

31381 lines detected.
261 columns detected.


x.AA261 <- pcadapt(AA261, K=20)



plot(x.AA261, option="screeplot")  ##PC for pop structure = on the steep curve
```

![alt_txt][nr_pcaclusters]

[nr_pcaclusters]:https://user-images.githubusercontent.com/12142475/54189040-e1807f00-44a8-11e9-9d0f-93f9d59b3652.png



Based on this I choose K=2. The random eigenvalues fall in the straight line, while the ones explaining structure are above this.

See vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html on choosing K.

Plot the PCA using population information


```
pop.AA261 <- read.table("AA.261.names", header=F) ##read in names acquired from the vcf file. 

pop.AA264$V1[!pop.AA264$V1 %in% pop.AA261$V1]  #check names of individuals that were removed
[1] BRO_6_2013  WIS_13_2013 WIS_16_2013
264 Levels: BAR_10_2013 BAR_11_2014 BAR_12_2013 BAR_13_2014 ... WIS_9_2014

pop.AA261 <- pop.AA264[c(-80, -223, -225),] ##remove the three indivs from the full file

summary(pop.AA261)
           V1            V2            V3        V4            V5       
 BAR_10_2013:  1   WIS    :45   geranium:118   new:116   Min.   :1.000  
 BAR_11_2014:  1   SWD    :39   rockrose:146   old:148   1st Qu.:3.000  
 BAR_12_2013:  1   BCH    :38                            Median :5.000  
 BAR_13_2014:  1   HOD    :30                            Mean   :4.727  
 BAR_14_2013:  1   BAR    :28                            3rd Qu.:7.000  
 BAR_14_2014:  1   MOF    :25                            Max.   :9.000  
 (Other)    :258   (Other):59                                           
       V6              V7                V8                  V9     
 Min.   :51.00   Min.   :-1.2189   Min.   :22.00   darkorchid3: 88  
 1st Qu.:51.62   1st Qu.:-1.0394   1st Qu.:22.00   darkorchid4: 30  
 Median :52.97   Median :-0.6404   Median :22.00   gold1      : 28  
 Mean   :52.50   Mean   :-0.4365   Mean   :22.88   goldenrod  :118  
 3rd Qu.:53.19   3rd Qu.:-0.1689   3rd Qu.:24.00                    
 Max.   :54.16   Max.   : 1.2694   Max.   :24.00                    
                                                                    
summary(pop.AA261$V2)
BAR BCH BRO FOR HOD LYD MOF SWD WIS 
 28  38  17  20  30  21  25  39  43 



poplist <- as.character(pop.AA261[,2])

plot(x.AA261, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA261[,3]) ##pca coloured by host plant presence. 
plot(x.AA261, option="scores", pop=poplist) ##pca with all populations specified

poplist <- as.character(pop.AA261[,4]) ##pca coloured by new vs old
plot(x.AA261, option="scores", pop=poplist) ##pca with all populations specified
```


![alt_txt][PCApops]

[PCApops]:https://user-images.githubusercontent.com/12142475/54190902-b6982a00-44ac-11e9-99a0-50492a6039f9.png


![alt_txt][PCAcolHist]

[PCAcolHist]:https://user-images.githubusercontent.com/12142475/54190916-c3b51900-44ac-11e9-837a-ca3ee7dc9f74.png



![alt_txt][PCAhostplant]

[PCAhostplant]:https://user-images.githubusercontent.com/12142475/54190961-d596bc00-44ac-11e9-8c19-b950830937e1.png



Plot with the correct shape and colours. 
```
library(ggplot2)

pop.AA264$PC1 <- x.AA264$scores[,1]
pop.AA264$PC2 <- x.AA264$scores[,2]

pdf("PCA264.forMS.pdf")
ggplot(pop.AA264, aes(x=PC1, y=PC2, colour=V3, shape=V4, size=1)) + geom_point() + scale_shape_manual(values = c(17,15)) + scale_color_manual(values=c("purple", "goldenrod1"))
dev.off()

```


![alt_txt][PCAfinal]

[PCAfinal]:https://user-images.githubusercontent.com/12142475/54742011-09ae6300-4bb8-11e9-9a96-172bdc3595cd.png


### fineRADstructure

[fineRADstructure](http://cichlid.gurdon.cam.ac.uk/fineRADstructure.html)[(Malinksy et al. 2018)](https://academic.oup.com/mbe/article/35/5/1284/4883220)
uses haplotype information from RAD data to infer a co-ancestry matrix. The dataset utilises linkage information, so all loci should be kept. 

To create haplotype data, all variant loci from a single scaffold is concatenated into a sequence string. These data can be extracted from the vcf file. 

I've used the [vcf2hapmatrix.py](https://github.com/pimbongaerts/radseq/blob/master/vcf2hapmatrix.py) script to convert my vcf file to an input file for fineRADstructure. 
This script assumes that the scaffolds are in order. It concatenates all the nucleotide bases found on a specific scaffold together for each individual. This means many linked loci are preserved in the final dataset. 

The file conversion uses python3, and needs the module PyVCF. Conversions are run on the mac laptop. 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure

./vcf2hapmatrix.py AA261.0.5miss.9popsMerged.vcf > AA261.fineradstr

cp ~/Software/fineRADstructure/RADpainter .

./RADpainter paint AA261.fineradstr 
Painting RAD tags from: AA261.fineradstr
The file seems to be in a Matrix format
Processed: 100 tag loci in 7.10749 seconds(0.0710749 seconds per RAD locus)
...
```




### fastStructure

Get the population assignment Q values at K=2

Convert vcf to fastStructure format using pgdspider. Remember to add the extension .str to the input file. Run at K=2
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fastStructure

python /Users/alexjvr/Applications/fastStructure/fastStructure/structure.py -K 2 --format=str --input=AA261.thin.faststr --output=AA261.faststr_K2.2


```

Import the Q-values into R and plot

```
library(ggplot2)
library(reshape2)

df <- read.table("AA261.faststr_K2.2.meanQ", header=F)
popnames <- read.table("../Bayescan/InputFiles/popnames.pgdspider")
df$pop <- popnames$V2
df$indiv <- popnames$V1

df$N2S <- df$pop
head(df$N2S)
df$N2S <- gsub("BCH", "1", df$N2S)
df$N2S <- gsub("LYD", "2", df$N2S)
df$N2S <- gsub("SWD", "3", df$N2S)
df$N2S <- gsub("BAR", "4", df$N2S)
df$N2S <- gsub("HOD", "5", df$N2S)
df$N2S <- gsub("MOF", "6", df$N2S)
df$N2S <- gsub("WIS", "7", df$N2S)
df$N2S <- gsub("BRO", "8", df$N2S)
df$N2S <- gsub("FOR", "9", df$N2S)

df$index <- 1:nrow(df)

attach(df)
df.sorted <- df[order(N2S),]
df2 <- melt(df.sorted, id=c("pop", "indiv", "index", "N2S"),)

pdf("AA261.faststrK2.pdf")
ggplot(df2) + geom_bar(aes(x=index, y=value, fill=variable), stat="identity")
dev.off()
```


### Figure

Combine the map, PCA, and fastructure together in a figure. 

![alt_txt][Fig1]

[Fig1]:https://user-images.githubusercontent.com/12142475/54742678-fdc3a080-4bb9-11e9-9483-c5d10468f1a7.png
