# UK colonisation history

Where were the new populations colonised from? We will use fastSimCoal to estimate the most likely colonisation history. 

We do not have the ancestral state, so I have to generate the [folded SFS](https://www.mun.ca/biology/scarr/SFS_&_FSFS.html) for each of the populations or groups of populations. That is, if all derived alleles are identified as compared with Indiv 1 (a random indivi in the population), then the frequency of the derived alleles are combined for classes i and (n-i), where i is the number of indivs in the population. 

An [example paper](https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.14951#nph14951-fig-0002) from the Sork lab. 

And [another](https://onlinelibrary.wiley.com/doi/full/10.1111/mec.15132?casa_token=iTjBMCiloY4AAAAA%3AY2wNHk41rqTXJXb077LRnuAsWMoe0LGFBuPcFpLA0qLJ3gCpHeGePPmHqnMQc2C_LVXe7nTOcfbNcE8) from the Laurila group 


## Filtering

1. The dataset is filtered to remove any loci under selection

2. Populations are downsampled to exclude all missing data using a python script available from the paper. 

3. To exclude linked loci only one variant is included per RADtag. 

This was done on my computer
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/FastSimCoal

vcftools --vcf AA261.0.5miss.9popsMerged.vcf 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.0.5miss.9popsMerged.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 31381 out of a possible 31381 Sites
```

First remove all loci under selection. For this I need to identify a list of outliers for 1) host plant preference, and 2) colonisation history. One of these overlaps. I'll remove the entire scaffold or contig that contains outlier loci. 
```


```

Split the data into four groups of interest
```


```



## Input files

Generating the input files for fastSimCoal2

### 1) SFS

I'm using [this](https://github.com/shenglin-liu/vcf2sfs) R script to generate 2D folded SFS for FOR (called North), HOD, South, and New populations. 



### 2) Priors

I'm defining priors for fastSimCoal rather than specifying values for the parameters. 


mutation rate - 

nucleotide diversity - 


### 3) Population input files



## Scenarios

We're testing three main scenarios. 

1. Col from the south northwards

2. Col from the east

3. Col from the north. 

We'll also include colonisation from all three with mixture between them. 

The scenarios are complicated because we don't know the relationship between all the old populations. So I will run an initial analysis to ascertain the most likely colonisation history of the old populations first. 

```


```



## Folded SFS

Input data needs to be filtered, and the the (joint) folded SFS should be calculated for each of the groups of populations. 

1. North old

2. South old

3. HOD

4. New


### Data filtering: 


```


```
