# UK colonisation history

Where were the new populations colonised from? We will use fastSimCoal to estimate the most likely colonisation history. 

We do not have the ancestral state, so I have to generate the [folded SFS](https://www.mun.ca/biology/scarr/SFS_&_FSFS.html) for each of the populations or groups of populations. That is, if all derived alleles are identified as compared with Indiv 1 (a random indiv in the population), then the frequency of the derived alleles are combined for classes i and (n-i), where i is the number of indivs in the population. 

An [example paper](https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.14951#nph14951-fig-0002) from the Sork lab. 

And [another](https://onlinelibrary.wiley.com/doi/full/10.1111/mec.15132?casa_token=iTjBMCiloY4AAAAA%3AY2wNHk41rqTXJXb077LRnuAsWMoe0LGFBuPcFpLA0qLJ3gCpHeGePPmHqnMQc2C_LVXe7nTOcfbNcE8) from the Laurila group 


## Filtering

1. The dataset is filtered to remove any loci under selection

2. Populations are downsampled to exclude all missing data using a python script available from the paper. 

3. To exclude linked loci only one variant is included per RADtag. 

This was done on my computer
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/FastSimCoal

vcftools --vcf AA261.0.5miss.9popsMerged.vcf 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.0.5miss.9popsMerged.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 31381 out of a possible 31381 Sites
```

First remove all loci under selection. For this I need to identify a list of outliers for 1) host plant preference, and 2) colonisation history. Three of these overlaps. I'll remove the entire scaffold or contig that contains outlier loci. These vcf files have already been generated for fineRADstructure: 
```
## ColHistory 

--vcf /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.ColHistOutlierScaffolds.recode.vcf 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.ColHistOutlierScaffolds.recode.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 169 out of a possible 169 Sites
Run Time = 1.00 seconds


Col hist
	--chr contig_16912
	--chr contig_1784
	--chr contig_18281
	--chr contig_1883
	--chr contig_19343	** in both
	--chr contig_19564
	--chr contig_19566
	--chr contig_24918
	--chr contig_3360
	--chr contig_3838	** in both  
	--chr contig_5345	** in both
	--chr contig_59667


## HostPlant

vcftools --vcf /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.HostPlantOutlierScaffolds.recode.vcf 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.HostPlantOutlierScaffolds.recode.vcf

After filtering, kept 261 out of 261 Individuals
After filtering, kept 109 out of a possible 109 Sites
Run Time = 0.00 seconds


HostPlant
	--chr contig_11951
	--chr contig_17378
	--chr contig_19343	** in both
	--chr contig_2518
	--chr contig_3838	** in both
	--chr contig_5345	** in both
	--chr contig_5407
	--chr m_scaff_962
	
#Remove these from main vcf file

vcftools --gzvcf AA261.0.5miss.9popsMerged.vcf.gz --not-chr m_scaff_962 --not-chr contig_5407 --not-chr contig_5345 --not-chr contig_3838 --not-chr contig_2518 --not-chr contig_19343 --not-chr contig_17378 --not-chr contig_11951 --not-chr contig_16912 --not-chr contig_1784 --not-chr contig_18281 --not-chr contig_1883 --not-chr contig_19564 --not-chr contig_19566 --not-chr contig_24918 --not-chr contig_3360 --not-chr contig_59667 --recode --recode-INFO-all --out AA261.0.5miss.9popsMerged.Neutral


...After filtering, kept 31158 out of a possible 31381 Sites
```

Downsample to one SNP per locus to minimise linkage disequilibrium in the dataset. RAD tags are max 600bp, but linkage blocks could be larger. I'm using --thin to downsample to one SNP per 10kb. 
```
vcftools --vcf AA261.0.5miss.9popsMerged.Neutral.recode.vcf --thin 10000 --recode --recode-INFO-all --out AA261.Neutral.thin

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 3486 out of a possible 31158 Sites
Run Time = 4.00 seconds


##If thinning for 600bp (i.e. size of RAD tags) we get 3798 loci. 
```

Remove missing data
```

```


### This bit didn't work. See above for correct code
Copy these vcf files to fastSimCoal folder and remove scaffolds from vcffile
```
pwd
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/FastSimCoal

cp /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.HostPlantOutlierScaffolds.recode.vcf .
cp /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure/AA261.ColHistOutlierScaffolds.recode.vcf .


##Merge outlier vcf files 

I had to install a new version of vcftools in the Applications folder: 
git clone https://github.com/vcftools/vcftools.git

vcftools

VCFtools (0.1.17)
Â© Adam Auton and Anthony Marcketta 2009

Process Variant Call Format files

For a list of options, please go to:
	https://vcftools.github.io/man_latest.html

Alternatively, a man page is available, type:
	man vcftools

Questions, comments, and suggestions should be emailed to:
	vcftools-help@lists.sourceforge.net


## Remove all the outlier scaffolds with vcftools 

#First link to the perl modules
export PERL5LIB=/Users/alexjvr/Applications/vcftools/src/perl

#Concat the outlier vcf files
vcf-concat AA261.ColHistOutlierScaffolds.recode.vcf AA261.HostPlantOutlierScaffolds.recode.vcf > AA261.OutlierScaffolds.vcf

```
### End of approach that didn't work. 



Split the data into four groups of interest

Popnames
```
#Extract all sample names
bcftools query -l AA261.0.5miss.9popsMerged.Neutral.recode.vcf

#create files for each population subset
head HOD.popnames 
HOD_10_2013
HOD_1_2013
HOD_12_2014
HOD_14_2013
HOD_15_2014_all
HOD_16_2013
HOD_16_2014


head South.old.popnames 
BCH_10_2013
BCH_1_2014
BCH_12_2013
BCH_12_2014
BCH_14_2014
BCH_15_2013
BCH_15_2014
BCH_16_2013
BCH_19_2013
BCH_19_2014

North.Old.popnames 
FOR_11_2014
FOR_12_2014
FOR_13_2014
FOR_14_2014
FOR_15_2014
FOR_17_2014
FOR_19_2014
FOR_20_2014


```


```
bcftools view -S HOD.popnames AA261.Neutral.thin.recode.vcf > HOD.Neutral.vcf
...After filtering, kept 30 out of 30 Individuals
...After filtering, kept 3486 out of a possible 3486 Sites


bcftools view -S South.old.popnames AA261.Neutral.thin.recode.vcf > South.Neutral.vcf
...After filtering, kept 98 out of 98 Individuals
...After filtering, kept 3486 out of a possible 3486 Sites


bcftools view -S North.Old.popnames AA261.Neutral.thin.recode.vcf > North.Neutral.vcf
...After filtering, kept 20 out of 20 Individuals
...After filtering, kept 3486 out of a possible 3486 Sites


```



## Input files

Generating the input files for fastSimCoal2

### 1) SFS

I'm using [this](https://github.com/shenglin-liu/vcf2sfs) R script to generate 2D folded SFS for FOR (called North), HOD, South, and New populations. 

I need a vcf file with a popmap file: 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/FastSimCoal

#write all the indiv names to file
bcftools query -l AA261.Neutral.thin.recode.vcf > popmap.AA261 
#write all the popnames to file
awk -F'_' '{print $0}' popmap.AA261 > popnames
#add together
print popmap.AA261 popnames > popmap.AA261.2
rm popmap.AA261 && mv popmap.AA261.2 popmap.AA261
head popmap.AA261 
BAR_10_2013	BAR
BAR_11_2014	BAR
BAR_12_2013	BAR
BAR_13_2014	BAR
```

And download the R script for the above link. The function vcf2sfs.r is stored in the FastSimCoal folder where all these analyses are being run.
Open R: 
```
source("vcf2sfs.r")
mygt <- vcf2gt("AA261.Neutral.thin.recode.vcf", "popmap.AA261")   ##read in the vcf file and popmap files

viewMissing(mygt) ##look at the distribution of missing data
```

#### Distribution of missing data in the dataset

![alt_txt][Missing.data]

[Missing.data]:https://user-images.githubusercontent.com/12142475/76212667-e3d30180-6200-11ea-9427-de8c11ee6a36.png



### 2) Priors

I'm defining priors for fastSimCoal rather than specifying values for the parameters. 


mutation rate - 

nucleotide diversity - 


### 3) Population input files



## Scenarios

We're testing three main scenarios. 

1. Col from the south northwards

2. Col from the east

3. Col from the north. 

We'll also include colonisation from all three with mixture between them. 

The scenarios are complicated because we don't know the relationship between all the old populations. So I will run an initial analysis to ascertain the most likely colonisation history of the old populations first. 

```


```



## Folded SFS

Input data needs to be filtered, and the the (joint) folded SFS should be calculated for each of the groups of populations. 

1. North old

2. South old

3. HOD

4. New


### Data filtering: 


```


```
