# ANGSD GL files and filtering

I'm using ANGSD to call GL for the fastSimCoal analysis. As this is a new pipeline I'm testing I need to test all the different filter parameters to use. 

I'm starting with bam files, so there are already some filters on the mapping quality of the sequences. Prior to that there are some crude filters during the demultiplexing and trimming steps. 

For this analysis I should use an outgroup or ancestral genotype. I could use A. arctaxerxes if enough SNPs are called. Otherwise I'll use the A.agestis genome and use the folded SFS. Note that Vitor Sousa and Laurent Excoffier suggest that ANGSDs folded SFS is incorrect. 
Thus I should generate unfolded SFS and fold using Vitor's script on his gitHub page. 

## Filters

### Input bam files

-b[filelist]

-remove_bads 1 : remove reads with 255 flag (not primary, failure and duplicate reads) (1=default)

-uniqueOnly 1 : remove reads with multiple best hits 

-minMapQ 20 : PHRED 20. This should already be in place during the mapping. 

-minQ 20 : PHRED 20 for individual base score. 

-only_proper_pairs 1 : include only properly paired reads (default)

-trim 0 : We're not trimming any data

-baq 1 : estimate base alignment quality using samtools method. 

### Filters for bam files

-minMaf 0.05 : set minimum MAF to 5% to exclude spurious data. 

- 

### Allele frequency estimation

-doMajorMinor 1 : estimate major and minor alleles from genotype likelihoods directly from bam files. 

-doMajorMinor 4 : Force Major allele based on reference (e.g. if using A.arctaxerxes). The minor allele is then inferred using doMajorMinor 1. 

-ref [..fasta] : For doMM 4 above we need to specify a reference genome. 

-doMaf 1 : calculate minor allele frequency 

-SNP_pval [float]: Only work with SNPs with a p-value above [float]



### Genotype likelihood

-GL 1 : I will estimate genotype likelihoods using the SAMtools model
-minInd 18 : I will remove loci where less than 18 individuals have been genotyped
