# Creating SFS inputs with ANGSD

We decided to use Angsd for the SFS for Brown Argus because the ddRAD data is very gappy - i.e. there is a lot of missingness in the data. This was also recommended to me by Vitor Sousa. He suggests ANGSD and resampling data iso imputing data. 

BUT ANGSD's folded SFS is not correct. So we have to estimate the unfolded SFS using either and ancestral genome (which we don't have) or an outgroup (A. artaxerxes would work). 

EDIT - I get very few loci when mapping to A. artaxerxes (<300 loci for most indivs!). So I'm using A. agestis genome again. 



First compile a list of bamfiles for the individuals from the final dataset for the rest of the paper: BA261
```
on BlueCrystal

bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/ANGSD/BA261.bamlist
```

Then generate SFS for all indivs in the bamlist by parallelising across the regions in the genome (as defined by the regions file from previous variant calling analyses. 
Use [this](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/angsd.sfs.sh) script. 


### Filtering

I'm using the following filters: 

minMAPQ (minimum mapping quality): PHRED 20

MINQ (minimum 

MinInd (min number of indivs): 18 per population

uniqueOnly : remove reads with multiple best hits

remove_bads

only_proper_pairs : keep only properly paired reads


Min and max depth: [Fumegalli et al. 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3813878/) showed that more loci (even at lower coverage) better estimates pop gen stats than fewer high coverage loci. 
So I will determine what the best depth range is for A. agestis. 





### Prepare bash script for ANGSD


The genome here has been split into 18 parts. So the SAF regions files need to be concatenated together: 

```
module load languages/gcc-6.1

~/bin/angsd/misc/realSFS cat BA.regionsab.saf.idx BA.regionsac.saf.idx -outnames testmerge
        -> This will cat together .saf files from angsd
	-> regions has to be disjoint between saf files. This WONT be checked (alot) !
	-> This has only been tested on safs for different chrs !
	-> Version of fname:BA.regionsab.saf.idx is:2
	-> Assuming .saf.gz file: BA.regionsab.saf.gz
	-> Assuming .saf.pos.gz: BA.regionsab.saf.pos.gz
	-> Version of fname:BA.regionsac.saf.idx is:2
	-> Assuming .saf.gz file: BA.regionsac.saf.gz
	-> Assuming .saf.pos.gz: BA.regionsac.saf.pos.gz
	-> outnames: 'testmerge' number of safs:2
	-> Merging 0/2 
	-> Merging 1/2 
  
~/bin/angsd/misc/realSFS cat BA.HOD.regionsaa.saf.idx BA.HOD.regionsab.saf.idx BA.HOD.regionsac.saf.idx BA.HOD.regionsad.saf.idx BA.HOD.regionsae.saf.idx BA.HOD.regionsaf.saf.idx BA.HOD.regionsag.saf.idx BA.HOD.regionsah.saf.idx BA.HOD.regionsai.saf.idx BA.HOD.regionsaj.saf.idx BA.HOD.regionsak.saf.idx BA.HOD.regionsal.saf.idx BA.HOD.regionsam.saf.idx BA.HOD.regionsan.saf.idx BA.HOD.regionsao.saf.idx BA.HOD.regionsap.saf.idx BA.HOD.regionsaq.saf.idx BA.HOD.regionsar.saf.idx -outnames BA.HOD.merge

~/bin/angsd/misc/realSFS cat BA.FOR.regionsaa.saf.idx BA.FOR.regionsab.saf.idx BA.FOR.regionsac.saf.idx BA.FOR.regionsad.saf.idx BA.FOR.regionsae.saf.idx BA.FOR.regionsaf.saf.idx BA.FOR.regionsag.saf.idx BA.FOR.regionsah.saf.idx BA.FOR.regionsai.saf.idx BA.FOR.regionsaj.saf.idx BA.FOR.regionsak.saf.idx BA.FOR.regionsal.saf.idx BA.FOR.regionsam.saf.idx BA.FOR.regionsan.saf.idx BA.FOR.regionsao.saf.idx BA.FOR.regionsap.saf.idx BA.FOR.regionsaq.saf.idx BA.FOR.regionsar.saf.idx -outnames BA.FOR.merge

~/bin/angsd/misc/realSFS cat BA.South.regionsaa.saf.idx BA.South.regionsab.saf.idx BA.South.regionsac.saf.idx BA.South.regionsad.saf.idx BA.South.regionsae.saf.idx BA.South.regionsaf.saf.idx BA.South.regionsag.saf.idx BA.South.regionsah.saf.idx BA.South.regionsai.saf.idx BA.South.regionsaj.saf.idx BA.South.regionsak.saf.idx BA.South.regionsal.saf.idx BA.South.regionsam.saf.idx BA.South.regionsan.saf.idx BA.South.regionsao.saf.idx BA.South.regionsap.saf.idx BA.South.regionsaq.saf.idx BA.South.regionsar.saf.idx -outnames BA.South.merge

~/bin/angsd/misc/realSFS cat BA.New.regionsaa.saf.idx BA.New.regionsab.saf.idx BA.New.regionsac.saf.idx BA.New.regionsad.saf.idx BA.New.regionsae.saf.idx BA.New.regionsaf.saf.idx BA.New.regionsag.saf.idx BA.New.regionsah.saf.idx BA.New.regionsai.saf.idx BA.New.regionsaj.saf.idx BA.New.regionsak.saf.idx BA.New.regionsal.saf.idx BA.New.regionsam.saf.idx BA.New.regionsan.saf.idx BA.New.regionsao.saf.idx BA.New.regionsap.saf.idx BA.New.regionsaq.saf.idx BA.New.regionsar.saf.idx -outnames BA.New.merge
```

And then a folded SFS can be created for each of the populations: 
```
~/bin/angsd/misc/realSFS testmerge.saf.idx -fold 1 > testmerge.saf.sfs

~/bin/angsd/misc/realSFS BA.HOD.saf.idx -fold 1 > BA.HOD.saf.sfs
~/bin/angsd/misc/realSFS BA.FOR.saf.idx -fold 1 > BA.FOR.saf.sfs
~/bin/angsd/misc/realSFS BA.South.saf.idx -fold 1 > BA.South.saf.sfs
~/bin/angsd/misc/realSFS BA.New.saf.idx -fold 1 > BA.New.saf.sfs
```

We need to create 2D+ SFS when using multiple populations. Test here how to make a 2D SFS with HOD and South: 
```
##2D folded SFS for initial test 
~/bin/angsd/misc/realSFS BA.South.merge.saf.idx BA.FOR.merge.saf.idx -fold 1 > BA.FOR.South.2dSFS.sfs


##3D folded SFS for base model (old sites only)

~/bin/angsd/misc/realSFS BA.South.merge.saf.idx BA.HOD.merge.saf.idx BA.FOR.merge.saf.idx -fold 1 > BA.South.HOD.FOR.3dSFS.sfs

##4D folded SFS for base model (all demes)
~/bin/angsd/misc/realSFS BA.South.merge.saf.idx BA.HOD.merge.saf.idx BA.FOR.merge.saf.idx BA.New.merge.saf.idx -fold 1 > BA.AllDemes.4dSFS.sfs
```

