# Are A.agestis specialised at range edge? 

Redundancy and pRDA to determine whether populations are specialised. 

Genetic variation can be explained by a number of things. We're interested in determining whether HostPlant prevalence or colonisation histroy has significantly affected the genetic variation. 


We'll used a contrained ordination (Redundancy Analysis - RDA) to test the marginal effect of each of these variables on the distribution of genetic variation. 

Variables of interest: 

1) Geographic distance, 2) HostPlant, or 3) Colonisation history. 


Input files

1:MAF within each population: 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --plink --out AA261.plink

plink --file AA261.plink --recodeA --out AA261.plink

head AA261.names

BAR_10_2013 BAR_10_2013 BAR
BAR_11_2014 BAR_11_2014 BAR
BAR_12_2013 BAR_12_2013 BAR
BAR_13_2014 BAR_13_2014 BAR
BAR_14_2013 BAR_14_2013 BAR
BAR_14_2014 BAR_14_2014 BAR
BAR_15_2014 BAR_15_2014 BAR
BAR_17_2014 BAR_17_2014 BAR
BAR_18_2014 BAR_18_2014 BAR
BAR_20_2014 BAR_20_2014 BAR


plink --file AA261.plink --freq --within AA261.names --out AA261.cluster.pop

PLINK v1.90b6.2 64-bit (12 Jun 2018)
Options in effect:
  --file AA261.plink
  --freq
  --out AA261.cluster.pop
  --within AA261.names

Hostname: eduroam-74-182.nomadic.bris.ac.uk
Working directory: /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RDA
Start time: Tue Mar 12 10:25:05 2019

Random number seed: 1552386305
16384 MB RAM detected; reserving 8192 MB for main workspace.
Scanning .ped file... done.
Performing single-pass .bed write (31381 variants, 261 people).
--file: AA261.cluster.pop-temporary.bed + AA261.cluster.pop-temporary.bim +
AA261.cluster.pop-temporary.fam written.
31381 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.cluster.pop.nosex .
--within: 9 clusters loaded, covering a total of 261 people.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921137.
--freq: Cluster-stratified allele frequencies (founders only) written to
AA261.cluster.pop.frq.strat .

End time: Tue Mar 12 10:25:06 2019

```


Read into R
```
AA261.MAF <- read.table("AA261.cluster.pop.frq.strat", header=T)



```



Run RDA
```
library(vegan)

GenData <- AA261.MAF4[,2:ncol(AA261.MAF4)]
Climate.Data <- read.table("environ.data", header=T)

RDA.AA261.full <- rda(GenData ~ colhist + HostPlant, Climate.Data)

RDA.AA261.full

Call: rda(formula = GenData ~ colhist + HostPlant, data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Constrained    57.9567     0.3599    2
Unconstrained 103.0931     0.6401    6
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2 
41.91 16.05 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5    PC6 
31.491 20.443 17.433 14.361 10.606  8.757 

##calculate the proportion variance explained by the model: 

RsquareAdj(RDA.AA261.full)
$r.squared
[1] 0.3598683

$adj.r.squared
[1] 0.146491


##test significance of full model

anova(RDA.AA261.full)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + HostPlant, data = Climate.Data)
         Df Variance      F Pr(>F)  
Model     2   57.957 1.6865  0.046 *
Residual  6  103.093                
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```


Partial RDA to determine the contribution of colonisation history and Host Plant: 
```
#1. Keep HostPlant constant -> determine effect of Col history

pRDA.AA261.conditionHostPlant <- rda(GenData~colhist + Condition(HostPlant), data=Climate.Data)
pRDA.AA261.conditionHostPlant
Call: rda(formula = GenData ~ colhist + Condition(HostPlant), data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    31.6712     0.1967    1
Constrained    26.2855     0.1632    1
Unconstrained 103.0931     0.6401    6
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
26.285 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5    PC6 
31.491 20.443 17.433 14.361 10.606  8.757 


RsquareAdj(pRDA.AA261.conditionHostPlant)
$r.squared
[1] 0.1632134

$adj.r.squared
[1] 0.06459979

anova(pRDA.AA261.conditionHostPlant)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(HostPlant), data = Climate.Data)
         Df Variance      F Pr(>F)  
Model     1   26.285 1.5298  0.087 .
Residual  6  103.093                
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

H0 accepted: Colonisation history does not explain a significant proportion of genetic variation. 
```


HostPlant
```
##Keep Colonisation history constant. 

pRDA.AA261.conditionColHist <- rda(GenData ~HostPlant + Condition(colhist), data=Climate.Data)

pRDA.AA261.conditionColHist
Call: rda(formula = GenData ~ HostPlant + Condition(colhist), data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    39.5323     0.2455    1
Constrained    18.4243     0.1144    1
Unconstrained 103.0931     0.6401    6
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
18.424 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5    PC6 
31.491 20.443 17.433 14.361 10.606  8.757 


RsquareAdj(pRDA.AA261.conditionColHist)
$r.squared
[1] 0.1144016

$adj.r.squared
[1] 0.008814786


anova(pRDA.AA261.conditionColHist)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ HostPlant + Condition(colhist), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   18.424 1.0723  0.381
Residual  6  103.093 

H0 accepted: Host Plant does not explain a significant proportion of genetic variation. 
```

Most of the genetic variation seems to be determined by geographic distance rather then environmental variables. (Or other environmental variables that we're not taking into account here). But HostPlant and Colonisation History do not explain a large proportion of the genetic variation. 


