# Are A.agestis specialised at range edge? 

Redundancy and pRDA to determine whether populations are specialised. 

Genetic variation can be explained by a number of things. We're interested in determining whether HostPlant prevalence or colonisation histroy has significantly affected the genetic variation. 


We'll used a contrained ordination (Redundancy Analysis - RDA) to test the marginal effect of each of these variables on the distribution of genetic variation. 

Variables of interest: 

1) Geographic distance, 2) HostPlant, or 3) Colonisation history. 


Input files

1:MAF within each population: 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --plink --out AA261.plink

plink --file AA261.plink --recodeA --out AA261.plink

head AA261.names

BAR_10_2013 BAR_10_2013 BAR
BAR_11_2014 BAR_11_2014 BAR
BAR_12_2013 BAR_12_2013 BAR
BAR_13_2014 BAR_13_2014 BAR
BAR_14_2013 BAR_14_2013 BAR
BAR_14_2014 BAR_14_2014 BAR
BAR_15_2014 BAR_15_2014 BAR
BAR_17_2014 BAR_17_2014 BAR
BAR_18_2014 BAR_18_2014 BAR
BAR_20_2014 BAR_20_2014 BAR


plink --file AA261.plink --freq --within AA261.names --out AA261.cluster.pop

PLINK v1.90b6.2 64-bit (12 Jun 2018)
Options in effect:
  --file AA261.plink
  --freq
  --out AA261.cluster.pop
  --within AA261.names

Hostname: eduroam-74-182.nomadic.bris.ac.uk
Working directory: /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RDA
Start time: Tue Mar 12 10:25:05 2019

Random number seed: 1552386305
16384 MB RAM detected; reserving 8192 MB for main workspace.
Scanning .ped file... done.
Performing single-pass .bed write (31381 variants, 261 people).
--file: AA261.cluster.pop-temporary.bed + AA261.cluster.pop-temporary.bim +
AA261.cluster.pop-temporary.fam written.
31381 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.cluster.pop.nosex .
--within: 9 clusters loaded, covering a total of 261 people.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921137.
--freq: Cluster-stratified allele frequencies (founders only) written to
AA261.cluster.pop.frq.strat .

End time: Tue Mar 12 10:25:06 2019

```


Read into R
```
library(ggplot2)
library(reshape2)

AA261.MAF <- read.table("AA261.cluster.pop.frq.strat", header=T)
AA261.MAF2 <- AA261.MAF[,c(3,2,6)]
AA261.MAF3 <- melt(AA261.MAF2, id.vars=c("CLST", "SNP"), variable.name="MAF")
AA261.MAF4 <- dcast(AA261.MAF3, formula=CLST ~SNP)

##This tives the MAF for each locus within each populations. 
AA261.MAF4[,1:2]
  CLST contig_10002:9139
1  BAR            0.1852
2  BCH            0.3421
3  BRO            0.4412
4  FOR            0.6250
5  HOD            0.3000
6  LYD            0.4762
7  MOF            0.3800
8  SWD            0.2763
9  WIS            0.5000



Climate.Data <- read.table("environ.data", header=T)
head(Climate.Data)
  pop      Lat     Long colhist HostPlant
1 BAR 52.62936 -0.41379       2         1
2 BCH 50.99830 -1.13599       1         1
3 BRO 53.64526 -1.21895       2         2
4 FOR 54.16236 -0.39437       1         1
5 HOD 52.97278  0.54254       1         2
6 LYD 51.15359  1.26940       1         1

```



Run RDA
```
library(vegan)

GenData <- AA261.MAF4[,2:ncol(AA261.MAF4)]
dim(GenData)
[1]     9 31381

Climate.Data <- read.table("environ.data", header=T)

RDA.AA261.full <- rda(GenData ~ colhist + HostPlant + Lat + Long, Climate.Data)

RDA.AA261.full

Call: rda(formula = GenData ~ colhist + HostPlant + Lat + Long, data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Constrained   104.9964     0.6519    4
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4 
52.38 19.93 16.92 15.77 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 

##This model explains ~65% of the genetic variation. The unconstrained PCs together explain just less than RDA1. 


##calculate the proportion variance explained by the model: 

RsquareAdj(RDA.AA261.full)
$r.squared
[1] 0.6519498

$adj.r.squared
[1] 0.3038996



##test significance of full model

anova(RDA.AA261.full)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + HostPlant + Lat + Long, data = Climate.Data)
         Df Variance      F Pr(>F)    
Model     4  104.996 1.8731  0.001 ***
Residual  4   56.053                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##Highly significant. 

##plot to see how much each variables explains. 
pdf("RDA.AA261.full.pdf")
plot(RDA.AA261.full)
dev.off()
```


Partial RDA to determine the contribution of colonisation history and Host Plant: 
```
#1. Keep HostPlant constant -> determine effect of Col history

pRDA.AA261.conditionHostPlant <- rda(GenData~colhist + Condition(HostPlant+Long+Lat), data=Climate.Data)
pRDA.AA261.conditionHostPlant
Call: rda(formula = GenData ~ colhist + Condition(HostPlant + Long +
Lat), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    83.6506     0.5194    3
Constrained    21.3458     0.1325    1
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
21.346 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 




RsquareAdj(pRDA.AA261.conditionHostPlant)
$r.squared
[1] 0.1325414

$adj.r.squared
[1] 0.07284609



anova(pRDA.AA261.conditionHostPlant)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(HostPlant + Long + Lat), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   21.346 1.5232  0.158
Residual  4   56.053            

H0 accepted: Colonisation history doesn't explains a significant proportion of genetic variation when keeping geographic variables and hostplant constant
```


HostPlant
```
##Keep Colonisation history and geog constant. 

pRDA.AA261.conditionColHist <- rda(GenData ~HostPlant + Condition(colhist+Lat+Long), data=Climate.Data)

pRDA.AA261.conditionColHist
Call: rda(formula = GenData ~ HostPlant + Condition(colhist + Lat +
Long), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    88.5508     0.5498    3
Constrained    16.4456     0.1021    1
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
16.446 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 


RsquareAdj(pRDA.AA261.conditionColHist)
$r.squared
[1] 0.1021148

$adj.r.squared
[1] 0.02416368



anova(pRDA.AA261.conditionColHist)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ HostPlant + Condition(colhist + Lat + Long), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   16.446 1.1736  0.346
Residual  4   56.053        

H0 accepted: Host Plant does not explain a significant proportion of genetic variation. 
```

Most of the genetic variation seems to be determined by geographic distance rather then environmental variables. (Or other environmental variables that we're not taking into account here). But HostPlant and Colonisation History do not explain a large proportion of the genetic variation. 


