# Are A.agestis specialised at range edge? 

Redundancy and pRDA to determine whether populations are specialised. 

Genetic variation can be explained by a number of things. We're interested in determining whether HostPlant prevalence or colonisation histroy has significantly affected the genetic variation. 


We'll used a contrained ordination (Redundancy Analysis - RDA) to test the marginal effect of each of these variables on the distribution of genetic variation. 

Variables of interest: 

1) Geographic distance, 2) HostPlant, or 3) Colonisation history. 


Input files

1:MAF within each population: 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --plink --out AA261.plink

plink --file AA261.plink --recodeA --out AA261.plink

head AA261.names

BAR_10_2013 BAR_10_2013 BAR
BAR_11_2014 BAR_11_2014 BAR
BAR_12_2013 BAR_12_2013 BAR
BAR_13_2014 BAR_13_2014 BAR
BAR_14_2013 BAR_14_2013 BAR
BAR_14_2014 BAR_14_2014 BAR
BAR_15_2014 BAR_15_2014 BAR
BAR_17_2014 BAR_17_2014 BAR
BAR_18_2014 BAR_18_2014 BAR
BAR_20_2014 BAR_20_2014 BAR


plink --file AA261.plink --freq --within AA261.names --out AA261.cluster.pop

PLINK v1.90b6.2 64-bit (12 Jun 2018)
Options in effect:
  --file AA261.plink
  --freq
  --out AA261.cluster.pop
  --within AA261.names

Hostname: eduroam-74-182.nomadic.bris.ac.uk
Working directory: /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RDA
Start time: Tue Mar 12 10:25:05 2019

Random number seed: 1552386305
16384 MB RAM detected; reserving 8192 MB for main workspace.
Scanning .ped file... done.
Performing single-pass .bed write (31381 variants, 261 people).
--file: AA261.cluster.pop-temporary.bed + AA261.cluster.pop-temporary.bim +
AA261.cluster.pop-temporary.fam written.
31381 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.cluster.pop.nosex .
--within: 9 clusters loaded, covering a total of 261 people.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921137.
--freq: Cluster-stratified allele frequencies (founders only) written to
AA261.cluster.pop.frq.strat .

End time: Tue Mar 12 10:25:06 2019

```


Read into R
```
library(ggplot2)
library(reshape2)

AA261.MAF <- read.table("AA261.cluster.pop.frq.strat", header=T)
AA261.MAF2 <- AA261.MAF[,c(3,2,6)]
AA261.MAF3 <- melt(AA261.MAF2, id.vars=c("CLST", "SNP"), variable.name="MAF")
AA261.MAF4 <- dcast(AA261.MAF3, formula=CLST ~SNP)

##This gives the MAF for each locus within each populations. 
AA261.MAF4[,1:2]
  CLST contig_10002:9139
1  BAR            0.1852
2  BCH            0.3421
3  BRO            0.4412
4  FOR            0.6250
5  HOD            0.3000
6  LYD            0.4762
7  MOF            0.3800
8  SWD            0.2763
9  WIS            0.5000



Climate.Data <- read.table("environ.data", header=T)
head(Climate.Data)
  pop      Lat     Long colhist HostPlant
1 BAR 52.62936 -0.41379       2         1
2 BCH 50.99830 -1.13599       1         1
3 BRO 53.64526 -1.21895       2         2
4 FOR 54.16236 -0.39437       2         1
5 HOD 52.97278  0.54254       1         2
6 LYD 51.15359  1.26940       1         1

```



Run RDA
```
library(vegan)

GenData <- AA261.MAF4[,2:ncol(AA261.MAF4)]
dim(GenData)
[1]     9 31381

Climate.Data <- read.table("environ.data", header=T)

RDA.AA261.full <- rda(GenData ~ colhist + hostplant2 + Lat + Long, Climate.Data)

RDA.AA261.full

Call: rda(formula = GenData ~ colhist + HostPlant + Lat + Long, data =
Climate.Data)

               Inertia Proportion Rank
Total         161.05      1.0000     
Constrained   105.496     0.6519    4
Unconstrained  55.554     0.345     4
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4 
52.96 18.97 16.84 16.73 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.419 15.185 11.061  8.890 

##This model explains ~65% of the genetic variation. The unconstrained PCs together explain just less than RDA1. 


##calculate the proportion variance explained by the model: 

RsquareAdj(RDA.AA261.full)
$r.squared
[1] 0.6550494

$adj.r.squared
[1] 0.3100988



##test significance of full model

anova(RDA.AA261.full)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + hostplant2 + Lat + Long, data = Climate.Data)
         Df Variance     F Pr(>F)    
Model     4  105.496 1.899  0.001 ***
Residual  4   55.554                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##Highly significant. 

##plot to see how much each variables explains. 
pdf("RDA.AA261.full.pdf")
plot(RDA.AA261.full)
dev.off()
```

### Choose best model

To choose the best model I'm removing non-significant variables one at a time. 

```
AA261.bestModel <- ordistep(RDA.AA261.full)

Start: GenData ~ colhist + hostplant2 + Lat + Long 

             Df    AIC      F Pr(>F)
- Long        1 45.518 1.2354  0.235
- Lat         1 45.722 1.3552  0.175
- hostplant2  1 45.877 1.4481  0.105
- colhist     1 46.081 1.5729  0.100

Step: GenData ~ colhist + hostplant2 + Lat 

             Df    AIC      F Pr(>F)
- Lat         1 45.574 1.2826  0.185
- hostplant2  1 45.719 1.3850  0.105
- colhist     1 45.948 1.5497  0.100

Step: GenData ~ colhist + hostplant2 

             Df    AIC      F Pr(>F)   
- hostplant2  1 45.457 1.3967  0.135   
- colhist     1 46.705 2.4965  0.005 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist 

          Df    AIC      F Pr(>F)   
- colhist  1 46.675 3.0092   0.01 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

Only Colhist are significant. 

### Is Colhist significant when keeping Lat constant? 

```
pRDA.LatConstant_ColhistVar <- rda(GenData ~ colhist + Condition(Lat), data=Climate.Data)


Call: rda(formula = GenData ~ colhist + Condition(Lat), data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    46.5561     0.2891    1
Constrained    21.6404     0.1344    1
Unconstrained  92.8533     0.5766    6
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1 
21.64 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5    PC6 
22.246 19.948 17.399 13.675 10.901  8.684 

RsquareAdj(pRDA.LatConstant_ColhistVar)
$r.squared
[1] 0.1343707

$adj.r.squared
[1] 0.04374738

anova(pRDA.LatConstant_ColhistVar)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(Lat), data = Climate.Data)
         Df Variance      F Pr(>F)  
Model     1   21.640 1.3984  0.072 .
Residual  6   92.853                
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```


### Conclusion

My conclusion is that ColHist and Lat both explain most of the explainable variance in the data, but because they're so correlated one is insignificant when the other is kept constant. 
HostPlant does not significantly explain genetic variation. This may be because only a few large effect loci are associated with adaptation to this new host plant. 
We also haven't taken female preference into account - our conclusions are based solely on host plant prevalence. 



#### partial RDA

For completeness I'm testing how HostPlant and Col History affect genetic variance conditioned on all other variables

Partial RDA to determine the contribution of colonisation history and Host Plant: 
```
#1. Keep HostPlant constant -> determine effect of Col history

pRDA.AA261.conditionHostPlant <- rda(GenData~colhist + Condition(hostplant2+Long+Lat), data=Climate.Data)
pRDA.AA261.conditionHostPlant
Call: rda(formula = GenData ~ colhist + Condition(hostplant2 + Long +
Lat), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    83.6506     0.5194    3
Constrained    21.8449     0.1356    1
Unconstrained  55.5542     0.3450    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
21.845 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.419 15.185 11.061  8.890 



RsquareAdj(pRDA.AA261.conditionHostPlant)
$r.squared
[1] 0.1356409

$adj.r.squared
[1] 0.07904525



anova(pRDA.AA261.conditionHostPlant)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(hostplant2 + Long + Lat), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   21.845 1.5729  0.158
Residual  4   55.554       

H0 accepted: Colonisation history doesn't explains a significant proportion of genetic variation when keeping geographic variables and hostplant constant
```


HostPlant
```
##Keep Colonisation history and geog constant. 

pRDA.AA261.conditionColHist <- rda(GenData ~hostplant2 + Condition(colhist+Lat+Long), data=Climate.Data)

pRDA.AA261.conditionColHist
Call: rda(formula = GenData ~ hostplant2 + Condition(colhist + Lat +
Long), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    85.3831     0.5302    3
Constrained    20.1124     0.1249    1
Unconstrained  55.5542     0.3450    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
20.112 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.419 15.185 11.061  8.890 


RsquareAdj(pRDA.AA261.conditionColHist)
$r.squared
[1] 0.1248833

$adj.r.squared
[1] 0.06183298


anova(pRDA.AA261.conditionColHist)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ hostplant2 + Condition(colhist + Lat + Long), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   20.112 1.4481  0.192
Residual  4   55.554              

H0 accepted: Host Plant does not explain a significant proportion of genetic variation. 
```

Most of the genetic variation seems to be determined by geographic distance rather then environmental variables. (Or other environmental variables that we're not taking into account here). But HostPlant and Colonisation History do not explain a large proportion of the genetic variation. 



## This holds true when I include BioClim and PCNM data into the list of variables. 


```
Climate.Data <- read.table("environ.data", header=T)

head(Climate.Data)
  X.CLST bio1 bio2 bio3 bio8 bio9 bio13 hostplant2 colhist       PCNM1
1    BAR   98   74   34  162   42    55          1       2 -0.02471747
2    BCH   99   76   36   48  146    93          1       1  0.08717326
3    BRO   95   73   37  132   59    61          2       2 -0.06095130
4    FOR   93   69   37   50  131    83          1       1 -0.05390263
5    HOD   96   69   33   44   61    61          2       1 -0.06129235
6    LYD   99   65   33  110   59    75          1       1  0.05793605
        PCNM2        PCNM3        PCNM4        PCNM5     Long      Lat
1 -0.09204650 -0.113347059 -0.004017264 -0.001691204 -0.41379 52.62936
2  0.10255800 -0.018926075  0.002623351  0.001170633 -1.13599 50.99830
3  0.01469557  0.008170790 -0.184648668 -0.205023077 -1.21895 53.64526
4  0.10016185  0.163042484 -0.013892345 -0.003037958 -0.39437 54.16236
5  0.01194874  0.002217105  0.153739223 -0.061819794  0.54254 52.97278
6 -0.12723788  0.186044440  0.006617196  0.002933059  1.26940 51.15359

```


The full model explains all of the variance!
```
RDA.AA261.full <- rda(GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat, Climate.Data)


RDA.AA261.full
Call: rda(formula = GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 +
hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +
Lat, data = Climate.Data)

              Inertia Proportion Rank
Total           161.1        1.0     
Constrained     161.1        1.0    8
Unconstrained     0.0        0.0    0
Inertia is variance 
Some constraints were aliased because they were collinear (redundant)

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4  RDA5  RDA6  RDA7  RDA8 
52.98 21.33 19.99 17.66 16.80 13.06 10.58  8.65 

```




Only PCNM variables are significant in the final models. 
```
AA261.bestModel <- ordistep(RDA.AA261.full)

Start: GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio1        0 -Inf         
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 +      PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 +      PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 +      PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 +      PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +      Lat 

          Df    AIC F Pr(>F)
- PCNM5    1 36.383         
- PCNM4    1 37.822         
- PCNM2    1 38.121         
- PCNM3    1 38.230         
- Long     1 38.338         
- colhist  1 38.455         
- Lat      1 38.934         
- PCNM1    1 39.438         

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + Long + Lat 

          Df    AIC      F Pr(>F)
- PCNM3    1 41.032 1.0934  0.395
- PCNM2    1 40.969 1.0788  0.380
- Long     1 41.069 1.1021  0.370
- PCNM1    1 41.711 1.2574  0.280
- PCNM4    1 41.641 1.2399  0.270
- Lat      1 41.598 1.2293  0.260
- colhist  1 43.408 1.7259  0.110

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM4 + Long + Lat 

          Df    AIC      F Pr(>F)  
- PCNM2    1 43.132 1.1541  0.265  
- Lat      1 43.786 1.3917  0.120  
- colhist  1 44.386 1.6256  0.115  
- PCNM1    1 44.671 1.7424  0.080 .
- Long     1 44.334 1.6048  0.075 .
- PCNM4    1 44.523 1.6812  0.065 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + PCNM1 + PCNM4 + Long + Lat 

          Df    AIC      F Pr(>F)  
- Long     1 44.866 1.5423  0.115  
- colhist  1 44.908 1.5638  0.100 .
- Lat      1 44.872 1.5456  0.095 .
- PCNM1    1 45.105 1.6647  0.075 .
- PCNM4    1 45.069 1.6460  0.060 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + PCNM1 + PCNM4 + Lat 

          Df    AIC      F Pr(>F)
- colhist  1 45.577 1.4064  0.165
- PCNM1    1 45.601 1.4208  0.165
- Lat      1 45.494 1.3565  0.150
- PCNM4    1 45.609 1.4252  0.150

Step: GenData ~ PCNM1 + PCNM4 + Lat 

        Df    AIC      F Pr(>F)  
- Lat    1 45.626 1.2783  0.235  
- PCNM1  1 45.816 1.4119  0.145  
- PCNM4  1 45.901 1.4728  0.080 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ PCNM1 + PCNM4 

        Df    AIC      F Pr(>F)   
- PCNM4  1 45.658 1.5196  0.055 . 
- PCNM1  1 47.164 2.8892  0.005 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```


## Northern populations only

Given that geographic structure can override the signal from environmental variables, I ran the RDA for the northern populations only. 



Read into R
```
##Done before
library(ggplot2)
library(reshape2)

#AA261.MAF <- read.table("AA261.cluster.pop.frq.strat", header=T)
#AA261.MAF2 <- AA261.MAF[,c(3,2,6)]
#AA261.MAF3 <- melt(AA261.MAF2, id.vars=c("CLST", "SNP"), variable.name="MAF")
#AA261.MAF4 <- dcast(AA261.MAF3, formula=CLST ~SNP)

##This gives the MAF for each locus within each populations. 
#AA261.MAF4[,1:2]
  CLST contig_10002:9139
1  BAR            0.1852
2  BCH            0.3421
3  BRO            0.4412
4  FOR            0.6250
5  HOD            0.3000
6  LYD            0.4762
7  MOF            0.3800
8  SWD            0.2763
9  WIS            0.5000



#Climate.Data <- read.table("environ.data", header=T)
head(Climate.Data)
  pop      Lat     Long colhist HostPlant
1 BAR 52.62936 -0.41379       2         1
2 BCH 50.99830 -1.13599       1         1
3 BRO 53.64526 -1.21895       2         2
4 FOR 54.16236 -0.39437       1         1
5 HOD 52.97278  0.54254       1         2
6 LYD 51.15359  1.26940       1         1


##Subset both datasets: 

AA261.MAF4.North <- AA261.MAF4[c(1,3,4,5,7,9),]
Climate.Data.North <- Climate.Data[c(1,3,4,5,7,9),]
```



Run RDA
```
library(vegan)

GenData <- AA261.MAF4.North[,2:ncol(AA261.MAF4.North)]
dim(GenData)
[1]     6 31381


RDA.AA261.full.North <- rda(GenData ~ colhist + hostplant2 + Lat + Long, Climate.Data.North)

RDA.AA261.full.North

Call: rda(formula = GenData ~ colhist + hostplant2 + Lat + Long, data =
Climate.Data.North)

               Inertia Proportion Rank
Total         137.7509     1.0000     
Constrained   122.4378     0.8888    4
Unconstrained  15.3131     0.1112    1
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4 
39.56 32.41 28.27 22.19 

Eigenvalues for unconstrained axes:
   PC1 
15.313 

##This model explains ~89% of the genetic variation. The unconstrained PC explain less than half of RDA1. 


##calculate the proportion variance explained by the model: 

RsquareAdj(RDA.AA261.full.North)
$r.squared
[1] 0.888835

$adj.r.squared
[1] 0.444175


##test significance of full model

anova(RDA.AA261.full.North)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 719

Model: rda(formula = GenData ~ colhist + hostplant2 + Lat + Long, data = Climate.Data.North)
         Df Variance      F   Pr(>F)   
Model     4  122.438 1.9989 0.006944 **
Residual  1   15.313                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


#Highly Significant. 

##plot to see how much each variables explains. 
pdf("RDA.AA261.full.North.pdf")
plot(RDA.AA261.full.North)
dev.off()
```

### Choose best model

To choose the best model I'm removing non-significant variables one at a time. 

```
AA261.North.bestModel <- ordistep(RDA.AA261.full.North)

Start: GenData ~ colhist + hostplant2 + Lat + Long 
Set of permutations < 'minperm'. Generating entire set.

             Df    AIC      F Pr(>F)  
- hostplant2  1 29.097 1.6372  0.075 .
- Lat         1 29.808 1.9692  0.040 *
- Long        1 29.302 1.7290  0.020 *
- colhist     1 29.587 1.8617  0.015 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

Lat, Long, colhist, and HostPlant are significant. 


### Is Colhist significant when keeping Lat constant? 

```
pRDA.LatConstant_ColhistVar.North <- rda(GenData ~ colhist + Condition(Lat), data=Climate.Data.North)


pRDA.LatConstant_ColhistVar.North
Call: rda(formula = GenData ~ colhist + Condition(Lat), data =
Climate.Data.North)

               Inertia Proportion Rank
Total         137.7509     1.0000     
Conditional    34.3763     0.2496    1
Constrained    33.7776     0.2452    1
Unconstrained  69.5971     0.5052    3
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1 
33.78 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3 
31.91 23.75 13.94 

RsquareAdj(pRDA.LatConstant_ColhistVar.North)
$r.squared
[1] 0.2452076

$adj.r.squared
[1] 0.09599343

anova(pRDA.LatConstant_ColhistVar.North)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 719

Model: rda(formula = GenData ~ colhist + Condition(Lat), data = Climate.Data.North)
         Df Variance     F Pr(>F)
Model     1   33.778 1.456 0.2306
Residual  3   69.597          

```


### Conclusion

My conclusion is that lat & long, and the ColHist and HostPlant all explain the variance in the data.

HostPlant explains the least of the genetic variance. This may be because only a few large effect loci are associated with adaptation to this new host plant. 
We also haven't taken female preference into account - our conclusions are based solely on host plant prevalence. 



### And with all the BioClim variables


```
Climate.Data <- read.table("environ.data", header=T)

head(Climate.Data)
  X.CLST bio1 bio2 bio3 bio8 bio9 bio13 hostplant2 colhist       PCNM1
1    BAR   98   74   34  162   42    55          1       2 -0.02471747
2    BCH   99   76   36   48  146    93          1       1  0.08717326
3    BRO   95   73   37  132   59    61          2       2 -0.06095130
4    FOR   93   69   37   50  131    83          1       1 -0.05390263
5    HOD   96   69   33   44   61    61          2       1 -0.06129235
6    LYD   99   65   33  110   59    75          1       1  0.05793605
        PCNM2        PCNM3        PCNM4        PCNM5     Long      Lat
1 -0.09204650 -0.113347059 -0.004017264 -0.001691204 -0.41379 52.62936
2  0.10255800 -0.018926075  0.002623351  0.001170633 -1.13599 50.99830
3  0.01469557  0.008170790 -0.184648668 -0.205023077 -1.21895 53.64526
4  0.10016185  0.163042484 -0.013892345 -0.003037958 -0.39437 54.16236
5  0.01194874  0.002217105  0.153739223 -0.061819794  0.54254 52.97278
6 -0.12723788  0.186044440  0.006617196  0.002933059  1.26940 51.15359


Climate.Data.North <- Climate.Data[c(1,3,4,5,7,9),]

```


```
RDA.AA261.full.North <- rda(GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat, Climate.Data.North)


RDA.AA261.full.North
Call: rda(formula = GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 +
hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +
Lat, data = Climate.Data.North)

              Inertia Proportion Rank
Total           137.8        1.0     
Constrained     137.8        1.0    5
Unconstrained     0.0        0.0    0
Inertia is variance 
Some constraints were aliased because they were collinear (redundant)

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4  RDA5 
39.72 32.75 29.14 22.20 13.93 
```


Latitude is the only highly significant variable in the best model, while colonisation history is also significant. All environmental variables drop out. 
```
AA261.North.bestModel <- ordistep(RDA.AA261.full.North)

Start: GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- bio1        0 -Inf         
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 +      PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 +      PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 +      PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 +      PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

             Df  AIC F Pr(>F)
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +      Lat 
Set of permutations < 'minperm'. Generating entire set.

          Df  AIC F Pr(>F)
- colhist  0 -Inf         
- PCNM1    0 -Inf         
- PCNM2    0 -Inf         
- PCNM3    0 -Inf         
- PCNM4    0 -Inf         
- PCNM5    0 -Inf         
- Long     0 -Inf         
- Lat      0 -Inf         

Step: GenData ~ PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

        Df  AIC F Pr(>F)
- PCNM1  0 -Inf         
- PCNM2  0 -Inf         
- PCNM3  0 -Inf         
- PCNM4  0 -Inf         
- PCNM5  0 -Inf         
- Long   0 -Inf         
- Lat    0 -Inf         

Step: GenData ~ PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

        Df  AIC F Pr(>F)
- PCNM2  0 -Inf         
- PCNM3  0 -Inf         
- PCNM4  0 -Inf         
- PCNM5  0 -Inf         
- Long   0 -Inf         
- Lat    0 -Inf         

Step: GenData ~ PCNM3 + PCNM4 + PCNM5 + Long + Lat 
Set of permutations < 'minperm'. Generating entire set.

        Df    AIC F Pr(>F)
- Lat    1 27.386         
- PCNM3  1 27.388         
- Long   1 27.410         
- PCNM4  1 27.416         
- PCNM5  1 27.422         

Step: GenData ~ PCNM3 + PCNM4 + PCNM5 + Long 
Set of permutations < 'minperm'. Generating entire set.

        Df    AIC      F Pr(>F)
- PCNM5  1 29.747 1.0687  0.400
- Long   1 29.917 1.1280  0.350
- PCNM4  1 29.924 1.1304  0.345
- PCNM3  1 30.653 1.4059  0.200

Step: GenData ~ PCNM3 + PCNM4 + Long 
Set of permutations < 'minperm'. Generating entire set.

        Df    AIC      F Pr(>F)  
- PCNM4  1 30.387 1.1052  0.380  
- Long   1 30.368 1.0954  0.375  
- PCNM3  1 30.893 1.3790  0.095 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ PCNM3 + Long 
Set of permutations < 'minperm'. Generating entire set.

        Df    AIC      F Pr(>F)  
- PCNM3  1 30.594 1.3339   0.07 .
- Long   1 30.920 1.5764   0.02 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```

# North conditional models

Col hist when hostplant and geography are kept constant
```

pRDA.AA261.conditionHostPlant.North <- rda(GenData~colhist + Condition(hostplant2+Long+Lat), data=Climate.Data.North)
Call: rda(formula = GenData ~ colhist + Condition(hostplant2 + Long +
Lat), data = Climate.Data.North)

               Inertia Proportion Rank
Total         137.7509     1.0000     
Conditional    93.9288     0.6819    3
Constrained    28.5090     0.2070    1
Unconstrained  15.3131     0.1112    1
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
28.509 

Eigenvalues for unconstrained axes:
   PC1 
15.313 


RsquareAdj(pRDA.AA261.conditionHostPlant.North)
$r.squared
[1] 0.2069603

$adj.r.squared
[1] 0.2394882


anova(pRDA.AA261.conditionHostPlant.North)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 719

Model: rda(formula = GenData ~ colhist + Condition(hostplant2 + Long + Lat), data = Climate.Data.North)
         Df Variance      F Pr(>F)
Model     1   28.509 1.8617 0.3319
Residual  1   15.313   
```

HostPlant when ColHist and geography are kept constant
```

pRDA.AA261.conditionColHist.North <- rda(GenData~hostplant2 + Condition(colhist+Long+Lat), data=Climate.Data.North)
Call: rda(formula = GenData ~ hostplant2 + Condition(colhist + Long +
Lat), data = Climate.Data.North)

               Inertia Proportion Rank
Total         137.7509     1.0000     
Conditional    97.3666     0.7068    3
Constrained    25.0712     0.1820    1
Unconstrained  15.3131     0.1112    1
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
25.071 

Eigenvalues for unconstrained axes:
   PC1 
15.313 




RsquareAdj(pRDA.AA261.conditionColHist.North)
$r.squared
[1] 0.1820042

$adj.r.squared
[1] 0.1770981


anova(pRDA.AA261.conditionColHist.North)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 719

Model: rda(formula = GenData ~ hostplant2 + Condition(colhist + Long + Lat), data = Climate.Data.North)
         Df Variance      F Pr(>F)
Model     1   25.071 1.6372 0.3653
Residual  1   15.313              

```


# Full Model: HostPlant + Geog

```
pRDA.AA261.HostPlantFull<- rda(GenData~Lat+Long+hostplant2, data=Climate.Data)

pRDA.AA261.HostPlantFull
Call: rda(formula = GenData ~ Lat + Long + hostplant2, data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Constrained    83.6506     0.5194    3
Unconstrained  77.3991     0.4806    5
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3 
48.44 18.43 16.78 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5 
22.135 20.406 15.184 10.785  8.890 

RsquareAdj(pRDA.AA261.HostPlantFull)
$r.squared
[1] 0.5194085

$adj.r.squared
[1] 0.2310535

anova(pRDA.AA261.HostPlantFull)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ Lat + Long + hostplant2, data = Climate.Data)
         Df Variance      F Pr(>F)   
Model     3   83.651 1.8013  0.003 **
Residual  5   77.399                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

Condition: HP only
```
pRDA.AA261.HostPlantFull.hponly<- rda(GenData~hostplant2 + Condition(Long+Lat), data=Climate.Data)
> pRDA.AA261.HostPlantFull.hponly
Call: rda(formula = GenData ~ hostplant2 + Condition(Long + Lat), data
= Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    64.3326     0.3995    2
Constrained    19.3180     0.1200    1
Unconstrained  77.3991     0.4806    5
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
19.318 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5 
22.135 20.406 15.184 10.785  8.890 


RsquareAdj(pRDA.AA261.HostPlantFull.hponly)
$r.squared
[1] 0.1199507

$adj.r.squared
[1] 0.03177655

anova(pRDA.AA261.HostPlantFull.hponly)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ hostplant2 + Condition(Long + Lat), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   19.318 1.2479  0.266
Residual  5   77.399            

```

Geog only
```
pRDA.AA261.HostPlantFull.GeogOnly
Call: rda(formula = GenData ~ Lat + Long + Condition(hostplant2), data
= Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    31.6712     0.1967    1
Constrained    51.9794     0.3228    2
Unconstrained  77.3991     0.4806    5
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2 
35.17 16.80 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5 
22.135 20.406 15.184 10.785  8.890 

> RsquareAdj(pRDA.AA261.HostPlantFull.GeogOnly)
$r.squared
[1] 0.3227536

$adj.r.squared
[1] 0.1491623

> anova(pRDA.AA261.HostPlantFull.GeogOnly)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ Lat + Long + Condition(hostplant2), data = Climate.Data)
         Df Variance      F Pr(>F)  
Model     2   51.979 1.6789  0.026 *
Residual  5   77.399                
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1



```

