# Are A.agestis specialised at range edge? 

Redundancy and pRDA to determine whether populations are specialised. 

Genetic variation can be explained by a number of things. We're interested in determining whether HostPlant prevalence or colonisation histroy has significantly affected the genetic variation. 


We'll used a contrained ordination (Redundancy Analysis - RDA) to test the marginal effect of each of these variables on the distribution of genetic variation. 

Variables of interest: 

1) Geographic distance, 2) HostPlant, or 3) Colonisation history. 


Input files

1:MAF within each population: 
```
vcftools --vcf AA261.0.5miss.9popsMerged.vcf --plink --out AA261.plink

plink --file AA261.plink --recodeA --out AA261.plink

head AA261.names

BAR_10_2013 BAR_10_2013 BAR
BAR_11_2014 BAR_11_2014 BAR
BAR_12_2013 BAR_12_2013 BAR
BAR_13_2014 BAR_13_2014 BAR
BAR_14_2013 BAR_14_2013 BAR
BAR_14_2014 BAR_14_2014 BAR
BAR_15_2014 BAR_15_2014 BAR
BAR_17_2014 BAR_17_2014 BAR
BAR_18_2014 BAR_18_2014 BAR
BAR_20_2014 BAR_20_2014 BAR


plink --file AA261.plink --freq --within AA261.names --out AA261.cluster.pop

PLINK v1.90b6.2 64-bit (12 Jun 2018)
Options in effect:
  --file AA261.plink
  --freq
  --out AA261.cluster.pop
  --within AA261.names

Hostname: eduroam-74-182.nomadic.bris.ac.uk
Working directory: /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/RDA
Start time: Tue Mar 12 10:25:05 2019

Random number seed: 1552386305
16384 MB RAM detected; reserving 8192 MB for main workspace.
Scanning .ped file... done.
Performing single-pass .bed write (31381 variants, 261 people).
--file: AA261.cluster.pop-temporary.bed + AA261.cluster.pop-temporary.bim +
AA261.cluster.pop-temporary.fam written.
31381 variants loaded from .bim file.
261 people (0 males, 0 females, 261 ambiguous) loaded from .fam.
Ambiguous sex IDs written to AA261.cluster.pop.nosex .
--within: 9 clusters loaded, covering a total of 261 people.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 261 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.921137.
--freq: Cluster-stratified allele frequencies (founders only) written to
AA261.cluster.pop.frq.strat .

End time: Tue Mar 12 10:25:06 2019

```


Read into R
```
library(ggplot2)
library(reshape2)

AA261.MAF <- read.table("AA261.cluster.pop.frq.strat", header=T)
AA261.MAF2 <- AA261.MAF[,c(3,2,6)]
AA261.MAF3 <- melt(AA261.MAF2, id.vars=c("CLST", "SNP"), variable.name="MAF")
AA261.MAF4 <- dcast(AA261.MAF3, formula=CLST ~SNP)

##This tives the MAF for each locus within each populations. 
AA261.MAF4[,1:2]
  CLST contig_10002:9139
1  BAR            0.1852
2  BCH            0.3421
3  BRO            0.4412
4  FOR            0.6250
5  HOD            0.3000
6  LYD            0.4762
7  MOF            0.3800
8  SWD            0.2763
9  WIS            0.5000



Climate.Data <- read.table("environ.data", header=T)
head(Climate.Data)
  pop      Lat     Long colhist HostPlant
1 BAR 52.62936 -0.41379       2         1
2 BCH 50.99830 -1.13599       1         1
3 BRO 53.64526 -1.21895       2         2
4 FOR 54.16236 -0.39437       1         1
5 HOD 52.97278  0.54254       1         2
6 LYD 51.15359  1.26940       1         1

```



Run RDA
```
library(vegan)

GenData <- AA261.MAF4[,2:ncol(AA261.MAF4)]
dim(GenData)
[1]     9 31381

Climate.Data <- read.table("environ.data", header=T)

RDA.AA261.full <- rda(GenData ~ colhist + HostPlant + Lat + Long, Climate.Data)

RDA.AA261.full

Call: rda(formula = GenData ~ colhist + HostPlant + Lat + Long, data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Constrained   104.9964     0.6519    4
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4 
52.38 19.93 16.92 15.77 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 

##This model explains ~65% of the genetic variation. The unconstrained PCs together explain just less than RDA1. 


##calculate the proportion variance explained by the model: 

RsquareAdj(RDA.AA261.full)
$r.squared
[1] 0.6519498

$adj.r.squared
[1] 0.3038996



##test significance of full model

anova(RDA.AA261.full)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + HostPlant + Lat + Long, data = Climate.Data)
         Df Variance      F Pr(>F)    
Model     4  104.996 1.8731  0.001 ***
Residual  4   56.053                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##Highly significant. 

##plot to see how much each variables explains. 
pdf("RDA.AA261.full.pdf")
plot(RDA.AA261.full)
dev.off()
```

### Choose best model

To choose the best model I'm removing non-significant variables one at a time. 

```
AA261.bestModel <- ordistep(RDA.AA261.full)

Start: GenData ~ colhist + HostPlant + Lat + Long 

            Df    AIC      F Pr(>F)   
- HostPlant  1 45.492 1.1736  0.295   
- Long       1 45.570 1.2183  0.210   
- colhist    1 46.081 1.5232  0.100 . 
- Lat        1 46.982 2.1048  0.010 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + Lat + Long 

          Df    AIC      F Pr(>F)  
- Long     1 45.429 1.2007  0.195  
- colhist  1 46.086 1.6702  0.090 .
- Lat      1 46.742 2.1747  0.025 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + Lat 

          Df    AIC      F Pr(>F)  
- colhist  1 45.605 1.6406  0.080 .
- Lat      1 46.140 2.1094  0.015 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

Only Lat and colhist are significant. 

### Is Colhist significant when keeping Lat constant? 

```
pRDA.LatConstant_ColhistVar <- rda(GenData ~ colhist + Condition(Lat), data=Climate.Data)


Call: rda(formula = GenData ~ colhist + Condition(Lat), data =
Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    46.5561     0.2891    1
Constrained    24.5848     0.1527    1
Unconstrained  89.9089     0.5583    6
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
24.585 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4    PC5    PC6 
20.902 18.678 17.409 13.490 10.754  8.677 

RsquareAdj(pRDA.LatConstant_ColhistVar)
$r.squared
[1] 0.1526536

$adj.r.squared
[1] 0.06812455

anova(pRDA.LatConstant_ColhistVar)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(Lat), data = Climate.Data)
         Df Variance      F Pr(>F)   
Model     1   24.585 1.6406   0.01 **
Residual  6   89.909                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```


### Conclusion

My conclusion is that Lat explains most of the variance in the date, while colonisation history explains a small but significant proportion. 
HostPlant does not explain genetic variation. This may be because only a few large effect loci are associated with adaptation to this new host plant. 
We also haven't taken female preference into account - our conclusions are based solely on host plant prevalence. 



#### partial RDA

For completeness I'm testing how HostPlant and Col History affect genetic variance conditioned on all other variables

Partial RDA to determine the contribution of colonisation history and Host Plant: 
```
#1. Keep HostPlant constant -> determine effect of Col history

pRDA.AA261.conditionHostPlant <- rda(GenData~colhist + Condition(HostPlant+Long+Lat), data=Climate.Data)
pRDA.AA261.conditionHostPlant
Call: rda(formula = GenData ~ colhist + Condition(HostPlant + Long +
Lat), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    83.6506     0.5194    3
Constrained    21.3458     0.1325    1
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
21.346 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 




RsquareAdj(pRDA.AA261.conditionHostPlant)
$r.squared
[1] 0.1325414

$adj.r.squared
[1] 0.07284609



anova(pRDA.AA261.conditionHostPlant)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ colhist + Condition(HostPlant + Long + Lat), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   21.346 1.5232  0.158
Residual  4   56.053            

H0 accepted: Colonisation history doesn't explains a significant proportion of genetic variation when keeping geographic variables and hostplant constant
```


HostPlant
```
##Keep Colonisation history and geog constant. 

pRDA.AA261.conditionColHist <- rda(GenData ~HostPlant + Condition(colhist+Lat+Long), data=Climate.Data)

pRDA.AA261.conditionColHist
Call: rda(formula = GenData ~ HostPlant + Condition(colhist + Lat +
Long), data = Climate.Data)

               Inertia Proportion Rank
Total         161.0498     1.0000     
Conditional    88.5508     0.5498    3
Constrained    16.4456     0.1021    1
Unconstrained  56.0534     0.3481    4
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1 
16.446 

Eigenvalues for unconstrained axes:
   PC1    PC2    PC3    PC4 
20.443 15.897 10.797  8.916 


RsquareAdj(pRDA.AA261.conditionColHist)
$r.squared
[1] 0.1021148

$adj.r.squared
[1] 0.02416368



anova(pRDA.AA261.conditionColHist)
Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ HostPlant + Condition(colhist + Lat + Long), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     1   16.446 1.1736  0.346
Residual  4   56.053        

H0 accepted: Host Plant does not explain a significant proportion of genetic variation. 
```

Most of the genetic variation seems to be determined by geographic distance rather then environmental variables. (Or other environmental variables that we're not taking into account here). But HostPlant and Colonisation History do not explain a large proportion of the genetic variation. 



## This holds true when I include BioClim and PCNM data into the list of variables. 


```
Climate.Data <- read.table("environ.data", header=T)

head(Climate.Data)
  X.CLST bio1 bio2 bio3 bio8 bio9 bio13 hostplant2 colhist       PCNM1
1    BAR   98   74   34  162   42    55          1       2 -0.02471747
2    BCH   99   76   36   48  146    93          1       1  0.08717326
3    BRO   95   73   37  132   59    61          2       2 -0.06095130
4    FOR   93   69   37   50  131    83          1       1 -0.05390263
5    HOD   96   69   33   44   61    61          2       1 -0.06129235
6    LYD   99   65   33  110   59    75          1       1  0.05793605
        PCNM2        PCNM3        PCNM4        PCNM5     Long      Lat
1 -0.09204650 -0.113347059 -0.004017264 -0.001691204 -0.41379 52.62936
2  0.10255800 -0.018926075  0.002623351  0.001170633 -1.13599 50.99830
3  0.01469557  0.008170790 -0.184648668 -0.205023077 -1.21895 53.64526
4  0.10016185  0.163042484 -0.013892345 -0.003037958 -0.39437 54.16236
5  0.01194874  0.002217105  0.153739223 -0.061819794  0.54254 52.97278
6 -0.12723788  0.186044440  0.006617196  0.002933059  1.26940 51.15359

```


The full model explains all of the variance!
```
RDA.AA261.full <- rda(GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat, Climate.Data)


RDA.AA261.full
Call: rda(formula = GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 +
hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +
Lat, data = Climate.Data)

              Inertia Proportion Rank
Total           161.1        1.0     
Constrained     161.1        1.0    8
Unconstrained     0.0        0.0    0
Inertia is variance 
Some constraints were aliased because they were collinear (redundant)

Eigenvalues for constrained axes:
 RDA1  RDA2  RDA3  RDA4  RDA5  RDA6  RDA7  RDA8 
52.98 21.33 19.99 17.66 16.80 13.06 10.58  8.65 

```





Latitude is the only highly significant variable in the best model, while colonisation history is also significant. All environmental variables drop out. 
```
AA261.bestModel <- ordistep(RDA.AA261.full)

Start: GenData ~ bio1 + bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio1        0 -Inf         
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio2 + bio8 + bio9 + bio13 + hostplant2 + colhist +      PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio2        0 -Inf         
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio8 + bio9 + bio13 + hostplant2 + colhist + PCNM1 +      PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio8        0 -Inf         
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio9 + bio13 + hostplant2 + colhist + PCNM1 + PCNM2 +      PCNM3 + PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio9        0 -Inf         
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ bio13 + hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 +      PCNM4 + PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- bio13       0 -Inf         
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ hostplant2 + colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 +      PCNM5 + Long + Lat 

             Df  AIC F Pr(>F)
- hostplant2  0 -Inf         
- colhist     0 -Inf         
- PCNM1       0 -Inf         
- PCNM2       0 -Inf         
- PCNM3       0 -Inf         
- PCNM4       0 -Inf         
- PCNM5       0 -Inf         
- Long        0 -Inf         
- Lat         0 -Inf         

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + PCNM5 + Long +      Lat 

          Df    AIC F Pr(>F)
- PCNM5    1 37.246         
- PCNM2    1 37.670         
- PCNM4    1 37.817         
- Long     1 37.956         
- PCNM3    1 37.966         
- PCNM1    1 38.435         
- colhist  1 38.455         
- Lat      1 38.911         

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + Long + Lat 

          Df    AIC      F Pr(>F)
- Long     1 41.096 0.9154  0.640
- PCNM2    1 41.109 0.9183  0.615
- PCNM3    1 41.154 0.9279  0.610
- PCNM1    1 41.777 1.0660  0.485
- Lat      1 41.813 1.0744  0.455
- PCNM4    1 41.847 1.0821  0.370
- colhist  1 43.408 1.4765  0.155

Step: GenData ~ colhist + PCNM1 + PCNM2 + PCNM3 + PCNM4 + Lat 

          Df    AIC      F Pr(>F)
- PCNM2    1 43.336 1.2036  0.250
- PCNM3    1 44.166 1.5130  0.150
- PCNM1    1 43.931 1.4227  0.145
- Lat      1 44.070 1.4758  0.140
- PCNM4    1 44.227 1.5371  0.115
- colhist  1 44.456 1.6284  0.100

Step: GenData ~ colhist + PCNM1 + PCNM3 + PCNM4 + Lat 

          Df    AIC      F Pr(>F)
- PCNM1    1 44.558 1.2913  0.215
- PCNM4    1 44.767 1.3920  0.155
- Lat      1 44.904 1.4596  0.155
- colhist  1 44.924 1.4693  0.130
- PCNM3    1 44.876 1.4458  0.110

Step: GenData ~ colhist + PCNM3 + PCNM4 + Lat 

          Df    AIC      F Pr(>F)  
- PCNM3    1 45.146 1.3329  0.170  
- PCNM4    1 45.357 1.4591  0.140  
- colhist  1 45.635 1.6305  0.095 .
- Lat      1 46.526 2.2164  0.015 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + PCNM4 + Lat 

          Df    AIC      F Pr(>F)   
- PCNM4    1 45.429 1.4436   0.11   
- colhist  1 45.816 1.7266   0.03 * 
- Lat      1 46.508 2.2645   0.01 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ colhist + Lat 

          Df    AIC      F Pr(>F)   
- colhist  1 45.605 1.6406  0.105   
- Lat      1 46.140 2.1094  0.010 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Step: GenData ~ Lat 

      Df    AIC      F Pr(>F)   
- Lat  1 46.675 2.8464   0.01 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```



