# Adaptation at new sites

1) Adaptation to different host plants?

2) Selection for dispersing individuals? 

3) FlyBase to blast CDS of outlier scaffolds/contigs



## Outlier analyses


### Bayescan

Convert vcf to bayescan using pgdspider. 

Copy file to bluecrystal to run analysis. 

```
pwd
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/Bayescan

scp AA216.bayescan aj18951@bluecrystalp3.acrc.bris.ac.uk:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan/
```


I've created a submission script (bayescan.HostPlant.sh) on bluecrystal. To be run from the directory where the input file is located.
```
#!/bin/bash
#PBS -N bayescan1
#PBS -l nodes=1:ppn=1 #1nodes, 1 processor per node
#PBS -l mem=16gb #RAM
#PBS -l walltime=20:00:00 ##20 hours wall time.  
#PBS -j oe  #concatenates error and output files (with prefix job1)
##PBS -t 1-10

#run job in working directory
cd $PBS_O_WORKDIR  #run job in working directory
#cd 1a_Aricia_agestis_PopGenomics/Bayescan

# set a local temporary folder.  Useful if you create a lot of temp files and do not want to receive an email saying that you are filling up all the memory of the common temporary folder. It happened to me!
export TMPDIR=$HOME/.local

echo "START ----------------------------"

#load your program if it is installed globally or the modules you used to install your program locally (compilers, etc) 
#Check what is available with module avail
#module load languages/R-3.0.2

# Run program

~/software/BayeScan2.1/binaries/BayeScan2.1_linux64bits AA251.HostPlant.bayescan -output AA251.HostPlant.out -n 5000 -thin 10 -nbp 20 -pilot 5000 -burn 50000 -pr_odds 20


move the output on server to 
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan/Out_HostPlant

```

This runs for ~ 10 -15 hours. 




Copy these reults to the mac. Check that both runs have converged and plot the outliers: 

```
source("plot_R.r")
library(coda)

chain <- read.table("AA251.HostPlant.baye.sel", header=T)
chain <- mcmc(chain, thin=10)

plot(chain)  ##check for convergence
summary(chain)


Iterations = 1:49981
Thinning interval = 10 
Number of chains = 1 
Sample size per chain = 4999 

1. Empirical mean and standard deviation for each variable,
   plus standard error of the mean:

           Mean        SD  Naive SE Time-series SE
logL -2.741e+05 2.575e+02 3.642e+00      1.126e+01
Fst1  1.273e-02 4.552e-04 6.438e-06      2.635e-05
Fst2  2.235e-02 4.695e-04 6.640e-06      2.386e-05

2. Quantiles for each variable:

           2.5%        25%        50%        75%      97.5%
logL -2.746e+05 -2.742e+05 -2.741e+05 -2.739e+05 -2.736e+05
Fst1  1.182e-02  1.243e-02  1.273e-02  1.304e-02  1.363e-02
Fst2  2.145e-02  2.203e-02  2.235e-02  2.267e-02  2.331e-02


autocorr.diag(chain) ## check correlation between the chains. Make sure the chains didn't get stuck

                 logL        Fst1         Fst2
Lag 0    1.0000000000  1.00000000  1.000000000
Lag 10   0.6252686432  0.87207608  0.748341112
Lag 50   0.3133485444  0.55081485  0.430573934
Lag 100  0.1454912583  0.30930027  0.252864542
Lag 500 -0.0003691996 -0.01810393 -0.009578494


effectiveSize(chain) ##check that this is close to the sample size (here 5000). If there is correlation (chain got stuck) the sample size will be much smaller than the input
      logL     Fst1     Fst2 
522.6054     298.3115   386.9771 

geweke.diag(chain, frac1=0.1, frac2=0.5)  ##The diagnostic reports the z-scores for each parameter. For example, with α = 0.05, the critical values of z are – 1.96 and +1.96. We reject H0 (equality of means => convergence) if z < -1.96 or z > +1.96.

Fraction in 1st window = 0.1
Fraction in 2nd window = 0.5 

   logL    Fst1    Fst2 
-0.1583 -0.9004  1.5322 
 
 

heidel.diag(chain, eps=0.1, pvalue=0.05) ##another test whether the chains have reached stationarity. 

     Stationarity start     p-value
     test         iteration        
logL passed       1         0.222  
Fst1 passed       1         0.812  
Fst2 passed       1         0.572  
                                  
     Halfwidth Mean      Halfwidth
     test                         
logL passed    -2.74e+05 2.21e+01 
Fst1 passed     1.27e-02 5.17e-05 
Fst2 passed     2.24e-02 4.68e-05 
```

All parameters have converged

![alt_txt][bayescanHP.conv]

[bayescanHP.conv]:https://user-images.githubusercontent.com/12142475/119225261-a2273180-bafa-11eb-8cc0-5f32b7b96558.png



Plot cut-off for outliers
```
AA.results <- read.table("AA251.HostPlant.baye_fst.txt")

pdf("AA.HostPlant.bayescan.results")
par(mfrow=c(1,2))
plot_bayescan(AA.results, FDR=0.05, add_text=T)
plot_bayescan(AA.results, FDR=0.01, add_text=T)
dev.off()


plot_bayescan(AA.results, FDR=0.05, add_text=T)
$outliers
 [1]   697   698   700   701   702  1059  1196  1199  2530  4513  4514  6205
[13]  8436 11200 12145 12405 12413 12414 14062 14726 14727 14730 16019 17041
[25] 18124 20759 21106 21116 25257 27621 29143 34306

$nb_outliers
[1] 32


plot_bayescan(AA.results, FDR=0.01, add_text=T)
$outliers
 [1]   697   698   700   701   702  1059 12405 12413 12414 14726 14730 16019
[13] 18124 21116 34306

$nb_outliers
[1] 15


#What are these loci? 
#Get a list of locus names from the vcf file
#On bluecrystal p3
#/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF
vcftools --vcf AAgestis.251.MAF0.05.missing0.5perpop.noMT.vcf --site-mean-depth

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.251.MAF0.05.missing0.5perpop.noMT.vcf
	--site-mean-depth

After filtering, kept 251 out of 251 Individuals
Outputting Depth for Each Site
After filtering, kept 35462 out of a possible 35462 Sites
Run Time = 3.00 seconds

#Back to mac
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/Bayescan
scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF/out.ldepth.mean .


#Read into R

AA251.loci <- read.table("out.ldepth.mean", header=T)
AA251.loci$index <- c(1:(nrow(AA251.loci)))  ##to check the correct lines are selected

AA251.loci[c(697,698,700,701,702,1059,12405,12413,12414,14726,14730,16019,18124,21116,34306),]
         CHROM      POS MEAN_DEPTH VAR_DEPTH index
697    SUPER_9  8813929   223.9480  47422.00   697
698    SUPER_9  8813944   228.2830  48424.50   698
700    SUPER_9  8813959   188.4260  46612.70   700
701    SUPER_9  8813963   227.9040  48533.50   701
702    SUPER_9  8813971   227.4180  48690.30   702
1059   SUPER_9 12099911   435.8840 156694.00  1059
12405 SUPER_16 13191358    60.4462   8162.67 12405
12413 SUPER_16 13191434    60.7331   8243.13 12413
12414 SUPER_16 13191435    60.7331   8244.45 12414
14726 SUPER_18  7568053   148.7570  26489.90 14726
14730 SUPER_18  7568229   147.7130  26099.90 14730
16019  SUPER_Z  9249052    95.1116  24321.60 16019
18124 SUPER_19 11604518    86.1713  14226.20 18124
21116 SUPER_22  3379804   355.4980  94578.90 21116
34306  SUPER_8  3986867    37.6016   3670.48 34306


AA251.loci[c(697,698,700,701,702,1059,1196,1199,2530,4513,4514,6205,8436,11200,12145,12405,12413,12414,14062,14726,14727,14730,16019,17041,18124,20759,21106,21116,25257,27621,29143,34306),]

         CHROM      POS MEAN_DEPTH  VAR_DEPTH index
697    SUPER_9  8813929   223.9480  47422.000   697
698    SUPER_9  8813944   228.2830  48424.500   698
700    SUPER_9  8813959   188.4260  46612.700   700
701    SUPER_9  8813963   227.9040  48533.500   701
702    SUPER_9  8813971   227.4180  48690.300   702
1059   SUPER_9 12099911   435.8840 156694.000  1059
1196   SUPER_9 13750697   361.8650 113166.000  1196  ##only in FDR 0.05
1199   SUPER_9 13750800   354.9920 108510.000  1199  ##only in FDR 0.05
2530  SUPER_10  9707814   209.7290  56610.700  2530  ##only in FDR 0.05
4513  SUPER_11 13819357   146.2430  40795.700  4513  ##only in FDR 0.05
4514  SUPER_11 13819358   146.2590  40799.400  4514  ##only in FDR 0.05
6205  SUPER_12 12602522    79.5857   5855.830  6205  ##only in FDR 0.05
8436  SUPER_14  3347465   230.8960  64459.400  8436  ##only in FDR 0.05
11200 SUPER_16  1044290   101.6330  19014.200 11200  ##only in FDR 0.05
12145 SUPER_16 10830169    42.4143   4590.180 12145  ##only in FDR 0.05
12405 SUPER_16 13191358    60.4462   8162.670 12405
12413 SUPER_16 13191434    60.7331   8243.130 12413
12414 SUPER_16 13191435    60.7331   8244.450 12414
14062 SUPER_18   483747   192.6100  34651.300 14062  ##only in FDR 0.05
14726 SUPER_18  7568053   148.7570  26489.900 14726
14727 SUPER_18  7568063   148.7210  26479.800 14727  ##only in FDR 0.05
14730 SUPER_18  7568229   147.7130  26099.900 14730
16019  SUPER_Z  9249052    95.1116  24321.600 16019
17041  SUPER_Z 39688189   243.5180  38099.800 17041  ##only in FDR 0.05
18124 SUPER_19 11604518    86.1713  14226.200 18124
20759 SUPER_22   415039   134.1950  23690.700 20759  ##only in FDR 0.05
21106 SUPER_22  3210013   110.9880  15782.700 21106  ##only in FDR 0.05
21116 SUPER_22  3379804   355.4980  94578.900 21116
25257  SUPER_2 23292047    54.4223   7353.360 25257  ##only in FDR 0.05
27621  SUPER_4  3746442    15.5458    205.817 27621  ##only in FDR 0.05
29143  SUPER_5  2534121    96.8526  15841.500 29143  ##only in FDR 0.05
34306  SUPER_8  3986867    37.6016   3670.480 34306

```


![alt_txt][HP.bayes.FDR0.05]

[HP.bayes.FDR0.05]:https://user-images.githubusercontent.com/12142475/119225343-211c6a00-bafb-11eb-9e99-9df387ef6666.png


![alt_txt][HP.bayes.FDR0.01]

[HP.bayes.FDR0.01]:https://user-images.githubusercontent.com/12142475/119225375-490bcd80-bafb-11eb-986a-4503cfa1b397.png




#### Jackknife for Bayescan

To determine if any single population affects the identification of outliers, I'm running bayescan with each of the populations removed. 

I've repeated this analysis with populations identified as new or established (old) under ColHistory. This had to be rerun because FOR was mis-identified as an established site in the initial runs in 2019. 

```
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan
```


Rerun the checks for convergence and identify outliers in each case: 

1. Copy all outputs to the mac
```
pwd 

bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan/Out_HostPlant

#On Mac
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan/Out_HostPlant


scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/Bayescan/Out_HostPlant/AA.no* .
```


Check that runs have converged and plot the outliers: 

```
source("Plot_R.r")
library(coda)

chain <- read.table("AA.noBCH.HostPlant.baye.sel", header=T)
chain <- mcmc(chain, thin=10)

plot(chain)  ##check for convergence

autocorr.diag(chain) ## check correlation between the chains. Make sure the chains didn't get stuck


effectiveSize(chain) ##check that this is close to the sample size (here 5000). If there is correlation (chain got stuck) the sample size will be much smaller than the input
      logL     Fst1     Fst2 
522.6054     298.3115   386.9771 

```

All parameters have converged



Plot cut-off for outliers
```
AA.noBAR.results <- read.table("AA.noBAR.HostPlant.baye_fst.txt")
AA.noBCH.results <- read.table("AA.noBCH.HostPlant.baye_fst.txt")
AA.noBRO.results <- read.table("AA.noBRO.HostPlant.baye_fst.txt")
AA.noFOR.results <- read.table("AA.noFOR.HostPlant.baye_fst.txt")
AA.noLYD.results <- read.table("AA.noLYD.HostPlant.baye_fst.txt")
AA.noMOF.results <- read.table("AA.noMOF.HostPlant.baye_fst.txt")
AA.noSWD.results <- read.table("AA.noSWD.HostPlant.baye_fst.txt")
AA.noWIS.results <- read.table("AA.noWIS.HostPlant.baye_fst.txt")
AA.noHOD.results <- read.table("AA.noHOD.HostPlant.baye_fst.txt")


plot_bayescan(AA.noBAR.results, FDR=0.05, add_text=T)
$outliers
 [1]   697   698   700   701   702  1059  1196  1199  1353  1668  1672  1675
[13]  1677  1681  2530  4644  6205  8436 10819 12405 12413 12414 14062 14726
[25] 14727 14730 14733 14734 14735 14741 16019 16215 16235 18124 18459 21116
[37] 25257 30926 34306

$nb_outliers
[1] 39


plot_bayescan(AA.noBAR.results, FDR=0.01, add_text=T)
$outliers
 [1]   697   698   700   702  1059  1196  1199  1353  2530  8436 12405 12413
[13] 12414 14726 14730 14741 16019 16215 16235 18124 34306

$nb_outliers
[1] 21


plot_bayescan(AA.noBCH.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noBCH.results, FDR=0.01, add_text=T)

$outliers
 [1]   183   697   698   700   701   702  1059  6686  8793  8798 10819 10895
[13] 11200 12145 12405 12413 12414 14076 15427 16019 17040 17041 18124 20759
[25] 21106 21116 21861 25256 25257 32680 33498 33921 34306

$nb_outliers
[1] 33

$outliers
 [1]   697   698   700   701   702  1059  8793  8798 10895 12413 15427 16019
[13] 18124 20759 21106 21861 25257

$nb_outliers
[1] 17


plot_bayescan(AA.noBRO.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noBRO.results, FDR=0.01, add_text=T)

$outliers
 [1]   697   698   700   701   702  1059  1196  1199  2530  6205  6686 11200
[13] 11566 12145 12405 12413 12414 14062 14726 14730 16019 17041 18124 20752
[25] 21106 21116 25256 25257 29143 32680 33498 33921 34306

$nb_outliers
[1] 33

$outliers
 [1]   697   698   700   701   702  2530 11200 12405 12413 12414 18124 25256
[13] 25257

$nb_outliers
[1] 13


plot_bayescan(AA.noMOF.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noMOF.results, FDR=0.01, add_text=T)

$outliers
 [1]  1196  1197  1199  1353  1502  2978  2985  4347  5670  8078 11765 11767
[13] 12786 12787 12791 14260 14261 14262 14464 14733 14734 14735 14738 14741
[25] 14847 14851 14852 16019 16659 17043 17813 18010 18124 18158 19036 19037
[37] 19038 20134 24356 24994 26584 29051 30257 30265 32152 32157 33237

$nb_outliers
[1] 47


$outliers
 [1]  1196  1197  1199  1353  2985  8078 12786 12787 12791 14260 14261 14262
[13] 14464 14733 14734 14735 14738 14741 16019 16659 17043 24994 26584 32157
[25] 33237

$nb_outliers
[1] 25


plot_bayescan(AA.noWIS.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noWIS.results, FDR=0.01, add_text=T)

$outliers
 [1]   183   697   698   700   701   702  1059  2530  4644  6205  7080  9033
[13]  9038  9039  9043 11200 12145 12405 12414 16019 16229 18124 19903 19904
[25] 21106 21116 29143 34306

$nb_outliers
[1] 28

$outliers
 [1]   697   698   700   702  1059  2530  6205  7080 11200 12145 16229 21106
[13] 21116 29143 34306

$nb_outliers
[1] 15


plot_bayescan(AA.noSWD.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noSWD.results, FDR=0.01, add_text=T)

$outliers
 [1]   697   698   700   701   702  1059  2530  4513  4514  7080  8460 11200
[13] 12145 12292 12405 12413 12414 14062 14726 14727 14730 16019 16199 17041
[25] 18124 18133 19901 19902 21116 29143 33729 34306

$nb_outliers
[1] 32


$outliers
 [1]   697   698   700   701   702  2530  8460 11200 12405 12413 12414 14726
[13] 14730 16019 17041 18124

$nb_outliers
[1] 16


plot_bayescan(AA.noHOD.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noHOD.results, FDR=0.01, add_text=T)

$outliers
 [1]   697   698   700   701   702   703  1196  1197  1199  1303  1353  9581
[13] 12405 12413 12414 14062 14726 14727 14730 16019 17041 18124 18459 21116
[25] 26740 26741 26742 26744

$nb_outliers
[1] 28

$outliers
 [1]   698   700   702  1196  1199  1353 12405 12413 12414 14726 14730 16019
[13] 17041 18124

$nb_outliers
[1] 14


plot_bayescan(AA.noLYD.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noLYD.results, FDR=0.01, add_text=T)

$outliers
 [1]   697   698   700   701   702  1059  1196  1199  1303  4513  4514  4644
[13]  6205 11200 12145 12405 12413 12414 14726 14727 14730 15369 15370 15371
[25] 15420 15428 16019 17041 18124 20759 21116 25256 25257 29143 29948 33921
[37] 34306

$nb_outliers
[1] 37


$outliers
 [1]   697   698   700   701   702  1059  4513  4514 12405 12413 12414 14726
[13] 14730 15420 16019 17041 18124 25257

$nb_outliers
[1] 18


plot_bayescan(AA.noFOR.results, FDR=0.05, add_text=T)
plot_bayescan(AA.noFOR.results, FDR=0.01, add_text=T)

$outliers
 [1]   697   698   700   701   702  1059  1196  1199  1303  2530  7080 11200
[13] 12292 12405 12413 12414 14726 14730 16019 16229 17041 18124 21106 21116
[25] 27621 29143 32140 32141 34306

$nb_outliers
[1] 29


$outliers
 [1]   697   698   700   702  1059  1196  1199 12405 12413 12414 16019 18124
[13] 21116 29143

$nb_outliers
[1] 14


#What are these loci? 
#Get a list of locus names from the vcf file
#On bluecrystal p3
#/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF
vcftools --vcf AAgestis.251.MAF0.05.missing0.5perpop.noMT.vcf --site-mean-depth

VCFtools - 0.1.17
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AAgestis.251.MAF0.05.missing0.5perpop.noMT.vcf
	--site-mean-depth

After filtering, kept 251 out of 251 Individuals
Outputting Depth for Each Site
After filtering, kept 35462 out of a possible 35462 Sites
Run Time = 3.00 seconds

#Back to mac
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/Bayescan
scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/FINAL_VCF/out.ldepth.mean .


#Read into R

AA251.loci <- read.table("out.ldepth.mean", header=T)
AA251.loci$index <- c(1:(nrow(AA251.loci)))  ##to check the correct lines are selected

AA251.loci[c(697,698,700,701,702,1059,12405,12413,12414,14726,14730,16019,18124,21116,34306),]
         CHROM      POS MEAN_DEPTH VAR_DEPTH index
697    SUPER_9  8813929   223.9480  47422.00   697
698    SUPER_9  8813944   228.2830  48424.50   698
700    SUPER_9  8813959   188.4260  46612.70   700
701    SUPER_9  8813963   227.9040  48533.50   701
702    SUPER_9  8813971   227.4180  48690.30   702
1059   SUPER_9 12099911   435.8840 156694.00  1059
12405 SUPER_16 13191358    60.4462   8162.67 12405
12413 SUPER_16 13191434    60.7331   8243.13 12413
12414 SUPER_16 13191435    60.7331   8244.45 12414
14726 SUPER_18  7568053   148.7570  26489.90 14726
14730 SUPER_18  7568229   147.7130  26099.90 14730
16019  SUPER_Z  9249052    95.1116  24321.60 16019
18124 SUPER_19 11604518    86.1713  14226.20 18124
21116 SUPER_22  3379804   355.4980  94578.90 21116
34306  SUPER_8  3986867    37.6016   3670.48 34306


AA251.loci[c(697,698,700,701,702,1059,1196,1199,2530,4513,4514,6205,8436,11200,12145,12405,12413,12414,14062,14726,14727,14730,16019,17041,18124,20759,21106,21116,25257,27621,29143,34306),]

         CHROM      POS MEAN_DEPTH  VAR_DEPTH index
697    SUPER_9  8813929   223.9480  47422.000   697
698    SUPER_9  8813944   228.2830  48424.500   698
700    SUPER_9  8813959   188.4260  46612.700   700
701    SUPER_9  8813963   227.9040  48533.500   701
702    SUPER_9  8813971   227.4180  48690.300   702
1059   SUPER_9 12099911   435.8840 156694.000  1059
1196   SUPER_9 13750697   361.8650 113166.000  1196  ##only in FDR 0.05
1199   SUPER_9 13750800   354.9920 108510.000  1199  ##only in FDR 0.05
2530  SUPER_10  9707814   209.7290  56610.700  2530  ##only in FDR 0.05
4513  SUPER_11 13819357   146.2430  40795.700  4513  ##only in FDR 0.05
4514  SUPER_11 13819358   146.2590  40799.400  4514  ##only in FDR 0.05
6205  SUPER_12 12602522    79.5857   5855.830  6205  ##only in FDR 0.05
8436  SUPER_14  3347465   230.8960  64459.400  8436  ##only in FDR 0.05
11200 SUPER_16  1044290   101.6330  19014.200 11200  ##only in FDR 0.05
12145 SUPER_16 10830169    42.4143   4590.180 12145  ##only in FDR 0.05
12405 SUPER_16 13191358    60.4462   8162.670 12405
12413 SUPER_16 13191434    60.7331   8243.130 12413
12414 SUPER_16 13191435    60.7331   8244.450 12414
14062 SUPER_18   483747   192.6100  34651.300 14062  ##only in FDR 0.05
14726 SUPER_18  7568053   148.7570  26489.900 14726
14727 SUPER_18  7568063   148.7210  26479.800 14727  ##only in FDR 0.05
14730 SUPER_18  7568229   147.7130  26099.900 14730
16019  SUPER_Z  9249052    95.1116  24321.600 16019
17041  SUPER_Z 39688189   243.5180  38099.800 17041  ##only in FDR 0.05
18124 SUPER_19 11604518    86.1713  14226.200 18124
20759 SUPER_22   415039   134.1950  23690.700 20759  ##only in FDR 0.05
21106 SUPER_22  3210013   110.9880  15782.700 21106  ##only in FDR 0.05
21116 SUPER_22  3379804   355.4980  94578.900 21116
25257  SUPER_2 23292047    54.4223   7353.360 25257  ##only in FDR 0.05
27621  SUPER_4  3746442    15.5458    205.817 27621  ##only in FDR 0.05
29143  SUPER_5  2534121    96.8526  15841.500 29143  ##only in FDR 0.05
34306  SUPER_8  3986867    37.6016   3670.480 34306

```



### What are these loci? 

```
AA251.loci[c(183, 697, 698, 700, 701, 702, 703, 1059, 1196, 1197, 1199, 1303, 1353, 1502, 1668, 1672, 1675, 1677, 1681, 2530, 2978, 2985, 4513, 4514, 4644, 4513, 4514, 4515, 5670, 6205, 6686, 7080, 8436, 8793, 8798, 9033, 9038, 9039, 9043, 9581, 10819, 10895, 11200, 11566, 11765, 11767, 12145, 12292, 12405, 12413, 12414, 12786,12787, 12791, 14062, 14076, 14260, 14261, 14262, 14464, 14726, 14727, 14730, 14733, 14734, 14735, 14738, 14741, 14847, 14851, 14852, 15369, 15370, 15371, 15420, 15427, 15428, 16019, 16199,16215, 16229, 16659, 17040, 17041, 17043, 17813, 18010, 18124, 18133, 18158, 18459, 19036, 19037,19038, 19901, 19902, 19903, 19904, 20134, 20752, 20759, 21106, 21116, 21861, 24356, 24994, 25256,25257, 26584, 26740, 26741, 26742, 26744, 27621, 29051, 29143, 29948, 30257, 30265, 32140, 32141, 32152, 32157, 32680, 33237, 33498, 33729, 33921, 34306 ),]

          CHROM      POS MEAN_DEPTH  VAR_DEPTH index
183     SUPER_9  2376094    71.7291   5283.570   183
697     SUPER_9  8813929   223.9480  47422.000   697
698     SUPER_9  8813944   228.2830  48424.500   698
700     SUPER_9  8813959   188.4260  46612.700   700
701     SUPER_9  8813963   227.9040  48533.500   701
702     SUPER_9  8813971   227.4180  48690.300   702
703     SUPER_9  8813992   281.0760  47649.900   703
1059    SUPER_9 12099911   435.8840 156694.000  1059
1196    SUPER_9 13750697   361.8650 113166.000  1196
1197    SUPER_9 13750721   361.6060 112994.000  1197
1199    SUPER_9 13750800   354.9920 108510.000  1199
1303    SUPER_9 15710563   128.4460  19856.100  1303
1353    SUPER_9 15951750   137.5220  23652.700  1353
1502    SUPER_9 17244878   405.5820  54397.900  1502
1668   SUPER_10   594511   117.1790   9509.680  1668
1672   SUPER_10   594536   117.4660   9551.110  1672
1675   SUPER_10   594553   117.4820   9552.510  1675
1677   SUPER_10   594562   117.4980   9557.680  1677
1681   SUPER_10   594580   117.5380   9556.390  1681
2530   SUPER_10  9707814   209.7290  56610.700  2530
2978   SUPER_10 14888691   162.8410  35966.800  2978
2985   SUPER_10 14888830   163.4660  35985.300  2985
4513   SUPER_11 13819357   146.2430  40795.700  4513
4514   SUPER_11 13819358   146.2590  40799.400  4514
4644   SUPER_11 15120248   391.2630  61411.600  4644
4513.1 SUPER_11 13819357   146.2430  40795.700  4513
4514.1 SUPER_11 13819358   146.2590  40799.400  4514
4515   SUPER_11 13819376   146.1630  40779.300  4515
5670   SUPER_12  7492320   159.2030  34529.100  5670
6205   SUPER_12 12602522    79.5857   5855.830  6205
6686   SUPER_13  1247995   105.3980  22787.700  6686
7080   SUPER_13  5040829   162.1550  32022.900  7080
8436   SUPER_14  3347465   230.8960  64459.400  8436
8793   SUPER_14  6984348   158.8880  21440.200  8793
8798   SUPER_14  6984462   159.5020  21607.200  8798
9033   SUPER_14  9498776   287.6410  78595.500  9033
9038   SUPER_14  9498820   284.4620  76694.200  9038
9039   SUPER_14  9498827   284.0080  76427.200  9039
9043   SUPER_14  9498914   288.1000  79328.000  9043
9581   SUPER_14 14556194   153.6410  35697.000  9581
10819  SUPER_15 12292122   265.0920  66244.000 10819
10895  SUPER_15 12947184   227.6690  37352.500 10895
11200  SUPER_16  1044290   101.6330  19014.200 11200
11566  SUPER_16  4912028    75.8247  11785.300 11566
11765  SUPER_16  6628829   195.9600  40765.000 11765
11767  SUPER_16  6628936   196.2350  40475.500 11767
12145  SUPER_16 10830169    42.4143   4590.180 12145
12292  SUPER_16 12325522   187.5420  66355.100 12292
12405  SUPER_16 13191358    60.4462   8162.670 12405
12413  SUPER_16 13191434    60.7331   8243.130 12413
12414  SUPER_16 13191435    60.7331   8244.450 12414
12786  SUPER_17  1198584   190.8960  60791.600 12786
12787  SUPER_17  1198594   185.2790  60790.400 12787
12791  SUPER_17  1198741   191.3590  61074.700 12791
14062  SUPER_18   483747   192.6100  34651.300 14062
14076  SUPER_18   550222   356.8840  85094.500 14076
14260  SUPER_18  2919521   116.9360  25297.000 14260
14261  SUPER_18  2919522   116.9280  25298.500 14261
14262  SUPER_18  2919538   116.9480  25306.800 14262
14464  SUPER_18  5119475    73.2629   6299.310 14464
14726  SUPER_18  7568053   148.7570  26489.900 14726
14727  SUPER_18  7568063   148.7210  26479.800 14727
14730  SUPER_18  7568229   147.7130  26099.900 14730
14733  SUPER_18  7677559   400.6530 145723.000 14733
14734  SUPER_18  7677561   400.6730 145618.000 14734
14735  SUPER_18  7677568   406.2830 144780.000 14735
14738  SUPER_18  7677579   405.7010 144328.000 14738
14741  SUPER_18  7677642   406.0720 144481.000 14741
14847  SUPER_18  8586474   160.1350  17459.000 14847
14851  SUPER_18  8586524   161.1200  17606.600 14851
14852  SUPER_18  8586525   161.1270  17606.300 14852
15369   SUPER_Z   417791    84.1952   5670.300 15369
15370   SUPER_Z   417792    84.1952   5668.540 15370
15371   SUPER_Z   417806    83.2351   5613.860 15371
15420   SUPER_Z   812539   147.1430  17566.000 15420
15427   SUPER_Z   826388    97.2470  11032.300 15427
15428   SUPER_Z   826392    93.7530   9891.480 15428
16019   SUPER_Z  9249052    95.1116  24321.600 16019
16199   SUPER_Z 13941340   443.7570 115979.000 16199
16215   SUPER_Z 14608282    86.0239   5081.260 16215
16229   SUPER_Z 14642760   120.8210  14967.800 16229
16659   SUPER_Z 28513286   139.5020  47182.200 16659
17040   SUPER_Z 39688177   243.5340  38098.900 17040
17041   SUPER_Z 39688189   243.5180  38099.800 17041
17043   SUPER_Z 39688283   243.0080  37236.000 17043
17813  SUPER_19  7875759   318.7330  50341.100 17813
18010  SUPER_19 10309835   348.6850 195697.000 18010
18124  SUPER_19 11604518    86.1713  14226.200 18124
18133  SUPER_19 11737503    38.9562   2345.170 18133
18158  SUPER_19 12082469   242.8330  74888.300 18158
18459  SUPER_20  1277305    98.4980  13748.900 18459
19036  SUPER_20  8466280    65.8247   4721.230 19036
19037  SUPER_20  8466283    65.8367   4723.520 19037
19038  SUPER_20  8466285    65.8645   4729.710 19038
19901  SUPER_21  2586748   135.8130  24349.300 19901
19902  SUPER_21  2586873   133.7890  23612.000 19902
19903  SUPER_21  2586874   133.9280  23656.100 19903
19904  SUPER_21  2586922   136.3780  24659.000 19904
20134  SUPER_21  5045894   139.5500  18002.100 20134
20752  SUPER_22   408410   177.1040  36450.900 20752
20759  SUPER_22   415039   134.1950  23690.700 20759
21106  SUPER_22  3210013   110.9880  15782.700 21106
21116  SUPER_22  3379804   355.4980  94578.900 21116
21861   SUPER_1  4151512   416.5580 144372.000 21861
24356   SUPER_2  7874723   222.9120  53974.500 24356
24994   SUPER_2 19346581   137.0000  44215.300 24994
25256   SUPER_2 23291976    54.4183   7360.000 25256
25257   SUPER_2 23292047    54.4223   7353.360 25257
26584   SUPER_3 13157714   208.1590  32943.200 26584
26740   SUPER_3 15356257   176.8450  27589.400 26740
26741   SUPER_3 15356303   176.5460  27564.500 26741
26742   SUPER_3 15356334   165.9280  25812.900 26742
26744   SUPER_3 15356479   175.9320  27347.400 26744
27621   SUPER_4  3746442    15.5458    205.817 27621
29051   SUPER_5  1546360   323.4860  65428.300 29051
29143   SUPER_5  2534121    96.8526  15841.500 29143
29948   SUPER_5 13174410    57.6892  10227.400 29948
30257   SUPER_5 17868004    61.1992   6103.860 30257
30265   SUPER_5 17868196    64.9203   6678.780 30265
32140   SUPER_6 19126556   385.0960  77119.600 32140
32141   SUPER_6 19126567   380.4180  78955.300 32141
32152   SUPER_6 19243803   261.2630 100005.000 32152
32157   SUPER_6 19243964   262.0360 100937.000 32157
32680   SUPER_7  4419088    61.5020  18298.500 32680
33237   SUPER_7 10650289    60.5458   6791.460 33237
33498   SUPER_7 14449727   190.4980  27688.900 33498
33729   SUPER_7 18118606    55.0717   4384.070 33729
33921   SUPER_8    25238   201.3510  49198.100 33921
34306   SUPER_8  3986867    37.6016   3670.480 34306
```


### BLAST of candidates

Working on the server
```
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/RefGenome
```

I have a list of the contigs/scaffolds where outliers for adaptation to host plant preference were found: 
```
grep ">" outlier.fasta >> outlier.names
sed -i 's/>//g' outlier.names
cat outlier.names
m_scaff_962
m_scaff_5443
contig_41650
m_scaff_4913
m_scaff_5318
m_scaff_5502
m_scaff_5215
m_scaff_4152
m_scaff_4399
m_scaff_6168
m_scaff_486
m_scaff_5063
```

Extract info from gff for each of the outliers: 
```
for f in `cat outlier.names`; do grep $f Aricia_agestis_Red_MESPA.gff >> outlier.bed; done
grep cds outlier.bed >> outlier2.bed
awk -F'\t' '{print $1,"\t",$4,"\t",$5}' outlier2.bed >> outlier3.bed
##sed -i 's/ //g' outlier3.bed ##uncomment if compaint about needing tab delimited file


module load apps/bedtools2
bedtools getfasta -fi Aricia_agestis_Red_MESPA.fasta -bed outlier3.bed -fo out.fasta
```

Submit to [FlyBase]()


### Go Ontology

Gene ontology enrichment in outlier loci

```
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("topGO", version = "3.8")



```



## THE Following analyses were NOT used in final paper ##

PCAdapt

BayEnv

LFMM

### PCAdapt

Initial PCAdapt steps completed in [3.GeneticDiversity&Structure.md](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/3.GeneticDiversity%26Structure.md)


Vignette: https://cran.r-project.org/web/packages/pcadapt/vignettes/pcadapt.html

Useful info about interpreting p-value distributions: http://varianceexplained.org/statistics/interpreting-pvalue-histogram/

I did this in R following the vignette quite closely:

I chose K=2 following the scree plot from PCAdapt

```
AA261.bed <- read.pcadapt("plink.bed", type="bed")
x.maf0.05 <- pcadapt(AA261.bed, K=2, min.maf=0.05)   ##calculate z-statistics and transformed values for chi-squared distribution
x.maf0.1 <- pcadapt(AA261.bed, K=2, min.maf=0.01) #calculate for maf 0.1

pdf(file="pcadapt.pvalues_AA261.pdf")
par(mfrow=c(2,1))
hist(x.maf0.05$pvalues,xlab="p-values maf0.05",main=NULL,breaks=50)
hist(x.maf0.1$pvalues,xlab="p-values maf0.01",main=NULL,breaks=50)
dev.off()

```


![alt_txt][maf]

[maf]:https://user-images.githubusercontent.com/12142475/54209447-b2343700-44d5-11e9-82ed-51999125f95c.png



I'll use a MAF cut-off of 1% as this yields more candidate loci, and the p-value distribution is fairly normal. 


use qvalue in R to specify FDR

```
library(qvalue)
alpha <- 0.05
qval <- qvalue(x.maf0.05$pvalues)$qvalues
outliers <- which(qval<alpha)
outliers
snp_pc <- get.pc(x.maf0.05,outliers) ##see PCs associated with the outliers
```

The outliers identified are mostly associated with principal component 2. I.e. they explain the divergence of HOD from the rest of the populations. 


Write qval for both analyses to file. These will be ordered later for comparison between all analyses
```
qval <- qvalue(x.maf0.05$pvalues)$qvalues
write.table(qval, "PCAdapt.qval.maf0.05_AA216", quote=F, row.names=F, col.names=F)

qval <- qvalue(x.maf0.1$pvalues)$qvalues
write.table(qval, "PCAdapt.qval.maf0.01_AA216", quote=F, row.names=F, col.names=F)
```


## Environmental Association Analyses

### BayEnv2

Input files:

https://bitbucket.org/tguenther/bayenv2_public/src/8e4039f64d61?at=default

Input file with subset of 500 SNPs for co-variance matrix (pop structure)

Input file with all SNPs for the association analysis

ENV input file with normalised environmental parameters. Here I use HostPlant at site. 

NB: population order in the input files should all be the same.

1. Convert to Bayenv

Use pgdspider2.1.0.3 and popfile for pgdspider

I'm using the filtered dataset: only one SNP per locus. This way I can calculate the neutral co-variance in the dataset.


2. Select 500 SNPs for covariance matrix


First convert vcf to bayenv using the thinned vcf file. This is ~1200 unlinked loci

```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/BayENV


vcftools --vcf AA261.0.5miss.9popsMerged.vcf --thin 600 --recode --recode-INFO-all --out AA261.thin

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.0.5miss.9popsMerged.vcf
	--recode-INFO-all
	--thin 600
	--out AA261.thin
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 3819 out of a possible 31381 Sites
Run Time = 6.00 seconds

```

Then use sed to select the first 500 loci in the file. These are called by contig, but have not been sorted using Heliconius, so should represent a random subset of the genome.

```
sed -n 1,1000p AA261.thin.BayEnv.txt > AA261.bayenv.500loci.txt
```

3. Estimate covariance matrix

This runs fairly quickly (~2hours for 132 indivs, 500loci), so I ran it on the mac.

The command needs to be run from the installation folder: /Users/alexjvr/Applications/bayenv2/compiled_on_a_mac
```
cp AA261.bayenv.500loci.txt /Users/alexjvr/Applications/bayenv2/compiled_on_a_mac/

cd /Users/alexjvr/Applications/bayenv2/compiled_on_a_mac/

./bayenv2 -i AA261.bayenv.500loci.txt -p 9 -k 100000 r23468 > AA261.500.matrix.out

./bayenv2 -i AA261.bayenv.500loci.txt -p 9 -k 100000 r67328 > AA261.500.matrix2.out
```
Check that the covariance matrix are highly correlated within and between runs.

I will copy 10 matrices randomly from each of the two runs, and visualise their correlation in R. As well as compare this with the pairwise Fst (this can be used to double check the order of the populations in the matrix).

I made new files for a matrix every 10k steps from both runs. ie. 10 matrices per run.
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/DataAnalysis_20181116/BayEnv

matrix1.10 <- read.table("matrix1.10")
matrix1.10 <- as.matrix(matrix1.10)

matrix2.10 <- read.table("matrix2.10")
matrix2.10 <- as.matrix(matrix2.10)


matrix1.100 <- read.table("matrix1.100")
matrix1.100 <- as.matrix(matrix1.100)

matrix2.100 <- read.table("matrix2.100")
matrix2.100 <- as.matrix(matrix2.100)


par(mfrow=c(1,2))
image(matrix1.10)
image(matrix2.10)


diss1.10 <- 1-cor(matrix1.10)
diss1.100 <- 1-cor(matrix1.100)
diss2.10 <- 1-cor(matrix2.10)
diss2.100 <- 1-cor(matrix2.100)


##calculate correlation between all individuals and determine whether matrixes are converged within and between runs: 

> cor.test(diss1.10, diss1.100)

	Pearson's product-moment correlation

data:  diss1.10 and diss1.100
t = Inf, df = 79, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 1 1
sample estimates:
cor 
  1 

> diss2.10 <- 1-cor(matrix2.10)
> cor.test(diss2.10, diss2.100)

	Pearson's product-moment correlation

data:  diss2.10 and diss2.100
t = Inf, df = 79, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 1 1
sample estimates:
cor 
  1 

> cor.test(diss1.10, diss2.10)

	Pearson's product-moment correlation

data:  diss1.10 and diss2.10
t = 16.801, df = 79, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8247593 0.9239490
sample estimates:
      cor 
0.8839262 

> cor.test(diss1.100, diss2.100)

	Pearson's product-moment correlation

data:  diss1.100 and diss2.100
t = 16.801, df = 79, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8247593 0.9239490
sample estimates:
      cor 
0.8839262 
```


Plot distance trees and visually inspect:
```
par(mfrow=c(2,1))
plot(hclust(as.dist(diss1.10)), main="matrix1.10")
plot(hclust(as.dist(diss2.10)), main="matrix2.10")
par(mfrow=c(2,1))
plot(hclust(as.dist(diss2.100)), main="matrix2.100")
plot(hclust(as.dist(diss1.100)), main="matrix1.100")
```

![alt_txt][hclust]

[hclust]:https://user-images.githubusercontent.com/12142475/54624096-a52ebf00-4a64-11e9-948a-e2989048ec01.png



##### Prepare all the input files

1. convert vcf to bayenv input using pgdspider

2. copy the last covariance matrix out of one of the matrix files and paste in a new file xx.MATRIX

3. Create the xx.ENV file by transposing the variables. ie. populations are columns. I'm not normalising these as they're just binary. 


The analysis is run in the folder with the source code.
```
/Users/alexjvr/Applications/bayenv2/compiled_on_a_mac/AAgestis261

##split the input file into single locus files

split -a 10 -l 2 AA261.0.5miss.9popsMerged.BayENV.txt snp_batch

head *ENV   ##the ENV file format. Populations in alphabetical order as in the vcf file. 

2	1	2	1	1	1	2	1	2
1	1	2	1	2	1	2	1	2



#Start the run

for f in $(ls snp_batch*); do ./bayenv2 -i $f -e AA261.9pops.ENV -m AA261.MATRIX -k 100000 -p 9 -n 2 -r $RANDOM -t -c -X; done
```


I submitted this in a script on bluecrystal: 
```
#!/bin/bash
#PBS -N bayEnv1
#PBS -l nodes=1:ppn=1 #1nodes, 1 processor per node
#PBS -l mem=16gb #RAM
#PBS -l walltime=100:00:00 ##20 hours wall time.  
#PBS -j oe  #concatenates error and output files (with prefix job1)
##PBS -t 1-100

#run job in working directory
cd $PBS_O_WORKDIR  #run job in working directory
#cd 1a_Aricia_agestis_PopGenomics/Bayescan

# set a local temporary folder.  Useful if you create a lot of temp files and do not want to receive an email saying that you are filling up all the memory of the common temporary folder. It happened to me!
export TMPDIR=$HOME/.local

echo "START ----------------------------"

#load your program if it is installed globally or the modules you used to install your program locally (compilers, etc) 
#Check what is available with module avail
#module load languages/R-3.0.2

##set up array

#NAME=$(sed "${PBS_ARRAYID}q;d" snp_batch)

#./bayenv2 -i $NAME -e AA261.9pops.ENV -m AA261.MATRIX -k 100000 -p 9 -n 2 -r $RANDOM -t -c -X

for f in $(ls snp_batch/snp_batch*); do ./bayenv2 -i $f -e AA261.9pops.ENV -m AA261.MATRIX -k 100000 -p 9 -n 2 -r $RANDOM -t -c -X; done
```






### LFMM

See tutorial for command line version: http://membres-timc.imag.fr/Olivier.Francois/lfmm/files/LEA_1.html

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/LFMM/SE132.FINAL

Input files

.env environmental file

1.Genotype file: vcf file of loci genotyped across all populations. 

2. Env file: environmental information for each individual


Env File
```
library(LEA)

env <- read.table("EnvVars_noheaders_AA216", header=F) ###read in the environmental data. One line perindividual. Same order as .vcf file

#Remember that the env variables need to be numeric. Here Var1 = Col history.
#old=1, new=2
#Var2=HostPlant
#Rockrose=1
#Geranium=2

write.env(env, "AA216.env")   ##convert to correct .env format
##Check that this file is numeric! i.e. remove any quote marks using sed in bash or in as.numberic in R. 
##This gives a very uninformative error if the input file is wrong. It says the first SNP is fixed across populations, and then it quits with "Error in obj.lfmm... "
```


Genotype file.
```
Convert .vcf to .lfmm in LEA

library(LEA)
genotype = vcf2lfmm("AA261.0.5miss.9popsMerged.vcf")

	- number of detected individuals:	261
	- number of detected loci:		31381

For SNP info, please check ./AA261.0.5miss.9popsMerged.vcfsnp.

0 line(s) were removed because these are not SNPs.
Please, check ./AA261.0.5miss.9popsMerged.removed file, for more informations.


	- number of detected individuals:	261
	- number of detected loci:		31381
```

Population structure with SNMF
The first step is to evaluate the population structure. SNMF works on the same principles as STRUCTURE, but is much faster.
```
Test K1:10

obj.snmf = snmf(genotype, K=1:10, entropy=T, ploidy=2, project="new")

##The project is saved into : AA216.thin.recode.snmfProject

##To load the project, use:

project = load.snmfProject("AA216.thin.recode.snmfProject")

##To remove the project, use: remove.snmfProject("AA216.thin.recode.snmfProject")

Since the program was called with the entropy option, we can plot the values of the cross-entropy criterion for each K. The value for which the function plateaus or increases is our estimate of K

plot(obj.snmf)
```


![alt_txt][snmf]

[snmf]:https://user-images.githubusercontent.com/12142475/54373447-2b1dc500-4675-11e9-9e76-47ab8b3d49c9.png



The minimum is at 2 (fastStructure also suggested 2). We can visualise the plot
```
barplot(t(Q(obj.snmf, K=2)), col=1:2)

```


![alt_txt][snmf_barplot]

[snmf_barplot]:https://user-images.githubusercontent.com/12142475/54373561-60c2ae00-4675-11e9-9a26-907307e873ae.png

Populations in alphabeticaal order


Run LFMM
```
obj.lfmm = lfmm(genotype, "AA216.env", K=2, rep=5, project="new")  #This was using HostPlant and ColHistory
```
I ran this on the fgcz server. 

```
/srv/kenlab/alexjvr_p1795/AAgestis/LFMM

```





## Concatenation of results



First sort the Aricia agestis gff file according to the Hmel gff file, so that we can assign linkage groups and start positions to each line. 
```

library(data.table)

### Load and reformat Hmel gff for later

Hmel25gff <- read.table("Hmel2.5_LG/Heliconius_melpomene_melpomene_Hmel2.5.gff3",sep="\t")

Hmel25gff_collapsed <- Hmel25gff[Hmel25gff$V3=="gene",]
Hmel25gff_collapsed$ID <- as.character(gsub("ID=", "\\1", Hmel25gff_collapsed$V9))



### Load and reformat Aricia gff
library(data.table)
library(dplyr)
library(ggplot2)

Argff <- read.table("Redundans_loop/Aricia_agestis/Aricia_agestis_Red_MESPA.gff",sep="\t")

Argff_collapsed <- Argff[Argff$V3=="cds",]

Argff_collapsed$ID <- as.character(gsub(".*Target=(HMEL[0-9]*..).*", "\\1", Argff_collapsed$V9))

Argff_collapsed$Scaff <- as.character(gsub(".*_(Hmel[0-9]*o) .*", "\\1", Argff_collapsed$V9))
Argff_collapsed$LG <- as.character(gsub(".*_Hmel2([0-9][0-9])[0-9]*o .*", "\\1", Argff_collapsed$V9))

##remove unassignable Hmel gene hits
Argff_collapsed <- Argff_collapsed[Argff_collapsed$LG != '00',]

Argff_unique_fulldata <- setDT(Argff_collapsed)[, lapply(.SD, paste, collapse = "; "), by = V1]

### Collapse Hmel cds hits into one line for each Aricia scaffold
Argff_unique <- setDT(Argff_collapsed)[, .(count = .N, var = uniqueN(Scaff)), by = V1]
##Make a vector of contigs with conflicting Hmel hits
Argff_unique <- Argff_unique$V1[Argff_unique$var==1]

##remove genes with conflicting hits from full data table
Argff_unique_fulldata <- Argff_unique_fulldata[Argff_unique_fulldata$V1 %in% Argff_unique]
##take first geneID from the cds hits
Argff_unique_fulldata2 <- as.data.frame(Argff_unique_fulldata[, paste0("Agene", 1) := tstrsplit(ID, "; ")[1]])
Assignment_table$LG_assigned <- substr(Assignment_table$LG, start = 1, stop = 2)

new <- Assignment_table %>% mutate(Hmel.pos=Hmel25gff_collapsed$V4[match(Agene1,Hmel25gff_collapsed$ID)]) ##look up the Hmel start position for each add a new column to Assignment table. 

ggplot(data=Assignment_table) + geom_point(aes(x=LG_assigned,y=Hmel.pos))
```

![alt_txt][Hmel.pos]

[Hmel.pos]:https://user-images.githubusercontent.com/12142475/54429390-47bf0900-4718-11e9-9de5-561cb89a28e3.png


get all the locus names. I used the lfmm output: 
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/LFMM

awk '{print $1, $2, $1"_"$2}' AA261.0.5miss.9popsMerged.vcfsnp  > locus.names

```


Back in R in:
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/AdaptationHostPlant

Create a file with 

1. AAgestis locus names

2. Hmel LG

3. Hmel pos

4. Outlier stats for each of the analyses
```

AA261.locus.stats <- read.table("locus.names_AA261.31381", header=F)
head(AA261.locus.stats)

         V1   V2             V3
1 m_scaff_5 5253 m_scaff_5_5253
2 m_scaff_5 5257 m_scaff_5_5257
3 m_scaff_5 5269 m_scaff_5_5269
4 m_scaff_5 5300 m_scaff_5_5300
5 m_scaff_5 5322 m_scaff_5_5322
6 m_scaff_5 5325 m_scaff_5_5325

##Add bayescan info
bayescan <- read.table("AA216.HostPlantPop.baye_fst.txt", header=T)
AA261.locus.stats$Bayescan <- bayescan$fst

##Add Hmel info to this table
AA261.locus.stats <- AA261.locus.stats %>% mutate(Agene1=Assignment_table$Agene1[match(V1, Assignment_table$V1.AAgestis.scaffold)])
AA261.locus.stats$LG.pos <- paste((as.numeric(AA261.locus.stats$LG_assigned)),(as.numeric(AA261.locus.stats$Hmel.pos)), sep="_")

pdf("Bayescan.ordere.pdf")
ggplot(AA261.locus.stats, aes(x=LG.pos, y=Bayescan)) + geom_point()
dev.off()
```

![alt_txt][bayescan.ordered]

[bayescan.ordered]:https://user-images.githubusercontent.com/12142475/54528772-ca90cf80-4975-11e9-8371-199a282dd134.png


1. Many of the outliers are on the unannotated bit of the genome.

2. There's a big outlier on ChromI (sex chrom). (sex-linked selection?) 

3. There are outliers across the whole genome. 


[Manhattan Plot](https://genome.sph.umich.edu/wiki/Code_Sample:_Generating_Manhattan_Plots_in_R)

```



```

