# Estimate divergence time between adaptive loci 


https://helda.helsinki.fi/bitstream/handle/10138/298668/journal.pgen.1007796.pdf?sequence=1&isAllowed=y


Here I aim to test: 

1) age of the outlier loci

2) origin of the outlier loci 


## fineRADstructure of outlier loci

fineRADstructure looks at the recent coancestry of loci. I can rerun this analysis using only the outlier loci identified by Bayescan. 


Filter vcf file for scaffolds that contain the outlier loci: 

HostPlant
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure

vcftools --vcf AA261.newnames.vcf --chr contig_5345 --chr contig_17378 --chr contig_5407 --chr contig_19343 --chr contig_3838 --chr contig_2518 --chr m_scaff_962 --chr contig_11951 --recode --recode-INFO-all --out AA261.HostPlantOutlierScaffolds

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.newnames.vcf
	--chr contig_11951
	--chr contig_17378
	--chr contig_19343
	--chr contig_2518
	--chr contig_3838
	--chr contig_5345
	--chr contig_5407
	--chr m_scaff_962
	--recode-INFO-all
	--out AA261.HostPlantOutlierScaffolds
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 109 out of a possible 31381 Sites
Run Time = 4.00 seconds


## Keep only individuals genotyped at more than 50% of the loci
vcftools --vcf AA261.HostPlantOutlierScaffolds.recode.vcf --keep keep.HostPlant --recode --recode-INFO-all --out AA250.HostPlantOutlierScaffolds

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.HostPlantOutlierScaffolds.recode.vcf
	--keep keep.HostPlant
	--recode-INFO-all
	--out AA250.HostPlantOutlierScaffolds
	--recode

Keeping individuals in 'keep' list
After filtering, kept 250 out of 261 Individuals
Outputting VCF file...
After filtering, kept 109 out of a possible 109 Sites
Run Time = 2.00 seconds
AJvR:fineRADstructure alexjvr$ vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --max-missing 0.5

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA250.HostPlantOutlierScaffolds.recode.vcf
	--max-missing 0.5

After filtering, kept 250 out of 250 Individuals
After filtering, kept 109 out of a possible 109 Sites
Run Time = 1.00 seconds
```


ColHist
```
vcftools --vcf AA261.newnames.vcf --chr contig_1784 --chr contig_5345 --chr contig_16912 --chr contig_3360 --chr contig_19343 --chr contig_59667 --chr contig_24918 --chr contig_3838 --chr contig_18281 --chr contig_1883 --chr contig_19566 --chr contig_19564 --recode --recode-INFO-all --out AA261.ColHistOutlierScaffolds


VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.newnames.vcf
	--chr contig_16912
	--chr contig_1784
	--chr contig_18281
	--chr contig_1883
	--chr contig_19343
	--chr contig_19564
	--chr contig_19566
	--chr contig_24918
	--chr contig_3360
	--chr contig_3838
	--chr contig_5345
	--chr contig_59667
	--recode-INFO-all
	--out AA261.ColHistOutlierScaffolds
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 169 out of a possible 31381 Sites
Run Time = 4.00 seconds

##Remove indivs with >50% missing data
vcftools --vcf AA261.ColHistOutlierScaffolds.recode.vcf --keep keep.indivs --recode --recode-INFO-all --out AA254.ColHistOutlierScaffolds

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.ColHistOutlierScaffolds.recode.vcf
	--keep keep.indivs
	--recode-INFO-all
	--out AA254.ColHistOutlierScaffolds
	--recode

Keeping individuals in 'keep' list
After filtering, kept 254 out of 261 Individuals
Outputting VCF file...
After filtering, kept 169 out of a possible 169 Sites
Run Time = 2.00 seconds
AJvR:fineRADstructure alexjvr$ vcftools --vcf AA254.ColHistOutlierScaffolds.recode.vcf --max-missing 0.5

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA254.ColHistOutlierScaffolds.recode.vcf
	--max-missing 0.5

After filtering, kept 254 out of 254 Individuals
After filtering, kept 169 out of a possible 169 Sites
Run Time = 0.00 seconds

```

Convert to fineRADstructure input
```
./vcf2hapmatrix.py AA261.ColHistOutlierScaffolds.recode.vcf > AA261.ColHistOutliers.fineradstr
./vcf2hapmatrix.py AA261.HostPlantOutlierScaffolds.recode.vcf > AA261.HostPlantOutliers.fineradstr

##with indivs removed
./vcf2hapmatrix.py AA254.ColHistOutlierScaffolds.recode.vcf > AA254.ColHistOutliers.fineradstr
./vcf2hapmatrix.py AA250.HostPlantOutlierScaffolds.recode.vcf > AA250.HostPlantOutlierScaffolds.fineradstr
```


Run fineRADstr
```
./RADpainter paint AA261.HostPlantOutliers.fineradstr 
Painting RAD tags from: AA261.HostPlantOutliers.fineradstr
The file seems to be in a Matrix format
BAR1	nan
BAR1	nan
Printing missingness per individual to: AA261.HostPlantOutliers_missingness.out
Theoretical c = 0.00769231
Jackknife c = nan
notInformative = 0
Printing the final coancestry matrix to AA261.HostPlantOutliers_chunks.out
Analysis completed in: 0.560336 seconds (0.080048 seconds per RAD locus)

```


## Haplotype network of outlier RADtags


Identify the scaffolds and region where the outlier loci are. I can see this in my locus.stats file in R

```
getwd()
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/AdaptationHostPlant

> outliers.HostPlant
                V1    V2                 V3       Agene1 LG_assigned Hmel.pos
2072   m_scaff_962 19860  m_scaff_962_19860 HMEL003137g1          01  3418280
10676 contig_11951 10614 contig_11951_10614         <NA>        <NA>       NA
11041  contig_5345   510    contig_5345_510         <NA>        <NA>       NA
11042  contig_5345   511    contig_5345_511         <NA>        <NA>       NA
11050  contig_5345   587    contig_5345_587         <NA>        <NA>       NA
12108 contig_17378  3882  contig_17378_3882         <NA>        <NA>       NA
16942  contig_5407  5785   contig_5407_5785         <NA>        <NA>       NA
17001 contig_19343  4711  contig_19343_4711         <NA>        <NA>       NA
21375  contig_3838  3376   contig_3838_3376         <NA>        <NA>       NA
21377  contig_3838  3378   contig_3838_3378         <NA>        <NA>       NA
21378  contig_3838  3382   contig_3838_3382         <NA>        <NA>       NA
21379  contig_3838  3408   contig_3838_3408         <NA>        <NA>       NA
26646  contig_2518 13094  contig_2518_13094         <NA>        <NA>       NA
      Bayescan    LG.pos     LFMM.zs   LFMM.invP chr     pos   pvalue
2072  0.075214 1_3418280 -1.29200000 0.852724236   1 3418280 0.075214
10676 0.070423     NA_NA -0.92842900 0.702313781  NA      NA 0.070423
11041 0.090865     NA_NA  0.00963256 0.008620760  NA      NA 0.090865
11042 0.092381     NA_NA  0.00433303 0.003877951  NA      NA 0.092381
11050 0.095242     NA_NA  0.06568980 0.058737760  NA      NA 0.095242
12108 0.082410     NA_NA  1.90364000 0.967262743  NA      NA 0.082410
16942 0.071641     NA_NA  1.39249000 0.881697843  NA      NA 0.071641
17001 0.095215     NA_NA -1.49713000 0.906909006  NA      NA 0.095215
21375 0.095010     NA_NA -0.75066800 0.600220766  NA      NA 0.095010
21377 0.092891     NA_NA -0.73351900 0.589366742  NA      NA 0.092891
21378 0.095510     NA_NA -0.76999000 0.612241154  NA      NA 0.095510
21379 0.089935     NA_NA -0.71849300 0.579714051  NA      NA 0.089935
26646 0.082731     NA_NA -1.31554000 0.859955662  NA      NA 0.082731
      BayescanColHist
2072         0.022119
10676        0.022163
11041        0.083876
11042        0.083073
11050        0.085587
12108        0.021469
16942        0.031380
17001        0.083373
21375        0.105340
21377        0.104240
21378        0.109400
21379        0.100140
26646        0.073927

```

Here I can see the scaffold and approximate position of the RAD tag of interest. Extract all these tags from the mapped bam files for the population data. 
This is done on the blue crystal server. 

```
pwd
#/newhome/aj18951/1a_Aricia_agestis_PopGenomics/mapped

module load apps/samtools-1.8
module load apps/bamtools-1.8
for i in $(ls *gz.bam); do samtools view -b -F4 $i m_scaff_962:19000-20000 >> $i.m_scaff_962.subset.bam; done
##remember -F4 to filter for properly paired and mapped reads only. 

```



## Haplotype network of outliers vs overall

I will extract all the outliers from the vcf file and draw a haplotype network for this in R. 


Create a vcf file for each of the scaffolds/contigs that outliers appear on: 
```
##Hostplant

vcftools --vcf AA261.newnames.vcf --chr contig_5345 --chr contig_17378 --chr contig_5407 --chr contig_19343 --chr contig_3838 --chr contig_2518 --chr m_scaff_962 --chr contig_11951 --recode --recode-INFO-all --out AA261.HostPlant.contig_5345


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_5345 --recode --recode-INFO-all --out AA250.HostPlant.contig_5345

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA250.HostPlantOutlierScaffolds.recode.vcf
	--chr contig_5345
	--recode-INFO-all
	--out AA250.HostPlant.contig_5345
	--recode

After filtering, kept 250 out of 250 Individuals
Outputting VCF file...
After filtering, kept 11 out of a possible 109 Sites
Run Time = 3.00 seconds

```


I've uploaded these onto bluecrystal
```
pwd

/newhome/aj18951/1a_Aricia_agestis_PopGenomics/HaplotypeNetwork

ls

AA250.HostPlantOutlierScaffolds.recode.vcf  
AA254.ColHistOutlierScaffolds.recode.vcf

```

Find length of all the scaffolds: 
```
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/RefGenome
nano neutralloci ##create a file with one line per sequence name

while IFS= read -r line; do grep "$line" Aricia_agestis_Red_MESPA.fasta.fai; done < neutralloci
#IFS= is to ensure leading whitespace isn't deleted. 
```


Phase these individual vcf files
```
module load languages/java-jdk-11.0.3

module load apps/beagle-4.1

#once loaded most of the shared packages can be found here: /cm/shared/apps/ 

##Check that beagle is working

java -jar /cm/shared/apps/Beagle-4.1/beagle.jar


#Phase the first vcf file

java -jar /cm/shared/apps/Beagle-4.1/beagle.jar gt=AA250.HostPlant.contig_5345.recode.vcf out=AA250.contig_5345.Beagle

gunzip AA250.contig_5345.Beagle.vcf.gz 
```

Extract fasta sequences: 
```
module load apps/bcftools-1.8

for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf; printf '\n'; done > test.fa


##write haplotype1 to file
awk -F"_" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)}' hap1.fa

#Or do it all together: 
for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)substr($11,2,2)}'; printf '\n'; done > loc1.allele1

##Add ".1" to all the names
sed -E -i 's:^>.+:&.1:g' loc1.allele1


for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($2,1,1)substr($3,1,1)substr($4,1,1)substr($5,1,1)substr($6,1,1)substr($7,1,1)substr($8,1,1)substr($9,1,1)substr($10,1,1)substr($11,1,1)}'; printf '\n'; done > loc1.allele2

##Add ".2" to all the names
sed -E -i 's:^>.+:&.2:g' loc1.allele2

##create one file. And remove all white spaces

cat loc1.allele1 loc1.allele2 > loc1.haplotypes
sed -i '/^$/d' loc1.haplotypes 
```


The haplotype network will be constructed in R using pegas
```
getwd()
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/HaplotypNetworkOutliers

library(pegas)
four.colours <- c("darkorchid4", "darkorchid", "gold1", "gold3")  ##for the groups Old.Ger.HOD, newGer, newRR, South.oldRR)

loc1.fa <- read.FASTA("loc1.haplotypes")
h <- haplotype(loc1.fa)
net <- haploNet(h)
plot(net)

loc1.popnames <- read.table("loc1.popnames", header=T)  ##read in file with pop info
   Indiv pop HostPlant ColHist HaplotypeGroup
1 BAR1.1 BAR  Rockrose     new          newRR
2 BAR2.1 BAR  Rockrose     new          newRR
3 BAR3.1 BAR  Rockrose     new          newRR
4 BAR4.1 BAR  Rockrose     new          newRR
5 BAR5.1 BAR  Rockrose     new          newRR
6 BAR6.1 BAR  Rockrose     new          newRR

summary(loc1.popnames$HaplotypeGroup)
 HOD.oldGer      newGer       newRR South.oldRR 
         58         168          86         188 

ind.hap<-with(stack(setNames(attr(h, "index"), rownames(h))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))     ##create a table to count haplotypes in each catagory

ind.hap

     pop
hap   HOD.oldGer newGer newRR South.oldRR
  I           10     16    43          38
  II          48    109    43         120
  III          0      0     0           8
  IV           0     36     0          22
  V            0      7     0           0


plot(net, scale.ratio = 2, cex = 0.8, pie=ind.hap, bg=four.colours)  
legend(1,10, colnames(ind.hap), col=four.colours), pch=20)

##To add info about HOD

three.colours <- c("gold1", "gold3", "darkorchid")
plot(net, scale.ratio = 2, cex = 0.8, pie=ind.hap, bg=three.colours)
legend(1,10, colnames(ind.hap), col=three.colours, pch=20)
```


Try this for the rest of the contigs/scaffolds: 
```
vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_17378 --recode --recode-INFO-all --out AA250.HostPlant.contig_17378
After filtering, kept 1 out of a possible 109 Sites


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_5407 --recode --recode-INFO-all --out AA250.HostPlant.contig_5407
After filtering, kept 11 out of a possible 109 Sites


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_19343 --recode --recode-INFO-all --out AA250.HostPlant.contig_19343
After filtering, kept 6 out of a possible 109 Sites


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_3838 --recode --recode-INFO-all --out AA250.HostPlant.contig_3838
After filtering, kept 38 out of a possible 109 Sites


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr m_scaff_962 --recode --recode-INFO-all --out AA250.HostPlant.m_scaff_962
After filtering, kept 5 out of a possible 109 Sites


vcftools --vcf AA250.HostPlantOutlierScaffolds.recode.vcf --chr contig_11951 --recode --recode-INFO-all --out AA250.HostPlant.contig_11951
After filtering, kept 7 out of a possible 109 Sites
```


Make sure that all the loci are likely to be linked. Check if loci are within 10kb of each other: 
```
for i in $(ls *contig*vcf); do vcftools --vcf $i --thin 10000; done

##all outputs should filter down to 1 variant per file. Otherwise check how far apart they are. 
```

Phase with beagle
```
for i in $(ls *contig*vcf); do java -jar /cm/shared/apps/Beagle-4.1/beagle.jar gt=$i out=$i.beagle; done

##remove the file that has only 1 variant site

rm AA250.HostPlant.contig_17378.recode.vcf

##phase the file with scaffold
java -jar /cm/shared/apps/Beagle-4.1/beagle.jar gt=AA250.HostPlant.m_scaff_962.recode.vcf out=AA250.HostPlant.m_scaff_962.recode.vcf.beagle

##unzip all
for i in $(ls *gz); do gunzip $i; done
```

Extract fasta sequences for all: 
```
module load apps/bcftools-1.8

for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf; printf '\n'; done > test.fa


##write haplotype1 to file
awk -F"|" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)}' hap1.fa

#Or do it all together: 
for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)substr($11,2,2)}'; printf '\n'; done > loc1.allele1

##Add ".1" to all the names
sed -E -i 's:^>.+:&.1:g' loc1.allele1


for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($2,1,1)substr($3,1,1)substr($4,1,1)substr($5,1,1)substr($6,1,1)substr($7,1,1)substr($8,1,1)substr($9,1,1)substr($10,1,1)substr($11,1,1)}'; printf '\n'; done > loc1.allele2

##Add ".2" to all the names
sed -E -i 's:^>.+:&.2:g' loc1.allele2

##create one file. And remove all white spaces

cat loc1.allele1 loc1.allele2 > loc1.haplotypes
sed -i '/^$/d' loc1.haplotypes 
```

Create plots for every haplotype 
```
#scaff_962 - Locus1

scaff962 <- read.FASTA("m_scaff_962.haplotype")
h962 <- haplotype(scaff962)
net962 <- haploNet(h962)
ind.hap.962.HostPlant <- with(stack(setNames(attr(h962, "index"), rownames(h962))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap.962.HostPlant
hap	HOD.oldGer	newGer	newRR	South.oldRR	
	I	10	16	43	38
	II	48	109	43	120
	III	0	0	0	8
	IV	0	36	0	22
	V	0	7	0	0

pdf("scaff962.HPhaplotype.pdf")
plot(net962, scale.ratio=1, cex=0.8, pie=ind.hap.962.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap.962.HostPlant), col=four.colours, pch=20)
title("scaff962 - Locus1")
dev.off()


#contig11951 - Locus2
loc2 <- read.FASTA("contig_11951.haplotype")
h2 <- haplotype(loc2)
net2 <- haploNet(h2)
ind.hap2.HostPlant <- with(stack(setNames(attr(h2, "index"), rownames(h2))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap2.HostPlant

hap    HOD.oldGer newGer newRR South.oldRR
  I            29     20    43          56
  II            0     64     2          38
  III          16     16    41          38
  IV           13     25     0          33
  V             0     17     0          23
  VI            0     14     0           0
  VII           0      1     0           0
  VIII          0      4     0           0
  IX            0      1     0           0
  X             0      1     0           0
  XI            0      2     0           0
  XII           0      1     0           0
  XIII          0      2     0           0

pdf("Locus2.HPhaplotype.pdf")
plot(net2, scale.ratio=1, cex=0.8, pie=ind.hap2.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap2.HostPlant), col=four.colours, pch=20)
title("contig11951 - Locus2")
dev.off()


#contig5345 - Locus3
loc3 <- read.FASTA("contig_5345.haplotype")
h3 <- haplotype(loc3)
net3 <- haploNet(h3)
ind.hap3.HostPlant <- with(stack(setNames(attr(h3, "index"), rownames(h3))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap3.HostPlant
HOD.oldGer	newGer	newRR	South.oldRR
29	49	43	94
0	35	5	0
29	16	38	45
0	14	0	11
0	11	0	30
0	0	0	5
0	0	0	2
0	33	0	1
0	8	0	0
0	1	0	0
0	1	0	0

pdf("Locus3.HPhaplotype.pdf")
plot(net3, scale.ratio=1, cex=0.8, pie=ind.hap3.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap3.HostPlant), col=four.colours, pch=20)
title("contig5345 - Locus3")
dev.off()

##Locus4/contig17378 - only one variant so no haplotype network

#contig5407 - Locus5
loc5 <- read.FASTA("contig_5407.haplotype")
h5 <- haplotype(loc5)
net5 <- haploNet(h5)
ind.hap5.HostPlant <- with(stack(setNames(attr(h5, "index"), rownames(h5))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap5.HostPlant
HOD.oldGer	newGer	newRR	South.oldRR
0	0	18	0
29	100	62	132
12	0	6	0
8	0	0	0
9	0	0	16
0	25	0	20
0	0	0	10
0	6	0	10
0	5	0	0
0	8	0	0
0	1	0	0
0	7	0	0
0	8	0	0
0	1	0	0
0	2	0	0
0	1	0	0
0	1	0	0
0	1	0	0
0	1	0	0
0	1	0	0

pdf("Locus5.HPhaplotype.pdf")
plot(net5, scale.ratio=1, cex=0.8, pie=ind.hap5.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap5.HostPlant), col=four.colours, pch=20)
title("contig5407 - Locus5")
dev.off()


#contig19343 - Locus6
loc6 <- read.FASTA("contig_19343.haplotype")
h6 <- haplotype(loc6)
net6 <- haploNet(h6)
ind.hap6.HostPlant <- with(stack(setNames(attr(h6, "index"), rownames(h6))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap6.HostPlant

hap	HOD.oldGer	newGer	newRR	South.oldRR
I	0	0	19	0
II	29	41	24	66
III	0	43	28	53
IV	0	11	0	13
V	29	30	15	52
VI	0	24	0	4
VII	0	7	0	0
VIII	0	12	0	0

pdf("Locus6.HPhaplotype.pdf")
plot(net6, scale.ratio=1, cex=0.8, pie=ind.hap6.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap6.HostPlant), col=four.colours, pch=20)
title("contig19343 - Locus6")
dev.off()

#contig3838 - Locus7
loc7 <- read.FASTA("contig_3838.haplotype")
h7 <- haplotype(loc7)
net7 <- haploNet(h7)
ind.hap7.HostPlant <- with(stack(setNames(attr(h7, "index"), rownames(h7))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap7.HostPlant
>128 haplotypes - see BA supp table

pdf("Locus7.HPhaplotype.pdf")
plot(net7, scale.ratio=1, cex=0.8, pie=ind.hap7.HostPlant, bg=four.colours)
legend(1,1, colnames(ind.hap7.HostPlant), col=four.colours, pch=20)
title("contig3838 - Locus7")
dev.off()
```



### Neutral haplotypes

Compare these haplotype networks with a set of random loci from the genome. 

First extract the names of all the loci from the vcf file
```
On bluecp3: /newhome/aj18951/1a_Aricia_agestis_PopGenomics/HaplotypeNetwork

module load apps/vcftools-0.1.12b

grep "^contig" AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf | awk '{print $1}' | awk '!a[$0]++' >> contig.names.neutral20
shuf -n20 contig.names.neutral >> contig.names.neutral20

cat contig.names.neutral20 
contig_10072
contig_12160
contig_10272
contig_18932
contig_4506
contig_2942
contig_2229
contig_37637
contig_25996
contig_662
contig_58892
contig_38039
contig_75445
contig_10781
contig_1985
contig_30084
contig_31360
contig_3461
contig_916
contig_2334
```

Extract all the contigs/scaffolds: 
```
vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_10072 --recode --recode-INFO-all --out AA250.HostPlant.contig_10072
After filtering, kept 16 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_12160 --recode --recode-INFO-all --out AA250.HostPlant.contig_12160
After filtering, kept 8 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_10272 --recode --recode-INFO-all --out AA250.HostPlant.contig_10272
After filtering, kept 3 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_18932 --recode --recode-INFO-all --out AA250.HostPlant.contig_18932
After filtering, kept 3 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_4506 --recode --recode-INFO-all --out AA250.HostPlant.contig_4506
After filtering, kept 7 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_2942 --recode --recode-INFO-all --out AA250.HostPlant.contig_2942
After filtering, kept 33 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_2229 --recode --recode-INFO-all --out AA250.HostPlant.contig_2229
After filtering, kept 6 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_37637 --recode --recode-INFO-all --out AA250.HostPlant.contig_37637
After filtering, kept 1 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_25996 --recode --recode-INFO-all --out AA250.HostPlant.contig_25996
After filtering, kept 3 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_662 --recode --recode-INFO-all --out AA250.HostPlant.contig_662
After filtering, kept 2 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_58892 --recode --recode-INFO-all --out AA250.HostPlant.contig_58892
After filtering, kept 13 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_38039 --recode --recode-INFO-all --out AA250.HostPlant.contig_38039
After filtering, kept 4 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_75445 --recode --recode-INFO-all --out AA250.HostPlant.contig_75445
After filtering, kept 2 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_10781 --recode --recode-INFO-all --out AA250.HostPlant.contig_10781
After filtering, kept 17 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_1985 --recode --recode-INFO-all --out AA250.HostPlant.contig_1985
After filtering, kept 9 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_30084 --recode --recode-INFO-all --out AA250.HostPlant.contig_30084
After filtering, kept 1 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_31360 --recode --recode-INFO-all --out AA250.HostPlant.contig_31360
After filtering, kept 5 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_3461 --recode --recode-INFO-all --out AA250.HostPlant.contig_3461
After filtering, kept 7 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_916 --recode --recode-INFO-all --out AA250.HostPlant.contig_916
After filtering, kept 1 out of a possible 18338 Sites

vcftools --vcf AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf --chr contig_2334 --recode --recode-INFO-all --out AA250.HostPlant.contig_2334
After filtering, kept 1 out of a possible 18338 Sites
```

Make sure that all the loci are likely to be linked. Check if loci are within 10kb of each other: 
```
for i in $(cat contig.names.neutral20); do vcftools --vcf $i --thin 10000; done

##all outputs should filter down to 1 variant per file. Otherwise check how far apart they are. 
```

Phase with beagle
```
module load languages/java-jdk-11.0.3

module load apps/beagle-4.1

#once loaded most of the shared packages can be found here: /cm/shared/apps/ 

##Check that beagle is working

java -jar /cm/shared/apps/Beagle-4.1/beagle.jar

## Phasing
for i in $(cat contig.names.neutral20); do java -jar /cm/shared/apps/Beagle-4.1/beagle.jar gt=$i out=$i.beagle; done

##remove the file that has only 1 variant site

rm AA250.HostPlant.contig_17378.recode.vcf

##move all to a new folder for Neutral Haplotypes
for i in $(cat contig.names.neutral20); do mv $i NeutralHaplotypes/; done
mv *gz NeutralHaplotypes/


##unzip all
pwd 
/newhome/aj18951/1a_Aricia_agestis_PopGenomics/HaplotypeNetwork/Neutral.Haplotypes
for i in $(ls *gz); do gunzip $i; done
```

Extract fasta sequences for all: 
```
module load apps/bcftools-1.8

for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf; printf '\n'; done > test.fa


##write haplotype1 to file
awk -F"|" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)}' hap1.fa

#Or do it all together: 
for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($1,1,1)substr($2,2,2)substr($3,2,2)substr($4,2,2)substr($5,2,2)substr($6,2,2)substr($7,2,2)substr($8,2,2)substr($9,2,2)substr($10,2,2)substr($11,2,2)}'; printf '\n'; done > loc1.allele1

##Add ".1" to all the names
sed -E -i 's:^>.+:&.1:g' loc1.allele1


for i in $(bcftools query -l AA250.contig_5345.Beagle.vcf); do printf '>'${i}'\n'; bcftools query -s ${i} -f '[%TGT]' AA250.contig_5345.Beagle.vcf | awk -F"|" '{print substr($2,1,1)substr($3,1,1)substr($4,1,1)substr($5,1,1)substr($6,1,1)substr($7,1,1)substr($8,1,1)substr($9,1,1)substr($10,1,1)substr($11,1,1)}'; printf '\n'; done > loc1.allele2

##Add ".2" to all the names
sed -E -i 's:^>.+:&.2:g' loc1.allele2

##create one file. And remove all white spaces

cat loc1.allele1 loc1.allele2 > loc1.haplotypes
sed -i '/^$/d' loc1.haplotypes 
```

Simple script for this: [ExtractHaplotypes.sh](https://github.com/alexjvr1/AriciaAgestis_PopGenMS/blob/master/ExtractHaplotypes.sh)

Create plots for every haplotype. 

First copy all the fasta files onto the mac
```
pwd /Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/HaplotypNetworkOutliers/NeutralLoci

scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/HaplotypeNetwork/Neutral.Haplotypes/popnames.info .
scp bluecp3:/newhome/aj18951/1a_Aricia_agestis_PopGenomics/HaplotypeNetwork/Neutral.Haplotypes/*loc1.allele* .
```

Networks will be drawn in R
```
library(pegas)
loc1.fa <- read.FASTA("AA250.HostPlant.contig_10072.recode.vcf.beagle.vcf.loc1.haplotypes")
h <- haplotype(loc1.fa)
net <- haploNet(h)
plot(net)

loc1.popnames <- read.table("loc1.popnames", header=T)  ##read in file with pop info. The names are doubled because there are two alleles. ie. all names + all names (not interleaved)
```


Networks will be drawn in R
```
getwd()
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/HaplotypNetworkOutliers/NeutralLoci

library(pegas)
four.colours <- c("darkorchid4", "darkorchid", "gold1", "gold3")  ##for the groups Old.Ger.HOD, newGer, newRR, South.oldRR
loc1.popnames <- read.table("Neutral.popnames", header=T)

#contig10072 - NeutralLocus1
loc1 <- read.FASTA("AA250.HostPlant.contig_10072.recode.vcf.beagle.vcf.loc1.haplotypes")
h1 <- haplotype(loc1)
net1 <- haploNet(h1)
ind.hap1.HostPlant <- with(stack(setNames(attr(h1, "index"), rownames(h1))), table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap1.HostPlant

plot(net, scale.ratio = 2, cex = 0.8, pie=ind.hap, bg=four.colours)  
legend(1,10, colnames(ind.hap), col=four.colours), pch=20)



#contig12160 - Locus2

loc2 <- read.FASTA("AA250.HostPlant.contig_12160.recode.vcf.beagle.vcf.loc1.haplotypes")
h2 <- haplotype(loc2)
net2 <- haploNet(h2)
ind.hap2.Neutral <- with(stack(setNames(attr(h2, "index"), rownames(h2))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap2.Neutral
hap	HOD.oldGer	newGer	newRR	South.oldRR	
	I	0	17	38	38
II	30	85	38	98
III	6	0	20	0
IV	28	7	0	18
V	0	18	0	14
VI	0	0	0	3
VII	0	15	0	21
VIII	0	6	0	0
IX	0	14	0	0
X	0	3	0	0
XI	0	1	0	0
XII	0	1	0	0
XIII	0	1	0	0
XIV	0	1	0	0
XV	0	1	0	0

pdf("NeutralLocus2.Haplotype.pdf")
plot(net2, scale.ratio=1, cex=0.8, pie=ind.hap2.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap2.Neutral), col=four.colours, pch=20)
title("scaff12160 - Locus2")
dev.off()


#contig10272 - Locus3

loc3 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h3 <- haplotype(loc3)
net3 <- haploNet(h3)
ind.hap3.Neutral <- with(stack(setNames(attr(h3, "index"), rownames(h3))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap3.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus3.Haplotype.pdf")
plot(net3, scale.ratio=1, cex=0.8, pie=ind.hap3.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap3.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus3")
dev.off()


#contig10272 - Locus4

loc4 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h4 <- haplotype(loc4)
net4 <- haploNet(h4)
ind.hap4.Neutral <- with(stack(setNames(attr(h4, "index"), rownames(h4))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap4.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus4.Haplotype.pdf")
plot(net4, scale.ratio=1, cex=0.8, pie=ind.hap4.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap4.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus4")
dev.off()

#contig10272 - Locus5

loc5 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h5 <- haplotype(loc5)
net5 <- haploNet(h5)
ind.hap5.Neutral <- with(stack(setNames(attr(h5, "index"), rownames(h5))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap5.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus5.Haplotype.pdf")
plot(net5, scale.ratio=1, cex=0.8, pie=ind.hap5.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap5.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus5")
dev.off()

#contig10272 - Locus6

loc6 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h6 <- haplotype(loc6)
net6 <- haploNet(h6)
ind.hap6.Neutral <- with(stack(setNames(attr(h6, "index"), rownames(h3))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap6.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus6.Haplotype.pdf")
plot(net6, scale.ratio=1, cex=0.8, pie=ind.hap6.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap6.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus6")
dev.off()

#contig10272 - Locus7

loc7 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h7 <- haplotype(loc7)
net7 <- haploNet(h7)
ind.hap7.Neutral <- with(stack(setNames(attr(h7, "index"), rownames(h7))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap7.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus7.Haplotype.pdf")
plot(net7, scale.ratio=1, cex=0.8, pie=ind.hap3.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap7.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus7")
dev.off()

#contig10272 - Locus8

loc8 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h8 <- haplotype(loc8)
net8 <- haploNet(h8)
ind.hap8.Neutral <- with(stack(setNames(attr(h8, "index"), rownames(h8))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap8.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus8.Haplotype.pdf")
plot(net8, scale.ratio=1, cex=0.8, pie=ind.hap8.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap8.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus8")
dev.off()

#contig10272 - Locus9

loc9 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h9 <- haplotype(loc9)
net9 <- haploNet(h9)
ind.hap9.Neutral <- with(stack(setNames(attr(h9, "index"), rownames(h3))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap9.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus9.Haplotype.pdf")
plot(net9, scale.ratio=1, cex=0.8, pie=ind.hap9.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap9.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus9")
dev.off()

#contig10272 - Locus10

loc10 <- read.FASTA("AA250.HostPlant.contig_10272.recode.vcf.beagle.vcf.loc1.haplotypes")
h10 <- haplotype(loc10)
net10 <- haploNet(h10)
ind.hap10.Neutral <- with(stack(setNames(attr(h10, "index"), rownames(h3))),table(hap=ind, pop=loc1.popnames$HaplotypeGroup))

ind.hap10.Neutral
hap   HOD.oldGer newGer newRR South.oldRR
  I           30     19    48          59
  II          34    108    48         107
  III          0     17     0          26
  IV           0     23     0           0
  V            0      1     0           0
  VI           0      2     0           0

pdf("NeutralLocus10.Haplotype.pdf")
plot(net10, scale.ratio=1, cex=0.8, pie=ind.hap10.Neutral, bg=four.colours)
legend(1,1, colnames(ind.hap10.Neutral), col=four.colours, pch=20)
title("scaff10272 - Locus10")
dev.off()


```
