# Estimate divergence time between adaptive loci 


https://helda.helsinki.fi/bitstream/handle/10138/298668/journal.pgen.1007796.pdf?sequence=1&isAllowed=y


Here I aim to test: 

1) age of the outlier loci

2) origin of the outlier loci 


## fineRADstructure of outlier loci

fineRADstructure looks at the recent coancestry of loci. I can rerun this analysis using only the outlier loci identified by Bayescan. 


Filter vcf file for scaffolds that contain the outlier loci: 

HostPlant
```
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/fineRADstructure

vcftools --vcf AA261.newnames.vcf --chr contig_5345 --chr contig_17378 --chr contig_5407 --chr contig_19343 --chr contig_3838 --chr contig_2518 --chr m_scaff_962 --chr contig_11951 --recode --recode-INFO-all --out AA261.HostPlantOutlierScaffolds

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.newnames.vcf
	--chr contig_11951
	--chr contig_17378
	--chr contig_19343
	--chr contig_2518
	--chr contig_3838
	--chr contig_5345
	--chr contig_5407
	--chr m_scaff_962
	--recode-INFO-all
	--out AA261.HostPlantOutlierScaffolds
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 109 out of a possible 31381 Sites
Run Time = 4.00 seconds
```


ColHist
```
vcftools --vcf AA261.newnames.vcf --chr contig_1784 --chr contig_5345 --chr contig_16912 --chr contig_3360 --chr contig_19343 --chr contig_59667 --chr contig_24918 --chr contig_3838 --chr contig_18281 --chr contig_1883 --chr contig_19566 --chr contig_19564 --recode --recode-INFO-all --out AA261.ColHistOutlierScaffolds


VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.newnames.vcf
	--chr contig_16912
	--chr contig_1784
	--chr contig_18281
	--chr contig_1883
	--chr contig_19343
	--chr contig_19564
	--chr contig_19566
	--chr contig_24918
	--chr contig_3360
	--chr contig_3838
	--chr contig_5345
	--chr contig_59667
	--recode-INFO-all
	--out AA261.ColHistOutlierScaffolds
	--recode

After filtering, kept 261 out of 261 Individuals
Outputting VCF file...
After filtering, kept 169 out of a possible 31381 Sites
Run Time = 4.00 seconds
```

Convert to fineRADstructure input
```
./vcf2hapmatrix.py AA261.ColHistOutlierScaffolds.recode.vcf > AA261.ColHistOutliers.fineradstr
./vcf2hapmatrix.py AA261.HostPlantOutlierScaffolds.recode.vcf > AA261.HostPlantOutliers.fineradstr


```



## Haplotype network of outlier RADtags

Use A. artaxerxes as an outlier. And perhaps a further ancestor with WGS?? 

Identify the scaffolds and region where the outlier loci are. I can see this in my locus.stats file in R

```
getwd()
/Users/alexjvr/2018.postdoc/BrownArgus_2018/201902_DataAnalysis/AdaptationHostPlant

> outliers.HostPlant
                V1    V2                 V3       Agene1 LG_assigned Hmel.pos
2072   m_scaff_962 19860  m_scaff_962_19860 HMEL003137g1          01  3418280
10676 contig_11951 10614 contig_11951_10614         <NA>        <NA>       NA
11041  contig_5345   510    contig_5345_510         <NA>        <NA>       NA
11042  contig_5345   511    contig_5345_511         <NA>        <NA>       NA
11050  contig_5345   587    contig_5345_587         <NA>        <NA>       NA
12108 contig_17378  3882  contig_17378_3882         <NA>        <NA>       NA
16942  contig_5407  5785   contig_5407_5785         <NA>        <NA>       NA
17001 contig_19343  4711  contig_19343_4711         <NA>        <NA>       NA
21375  contig_3838  3376   contig_3838_3376         <NA>        <NA>       NA
21377  contig_3838  3378   contig_3838_3378         <NA>        <NA>       NA
21378  contig_3838  3382   contig_3838_3382         <NA>        <NA>       NA
21379  contig_3838  3408   contig_3838_3408         <NA>        <NA>       NA
26646  contig_2518 13094  contig_2518_13094         <NA>        <NA>       NA
      Bayescan    LG.pos     LFMM.zs   LFMM.invP chr     pos   pvalue
2072  0.075214 1_3418280 -1.29200000 0.852724236   1 3418280 0.075214
10676 0.070423     NA_NA -0.92842900 0.702313781  NA      NA 0.070423
11041 0.090865     NA_NA  0.00963256 0.008620760  NA      NA 0.090865
11042 0.092381     NA_NA  0.00433303 0.003877951  NA      NA 0.092381
11050 0.095242     NA_NA  0.06568980 0.058737760  NA      NA 0.095242
12108 0.082410     NA_NA  1.90364000 0.967262743  NA      NA 0.082410
16942 0.071641     NA_NA  1.39249000 0.881697843  NA      NA 0.071641
17001 0.095215     NA_NA -1.49713000 0.906909006  NA      NA 0.095215
21375 0.095010     NA_NA -0.75066800 0.600220766  NA      NA 0.095010
21377 0.092891     NA_NA -0.73351900 0.589366742  NA      NA 0.092891
21378 0.095510     NA_NA -0.76999000 0.612241154  NA      NA 0.095510
21379 0.089935     NA_NA -0.71849300 0.579714051  NA      NA 0.089935
26646 0.082731     NA_NA -1.31554000 0.859955662  NA      NA 0.082731
      BayescanColHist
2072         0.022119
10676        0.022163
11041        0.083876
11042        0.083073
11050        0.085587
12108        0.021469
16942        0.031380
17001        0.083373
21375        0.105340
21377        0.104240
21378        0.109400
21379        0.100140
26646        0.073927

```

Here I can see the scaffold and approximate position of the RAD tag of interest. Extract all these tags from the mapped bam files for the population data. 
This is done on the blue crystal server. 

```
pwd
#/newhome/aj18951/1a_Aricia_agestis_PopGenomics/mapped

module load apps/samtools-1.8
module load apps/bamtools-1.8
for i in $(ls *gz.bam); do samtools view -b -F4 $i m_scaff_962:19000-20000 >> $i.m_scaff_962.subset.bam; done
##remember -F4 to filter for properly paired and mapped reads only. 

```

And a script if you provide a list of scaffolds and positions
```

```
