# GEMMA

I'll use [GEMMA](https://github.com/genetics-statistics/GEMMA/blob/master/doc/manual.pdf) to estimate the proportion of variance explained by the genetic data (PVE), and identify the most 
important loci associated with host plant choice for the A.agestis dataset. 


Gemma requires complete datasets (no missing data), hence I have to impute missing genotypes in the dataset. 


## Imputing genotypes in ddRAD dataset



## Convert genotype file to plink 

Gemma requires plink input files. However, plink doesn't recognise the chromosome names used in our reference genomes (e.g. contig_1234, and scaffold_1234). 

First rename all these regions in the vcf file: 
```
##Exctract all the chromosome names from the vcf file: 

grep "##contig=" AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf > allcontigs

head allcontigs
##contig=<ID=m_scaff_1=17153>
##contig=<ID=m_scaff_2=7134>
##contig=<ID=m_scaff_3=9191>
##contig=<ID=m_scaff_4=8133>
##contig=<ID=m_scaff_5=10012>
##contig=<ID=m_scaff_6=14523>
##contig=<ID=m_scaff_7=17602>
##contig=<ID=m_scaff_8=2819>
##contig=<ID=m_scaff_9=2835>
##contig=<ID=m_scaff_10=5218>

##extract the name only and index the lines (NR). Bcftools requires old names \t new names

awk -F "=" '{print $3"\t"NR}' allcontigs > contig.names

##bcftools 
module load apps/bcftools-1.8

bcftools annotate --rename-chrs contig.names AA261.0.5miss.9popsMerged.MAF0.05.recode.vcf > AA261.forGemma.vcf

##and check that this has worked correctly: 

grep "##contig" AA261.forGemma.vcf |head
##contig=<ID=1,length=17153>
##contig=<ID=2,length=7134>
##contig=<ID=3,length=9191>
##contig=<ID=4,length=8133>
##contig=<ID=5,length=10012>
##contig=<ID=6,length=14523>
##contig=<ID=7,length=17602>
##contig=<ID=8,length=2819>
##contig=<ID=9,length=2835>
##contig=<ID=10,length=5218>

grep "##contig" AA261.forGemma.vcf |tail
##contig=<ID=93559,length=18742>
##contig=<ID=93560,length=4109>
##contig=<ID=93561,length=14916>
##contig=<ID=93562,length=14921>
##contig=<ID=93563,length=14924>
##contig=<ID=93564,length=14927>
##contig=<ID=93565,length=14926>
##contig=<ID=93566,length=14928>
##contig=<ID=93567,length=14927>
##contig=<ID=93568,length=18020>
```

This should now convert to plink with no errors: 
Check number of indivs and SNPs
```
module load apps/vcftools-0.1.12b

vcftools --vcf AA261.forGemma.vcf --plink --out AA261.forGemma.plink

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf AA261.forGemma.vcf
	--out AA261.forGemma.plink
	--plink

After filtering, kept 261 out of 261 Individuals
Writing PLINK PED and MAP files ... 
Done.
After filtering, kept 18338 out of a possible 18338 Sites
Run Time = 2.00 seconds

```

Generate binary plink files: 
```
## this doesn't work with plink2, so use the older version
## to see and delete modules loaded on your bluecrystal environment: 

module list

Currently Loaded Modulefiles:
  1) shared                        5) default-environment           9) languages/perl-5.14.2        13) apps/bcftools-1.8
  2) dot                           6) languages/java-jdk-1.8.0-66  10) tools/zlib-1.2.8             14) apps/plink-2.00
  3) torque/4.2.4.1                7) languages/java-jdk-1.7.0-40  11) apps/vcftools-0.1.12b        15) apps/plink-1.90
  4) moab/7.2.9                    8) languages/R-3.0.2            12) languages/gcc-6.1

module del apps/plink-2.00

##check that the correct plink is being used: 

plink 
PLINK v1.90b3f 64-bit (2 Mar 2015)         https://www.cog-genomics.org/plink2
(C) 2005-2015 Shaun Purcell, Christopher Chang   GNU General Public License v3

  plink [input flag(s)...] {command flag(s)...} {other flag(s)...}
  plink --help {flag name(s)...}

Commands include --make-bed, --recode, --flip-scan, --merge-list,
--write-snplist, --list-duplicate-vars, --freqx, --missing, --test-mishap,
--hardy, --mendel, --ibc, --impute-sex, --indep-pairphase, --r2, --show-tags,
--blocks, --distance, --genome, --homozyg, --make-rel, --make-grm-gz,
--rel-cutoff, --cluster, --pca, --neighbour, --ibs-test, --regress-distance,
--model, --bd, --gxe, --logistic, --dosage, --lasso, --test-missing,
--make-perm-pheno, --tdt, --qfam, --annotate, --clump, --gene-report,
--meta-analysis, --epistasis, --fast-epistasis, and --score.

'plink --help | more' describes all functions (warning: long).

```

Conversion: 

Plink interprets numbers as chromosomes, hence expects only 23 (as in humans). We can specify that they are scaffolds by adding e.g. "contig" before the number (ie. back to the vcf format), but I have to do this directly on the plink file. 


```
awk -F "\t" '{print "contig."$1"\t""contig."$2"\t"$3"\t"$4}' A261.forGemma.plink.map > A261.forGemma.plink.map2

#Check that this has worked, and then move file to correct .map

mv A261.forGemma.plink.map2 A261.forGemma.plink.map

#Now we can convert: 

plink --file A261.forGemma.plink --make-bed --allow-extra-chr --out A261.forGemma.plink
```




## Add phenotype data 

Need a text file with three columns: family ID, indiv ID, and phenotype. See [PLINK manual](http://zzz.bwh.harvard.edu/plink/data.shtml). 

We have phenotypes for most but not all of the genotyped individuals (250/261 indivs). Maaike sent the Phenotype file on 13 June 2019 (HPPdata.xlsx). 


The phenotype data will be read as column6 in the .fam file from plink. This needs to be replaced by the input phenotype data. Here I will use a binary encoding of RR (0) and GM (1) host plant choice. Missing data can be specified with NA or -9. 
```
awk 'FNR==NR{a[NR]=$3;next}{$6=a[FNR]}1' A261.Pheno.forGemma A261.forGemma.plink.fam > A261.forGemma.plink.fam
```

Similarly, family information will be taken from column 2 of the .fam file. This can be replaced with the population information. 
```
awk 'FNR==NR{a[NR]=$6;next}{$2=a[FNR]}1' A261.Pheno.forGemma A261.forGemma.plink.fam > A261.forGemma.plink.fam
```


